<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>DESeq_Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="quarto_files/libs/clipboard/clipboard.min.js"></script>
<script src="quarto_files/libs/quarto-html/quarto.js"></script>
<script src="quarto_files/libs/quarto-html/popper.min.js"></script>
<script src="quarto_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="quarto_files/libs/quarto-html/anchor.min.js"></script>
<link href="quarto_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="quarto_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="quarto_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="quarto_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="quarto_files/libs/bootstrap/bootstrap-469117c77a8604b9ffbe744ae1d88387.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#import-data" id="toc-import-data" class="nav-link active" data-scroll-target="#import-data">Import Data</a></li>
  <li><a href="#deseq-initialization-visualization" id="toc-deseq-initialization-visualization" class="nav-link" data-scroll-target="#deseq-initialization-visualization">DESeq Initialization &amp; Visualization</a>
  <ul class="collapse">
  <li><a href="#star-rnaseq-dds" id="toc-star-rnaseq-dds" class="nav-link" data-scroll-target="#star-rnaseq-dds">STAR-RNAseq-dds</a></li>
  <li><a href="#salmon-rnaseq-dds-mus-genes" id="toc-salmon-rnaseq-dds-mus-genes" class="nav-link" data-scroll-target="#salmon-rnaseq-dds-mus-genes">Salmon-RNAseq-dds-Mus-Genes</a></li>
  <li><a href="#salmon-rnaseq-dds-mus-premrna" id="toc-salmon-rnaseq-dds-mus-premrna" class="nav-link" data-scroll-target="#salmon-rnaseq-dds-mus-premrna">Salmon-RNAseq-dds-Mus-PremRNA</a></li>
  <li><a href="#star-ttseq-dds" id="toc-star-ttseq-dds" class="nav-link" data-scroll-target="#star-ttseq-dds">STAR-TTseq-dds</a></li>
  <li><a href="#bowtie2-ttseq-dds" id="toc-bowtie2-ttseq-dds" class="nav-link" data-scroll-target="#bowtie2-ttseq-dds">Bowtie2-TTseq-dds</a></li>
  <li><a href="#results-instances-low-vs-wt" id="toc-results-instances-low-vs-wt" class="nav-link" data-scroll-target="#results-instances-low-vs-wt">Results Instances LOW vs WT</a></li>
  <li><a href="#ma-plots" id="toc-ma-plots" class="nav-link" data-scroll-target="#ma-plots">MA plots</a></li>
  <li><a href="#manual-ma-plots" id="toc-manual-ma-plots" class="nav-link" data-scroll-target="#manual-ma-plots">Manual MA plots</a></li>
  <li><a href="#qq-plot" id="toc-qq-plot" class="nav-link" data-scroll-target="#qq-plot">QQ plot</a></li>
  </ul></li>
  <li><a href="#rna-seq-analysis" id="toc-rna-seq-analysis" class="nav-link" data-scroll-target="#rna-seq-analysis">RNA-Seq Analysis</a>
  <ul class="collapse">
  <li><a href="#star" id="toc-star" class="nav-link" data-scroll-target="#star">STAR</a></li>
  <li><a href="#salmon" id="toc-salmon" class="nav-link" data-scroll-target="#salmon">Salmon</a></li>
  <li><a href="#c1-removed" id="toc-c1-removed" class="nav-link" data-scroll-target="#c1-removed">C1-Removed</a></li>
  </ul></li>
  <li><a href="#tt-seq-analysis" id="toc-tt-seq-analysis" class="nav-link" data-scroll-target="#tt-seq-analysis">TT-Seq Analysis</a>
  <ul class="collapse">
  <li><a href="#star-1" id="toc-star-1" class="nav-link" data-scroll-target="#star-1">STAR</a></li>
  <li><a href="#bowtie2" id="toc-bowtie2" class="nav-link" data-scroll-target="#bowtie2">Bowtie2</a></li>
  <li><a href="#c1-removed-bowtie2" id="toc-c1-removed-bowtie2" class="nav-link" data-scroll-target="#c1-removed-bowtie2">C1-Removed-Bowtie2</a></li>
  </ul></li>
  <li><a href="#karyoploter" id="toc-karyoploter" class="nav-link" data-scroll-target="#karyoploter">karyoploteR</a>
  <ul class="collapse">
  <li><a href="#compare_chr_ttsample" id="toc-compare_chr_ttsample" class="nav-link" data-scroll-target="#compare_chr_ttsample">Compare_Chr_TTsample</a></li>
  <li><a href="#compare_chr_2sample" id="toc-compare_chr_2sample" class="nav-link" data-scroll-target="#compare_chr_2sample">Compare_Chr_2sample</a></li>
  <li><a href="#compare_chr_1max_2sample" id="toc-compare_chr_1max_2sample" class="nav-link" data-scroll-target="#compare_chr_1max_2sample">Compare_Chr_1max_2sample</a></li>
  <li><a href="#ttdegs-gene-density" id="toc-ttdegs-gene-density" class="nav-link" data-scroll-target="#ttdegs-gene-density">TTDegs &amp; Gene Density</a></li>
  </ul></li>
  <li><a href="#limma" id="toc-limma" class="nav-link" data-scroll-target="#limma">limma</a>
  <ul class="collapse">
  <li><a href="#fold2change-rna-vs-tt-plot" id="toc-fold2change-rna-vs-tt-plot" class="nav-link" data-scroll-target="#fold2change-rna-vs-tt-plot">Fold2Change RNA vs TT Plot</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">DESeq_Analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="import-data" class="level2">
<h2 class="anchored" data-anchor-id="import-data">Import Data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tximport)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tximportData)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:limma':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biobase'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:MatrixGenerics':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    anyMissing, rowMedians</code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:matrixStats':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:GenomicRanges':

    intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:IRanges':

    collapse, desc, intersect, setdiff, slice, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:S4Vectors':

    first, intersect, rename, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    combine, intersect, setdiff, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(txdbmaker)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'txdbmaker'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:GenomicFeatures':

    browseUCSCtrack, getChromInfoFromBiomart, makeFDbPackageFromUCSC,
    makeFeatureDbFromUCSC, makePackageName, makeTxDb,
    makeTxDbFromBiomart, makeTxDbFromEnsembl, makeTxDbFromGFF,
    makeTxDbFromGRanges, makeTxDbFromUCSC, makeTxDbPackage,
    makeTxDbPackageFromBiomart, makeTxDbPackageFromUCSC,
    supportedMiRBaseBuildValues, supportedUCSCFeatureDbTables,
    supportedUCSCFeatureDbTracks, supportedUCSCtables,
    UCSCFeatureDbTableSchema</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ lubridate 1.9.4     ✔ tibble    3.3.0
✔ purrr     1.0.4     ✔ tidyr     1.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ lubridate::%within%()   masks IRanges::%within%()
✖ dplyr::collapse()       masks IRanges::collapse()
✖ dplyr::combine()        masks Biobase::combine(), BiocGenerics::combine()
✖ dplyr::count()          masks matrixStats::count()
✖ dplyr::desc()           masks IRanges::desc()
✖ tidyr::expand()         masks S4Vectors::expand()
✖ dplyr::filter()         masks stats::filter()
✖ dplyr::first()          masks S4Vectors::first()
✖ dplyr::lag()            masks stats::lag()
✖ ggplot2::Position()     masks BiocGenerics::Position(), base::Position()
✖ purrr::reduce()         masks GenomicRanges::reduce(), IRanges::reduce()
✖ dplyr::rename()         masks S4Vectors::rename()
✖ lubridate::second()     masks S4Vectors::second()
✖ lubridate::second&lt;-()   masks S4Vectors::second&lt;-()
✖ AnnotationDbi::select() masks dplyr::select()
✖ dplyr::slice()          masks IRanges::slice()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'reshape2'

The following object is masked from 'package:tidyr':

    smiths</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(karyoploteR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: regioneR</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>bowtie_matrix <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/home/franck/SCRATCH/fpan/Genome_Index/chR_index_files/BowtieQ30_counts_matrix.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>star_matrix <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/home/franck/SCRATCH/fpan/Genome_Index/chR_index_files/STARQ30_counts_matrix.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>description_file <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"/home/franck/SCRATCH/fpan/Genome_Index/chR_index_files/Ashish.SampleDescription.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>gtf_file <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">"/home/franck/SCRATCH/fpan/Genome_Index/chR_index_files/chR_TrExAdded.gtf"</span>)</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>latest_gtf <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">"/home/franck/SCRATCH/fpan/Genome_Index/chR_index_files/rsem_9_06_25_filtered.gtf"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>description_file<span class="sc">$</span>Genotype <span class="ot">&lt;-</span> <span class="fu">factor</span>(description_file<span class="sc">$</span>Genotype, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"WT"</span>, <span class="st">"LOW"</span>, <span class="st">"GapRWT"</span>))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(bowtie_matrix)[<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(bowtie_matrix))] <span class="ot">&lt;-</span> </span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gsub</span>(<span class="st">"B.*$"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(bowtie_matrix)[<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(bowtie_matrix))])</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(star_matrix)[<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(star_matrix))] <span class="ot">&lt;-</span> </span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gsub</span>(<span class="st">"Star.*$"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(star_matrix)[<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(star_matrix))])</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(description_file) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(star_matrix)[<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(star_matrix))]</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>description_file<span class="sc">$</span>sample_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(description_file)</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(star_matrix) <span class="ot">&lt;-</span> star_matrix<span class="sc">$</span>X</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(bowtie_matrix) <span class="ot">&lt;-</span> bowtie_matrix<span class="sc">$</span>X</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>star_matrix <span class="ot">&lt;-</span> star_matrix[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>bowtie_matrix <span class="ot">&lt;-</span> bowtie_matrix[, <span class="sc">-</span><span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(star_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  RNA_S11589Nr53 RNA_S11589Nr54 RNA_S11589Nr55 RNA_S11589Nr56
18S                       656533         321555         427012         626102
28S                      1505491         752992        1015094        1508968
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008              0              0              0              0
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr57 RNA_S11589Nr58 RNA_S11589Nr59 RNA_S11589Nr60
18S                       510606         497955         384400         473432
28S                      1285117        1271717         925302        1368124
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008              8              0              0              0
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr61 RNA_S11589Nr62 RNA_S11589Nr63 RNA_S11589Nr64
18S                       122439         158732         212367         117554
28S                       342581         439200         411157         301812
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008              0              0              0              0
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr69 RNA_S11589Nr70 RNA_S11589Nr71 RNA_S11589Nr72
18S                     11216638       12437014       10571964       11959462
28S                     20975929       24396074       19971006       21545826
Droso_FBgn0000003              4              0              2              0
Droso_FBgn0000008             64             33             61             90
Droso_FBgn0000014              2              0              0              2
Droso_FBgn0000015              0              2              0              2
                  RNA_S11589Nr73 RNA_S11589Nr74 RNA_S11589Nr75 RNA_S11589Nr76
18S                     12747183       14415866       11706650       12358688
28S                     24449436       29235185       21485212       21723967
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008             93             41             52             43
Droso_FBgn0000014              2              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr77 RNA_S11589Nr78 RNA_S11589Nr79 RNA_S11589Nr80
18S                     14730053       12101005        8481735        9545585
28S                     27060439       24025274       15339482       15719132
Droso_FBgn0000003              2              2              2              2
Droso_FBgn0000008            128             68             68            120
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              2              0              0              2
                  RNA_S11589Nr94 RNA_S11589Nr95 RNA_S11589Nr96 RNA_S11589Nr97
18S                       378741         334218         367715         654221
28S                       951003         960597         824511        1327514
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008              2              0              0              0
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              0              2              0              0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bowtie_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  RNA_S11589Nr53 RNA_S11589Nr54 RNA_S11589Nr55 RNA_S11589Nr56
18S                       649245         315216         418510         613063
28S                      1458468         703607         953357        1439333
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008              0              0              0              0
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr57 RNA_S11589Nr58 RNA_S11589Nr59 RNA_S11589Nr60
18S                       499555         489424         374485         468088
28S                      1231639        1200151         871643        1327091
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008              7              0              0              0
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr61 RNA_S11589Nr62 RNA_S11589Nr63 RNA_S11589Nr64
18S                       122364         157729         213119         114863
28S                       337238         421435         398415         288291
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008              0              0              0              0
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              1              0              0              0
                  RNA_S11589Nr69 RNA_S11589Nr70 RNA_S11589Nr71 RNA_S11589Nr72
18S                     10882366       12053086       10268220       11573945
28S                     19958365       22646075       18784274       20446596
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008             60             31             56             86
Droso_FBgn0000014              2              0              0              2
Droso_FBgn0000015              0              2              0              2
                  RNA_S11589Nr73 RNA_S11589Nr74 RNA_S11589Nr75 RNA_S11589Nr76
18S                     12361670       13976950       11322719       11957536
28S                     23250288       27128583       20212171       20674276
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008             92             39             53             44
Droso_FBgn0000014              2              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr77 RNA_S11589Nr78 RNA_S11589Nr79 RNA_S11589Nr80
18S                     14254622       11721764        8213901        9196647
28S                     25728264       22277257       14463580       14922666
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008            123             64             67            117
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              2              0              0              2
                  RNA_S11589Nr94 RNA_S11589Nr95 RNA_S11589Nr96 RNA_S11589Nr97
18S                       371747         327533         360453         640734
28S                       909545         915491         788546        1269009
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008              0              0              0              0
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              0              2              0              0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(description_file)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                      sample
RNA_S11589Nr53  H1_WT_C1_S2spike_RNAseq_Rep1
RNA_S11589Nr54  H1_WT_C2_S2spike_RNAseq_Rep1
RNA_S11589Nr55 H1_LOW_C1_S2spike_RNAseq_Rep1
RNA_S11589Nr56 H1_LOW_C2_S2spike_RNAseq_Rep1
RNA_S11589Nr57  H1_WT_C1_S2spike_RNAseq_Rep2
RNA_S11589Nr58  H1_WT_C2_S2spike_RNAseq_Rep2
                                                   fastq_1
RNA_S11589Nr53 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
RNA_S11589Nr54 S11589_1/01_fastq/RNA_S11589Nr54.1.fastq.gz
RNA_S11589Nr55 S11589_1/01_fastq/RNA_S11589Nr55.1.fastq.gz
RNA_S11589Nr56 S11589_1/01_fastq/RNA_S11589Nr56.1.fastq.gz
RNA_S11589Nr57 S11589_1/01_fastq/RNA_S11589Nr57.1.fastq.gz
RNA_S11589Nr58 S11589_1/01_fastq/RNA_S11589Nr58.1.fastq.gz
                                                   fastq_2 strandedness
RNA_S11589Nr53 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto
RNA_S11589Nr54 S11589_1/01_fastq/RNA_S11589Nr54.2.fastq.gz         auto
RNA_S11589Nr55 S11589_1/01_fastq/RNA_S11589Nr55.2.fastq.gz         auto
RNA_S11589Nr56 S11589_1/01_fastq/RNA_S11589Nr56.2.fastq.gz         auto
RNA_S11589Nr57 S11589_1/01_fastq/RNA_S11589Nr57.2.fastq.gz         auto
RNA_S11589Nr58 S11589_1/01_fastq/RNA_S11589Nr58.2.fastq.gz         auto
               Genotype Colonies Method Replicate Treatment    sample_name
RNA_S11589Nr53       WT       C1 RNAseq      Rep1      &lt;NA&gt; RNA_S11589Nr53
RNA_S11589Nr54       WT       C2 RNAseq      Rep1      &lt;NA&gt; RNA_S11589Nr54
RNA_S11589Nr55      LOW       C1 RNAseq      Rep1      &lt;NA&gt; RNA_S11589Nr55
RNA_S11589Nr56      LOW       C2 RNAseq      Rep1      &lt;NA&gt; RNA_S11589Nr56
RNA_S11589Nr57       WT       C1 RNAseq      Rep2      &lt;NA&gt; RNA_S11589Nr57
RNA_S11589Nr58       WT       C2 RNAseq      Rep2      &lt;NA&gt; RNA_S11589Nr58</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>rnaseq_description <span class="ot">&lt;-</span> description_file[<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>,]</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>ttseq_description <span class="ot">&lt;-</span> description_file[<span class="dv">13</span><span class="sc">:</span><span class="dv">24</span>,]</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>star_rnaseq <span class="ot">&lt;-</span> star_matrix[, <span class="fu">rownames</span>(rnaseq_description)]</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>star_ttseq <span class="ot">&lt;-</span> star_matrix[, <span class="fu">rownames</span>(ttseq_description)]</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>bowtie_ttseq <span class="ot">&lt;-</span> bowtie_matrix[, <span class="fu">rownames</span>(ttseq_description)]</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(star_ttseq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  RNA_S11589Nr69 RNA_S11589Nr70 RNA_S11589Nr71 RNA_S11589Nr72
18S                     11216638       12437014       10571964       11959462
28S                     20975929       24396074       19971006       21545826
Droso_FBgn0000003              4              0              2              0
Droso_FBgn0000008             64             33             61             90
Droso_FBgn0000014              2              0              0              2
Droso_FBgn0000015              0              2              0              2
                  RNA_S11589Nr73 RNA_S11589Nr74 RNA_S11589Nr75 RNA_S11589Nr76
18S                     12747183       14415866       11706650       12358688
28S                     24449436       29235185       21485212       21723967
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008             93             41             52             43
Droso_FBgn0000014              2              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr77 RNA_S11589Nr78 RNA_S11589Nr79 RNA_S11589Nr80
18S                     14730053       12101005        8481735        9545585
28S                     27060439       24025274       15339482       15719132
Droso_FBgn0000003              2              2              2              2
Droso_FBgn0000008            128             68             68            120
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              2              0              0              2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bowtie_ttseq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  RNA_S11589Nr69 RNA_S11589Nr70 RNA_S11589Nr71 RNA_S11589Nr72
18S                     10882366       12053086       10268220       11573945
28S                     19958365       22646075       18784274       20446596
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008             60             31             56             86
Droso_FBgn0000014              2              0              0              2
Droso_FBgn0000015              0              2              0              2
                  RNA_S11589Nr73 RNA_S11589Nr74 RNA_S11589Nr75 RNA_S11589Nr76
18S                     12361670       13976950       11322719       11957536
28S                     23250288       27128583       20212171       20674276
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008             92             39             53             44
Droso_FBgn0000014              2              0              0              0
Droso_FBgn0000015              0              0              0              0
                  RNA_S11589Nr77 RNA_S11589Nr78 RNA_S11589Nr79 RNA_S11589Nr80
18S                     14254622       11721764        8213901        9196647
28S                     25728264       22277257       14463580       14922666
Droso_FBgn0000003              0              0              0              0
Droso_FBgn0000008            123             64             67            117
Droso_FBgn0000014              0              0              0              0
Droso_FBgn0000015              2              0              0              2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>ttseq_description</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                     sample
RNA_S11589Nr69  H1_WT_C1_S2spike_TTseq_Rep1
RNA_S11589Nr70  H1_WT_C2_S2spike_TTseq_Rep1
RNA_S11589Nr71 H1_LOW_C1_S2spike_TTseq_Rep1
RNA_S11589Nr72 H1_LOW_C2_S2spike_TTseq_Rep1
RNA_S11589Nr73  H1_WT_C1_S2spike_TTseq_Rep2
RNA_S11589Nr74  H1_WT_C2_S2spike_TTseq_Rep2
RNA_S11589Nr75 H1_LOW_C1_S2spike_TTseq_Rep2
RNA_S11589Nr76 H1_LOW_C2_S2spike_TTseq_Rep2
RNA_S11589Nr77  H1_WT_C1_S2spike_TTseq_Rep3
RNA_S11589Nr78  H1_WT_C2_S2spike_TTseq_Rep3
RNA_S11589Nr79 H1_LOW_C1_S2spike_TTseq_Rep3
RNA_S11589Nr80 H1_LOW_C2_S2spike_TTseq_Rep3
                                                   fastq_1
RNA_S11589Nr69 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
RNA_S11589Nr70 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
RNA_S11589Nr71 S11589_2/01_fastq/RNA_S11589Nr71.1.fastq.gz
RNA_S11589Nr72 S11589_2/01_fastq/RNA_S11589Nr72.1.fastq.gz
RNA_S11589Nr73 S11589_2/01_fastq/RNA_S11589Nr73.1.fastq.gz
RNA_S11589Nr74 S11589_2/01_fastq/RNA_S11589Nr74.1.fastq.gz
RNA_S11589Nr75 S11589_2/01_fastq/RNA_S11589Nr75.1.fastq.gz
RNA_S11589Nr76 S11589_2/01_fastq/RNA_S11589Nr76.1.fastq.gz
RNA_S11589Nr77 S11589_2/01_fastq/RNA_S11589Nr77.1.fastq.gz
RNA_S11589Nr78 S11589_2/01_fastq/RNA_S11589Nr78.1.fastq.gz
RNA_S11589Nr79 S11589_2/01_fastq/RNA_S11589Nr79.1.fastq.gz
RNA_S11589Nr80 S11589_2/01_fastq/RNA_S11589Nr80.1.fastq.gz
                                                   fastq_2 strandedness
RNA_S11589Nr69 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto
RNA_S11589Nr70 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto
RNA_S11589Nr71 S11589_2/01_fastq/RNA_S11589Nr71.2.fastq.gz         auto
RNA_S11589Nr72 S11589_2/01_fastq/RNA_S11589Nr72.2.fastq.gz         auto
RNA_S11589Nr73 S11589_2/01_fastq/RNA_S11589Nr73.2.fastq.gz         auto
RNA_S11589Nr74 S11589_2/01_fastq/RNA_S11589Nr74.2.fastq.gz         auto
RNA_S11589Nr75 S11589_2/01_fastq/RNA_S11589Nr75.2.fastq.gz         auto
RNA_S11589Nr76 S11589_2/01_fastq/RNA_S11589Nr76.2.fastq.gz         auto
RNA_S11589Nr77 S11589_2/01_fastq/RNA_S11589Nr77.2.fastq.gz         auto
RNA_S11589Nr78 S11589_2/01_fastq/RNA_S11589Nr78.2.fastq.gz         auto
RNA_S11589Nr79 S11589_2/01_fastq/RNA_S11589Nr79.2.fastq.gz         auto
RNA_S11589Nr80 S11589_2/01_fastq/RNA_S11589Nr80.2.fastq.gz         auto
               Genotype Colonies Method Replicate Treatment    sample_name
RNA_S11589Nr69       WT       C1  TTseq      Rep1      &lt;NA&gt; RNA_S11589Nr69
RNA_S11589Nr70       WT       C2  TTseq      Rep1      &lt;NA&gt; RNA_S11589Nr70
RNA_S11589Nr71      LOW       C1  TTseq      Rep1      &lt;NA&gt; RNA_S11589Nr71
RNA_S11589Nr72      LOW       C2  TTseq      Rep1      &lt;NA&gt; RNA_S11589Nr72
RNA_S11589Nr73       WT       C1  TTseq      Rep2      &lt;NA&gt; RNA_S11589Nr73
RNA_S11589Nr74       WT       C2  TTseq      Rep2      &lt;NA&gt; RNA_S11589Nr74
RNA_S11589Nr75      LOW       C1  TTseq      Rep2      &lt;NA&gt; RNA_S11589Nr75
RNA_S11589Nr76      LOW       C2  TTseq      Rep2      &lt;NA&gt; RNA_S11589Nr76
RNA_S11589Nr77       WT       C1  TTseq      Rep3      &lt;NA&gt; RNA_S11589Nr77
RNA_S11589Nr78       WT       C2  TTseq      Rep3      &lt;NA&gt; RNA_S11589Nr78
RNA_S11589Nr79      LOW       C1  TTseq      Rep3      &lt;NA&gt; RNA_S11589Nr79
RNA_S11589Nr80      LOW       C2  TTseq      Rep3      &lt;NA&gt; RNA_S11589Nr80</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>gtf_genes <span class="ot">&lt;-</span> gtf_file[gtf_file<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"gene"</span>, ]</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>MusMusculus_Genes <span class="ot">&lt;-</span> gtf_genes[<span class="fu">grep</span>(<span class="st">"^ENS"</span>, gtf_genes<span class="sc">$</span>gene_id), ]</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>MusGeneNames <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">data.frame</span>(<span class="fu">mcols</span>(MusMusculus_Genes[,<span class="fu">c</span>(<span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>)])))</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(MusGeneNames)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                gene_id gene_name
1  ENSMUSG00000000001.4     Gnai3
2 ENSMUSG00000000003.15      Pbsn
3 ENSMUSG00000000028.15     Cdc45
4 ENSMUSG00000000031.16       H19
5 ENSMUSG00000000037.17     Scml2
6 ENSMUSG00000000049.11      Apoh</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>base_dir <span class="ot">&lt;-</span> <span class="st">"/home/franck/SCRATCH/fpan/Genome_Index/chR_index_files/SalmonRNA/QuantOutput_gcBias"</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(base_dir, <span class="fu">paste0</span>(rnaseq_description<span class="sc">$</span>sample_name, <span class="st">"_quant"</span>), <span class="st">"quant.sf"</span>)</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(files) <span class="ot">&lt;-</span> rnaseq_description<span class="sc">$</span>sample_name</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>tx2gene <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">file.path</span>(base_dir, <span class="st">"tx2gene.rsem.filtered.csv"</span>), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>txi_tx <span class="ot">&lt;-</span> <span class="fu">tximport</span>(files, <span class="at">type=</span><span class="st">"salmon"</span>, <span class="at">txOut=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>reading in files with read_tsv</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tx2gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 2
  TXNAME                      GENEID              
  &lt;chr&gt;                       &lt;chr&gt;               
1 ENSMUST00000193812.1        ENSMUSG00000102693.1
2 ENSMUST00000082908.1        ENSMUSG00000064842.1
3 ENSMUST00000192857.1        ENSMUSG00000102851.1
4 ENSMUST00000161581.1        ENSMUSG00000089699.1
5 PreMrnaENSMUSG00000089699.1 ENSMUSG00000089699.1
6 ENSMUST00000192183.1        ENSMUSG00000103147.1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>TTbase_dir <span class="ot">&lt;-</span> <span class="st">"/home/franck/SCRATCH/fpan/Genome_Index/chR_index_files/SalmonTT/QuantOutput"</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>TTfiles <span class="ot">&lt;-</span> <span class="fu">file.path</span>(TTbase_dir, <span class="fu">paste0</span>(ttseq_description<span class="sc">$</span>sample_name, <span class="st">"_quant"</span>), <span class="st">"quant.sf"</span>)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(TTfiles) <span class="ot">&lt;-</span> ttseq_description<span class="sc">$</span>sample_name</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>TTtx2gene <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">file.path</span>(base_dir, <span class="st">"tx2gene.rsem.filtered.csv"</span>), <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>TTtxi_tx <span class="ot">&lt;-</span> <span class="fu">tximport</span>(TTfiles, <span class="at">type=</span><span class="st">"salmon"</span>, <span class="at">txOut=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>reading in files with read_tsv</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1 2 3 4 5 6 7 8 9 10 11 12 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TTtx2gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 2
  TXNAME                      GENEID              
  &lt;chr&gt;                       &lt;chr&gt;               
1 ENSMUST00000193812.1        ENSMUSG00000102693.1
2 ENSMUST00000082908.1        ENSMUSG00000064842.1
3 ENSMUST00000192857.1        ENSMUSG00000102851.1
4 ENSMUST00000161581.1        ENSMUSG00000089699.1
5 PreMrnaENSMUSG00000089699.1 ENSMUSG00000089699.1
6 ENSMUST00000192183.1        ENSMUSG00000103147.1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>gtf_mono_exon <span class="ot">&lt;-</span> latest_gtf</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>exons_gtf <span class="ot">&lt;-</span> gtf_mono_exon[gtf_mono_exon<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"exon"</span>]</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>df_subset <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(exons_gtf)</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>exon_counts <span class="ot">&lt;-</span> df_subset <span class="sc">%&gt;%</span></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(transcript_id) <span class="sc">%&gt;%</span></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">exon_count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(exon_count <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>mono_exon_id <span class="ot">&lt;-</span> exon_counts<span class="sc">$</span>transcript_id</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>mono_exon_id <span class="ot">&lt;-</span> mono_exon_id[<span class="fu">grepl</span>(<span class="st">"^ENS"</span>, mono_exon_id)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="deseq-initialization-visualization" class="level2">
<h2 class="anchored" data-anchor-id="deseq-initialization-visualization">DESeq Initialization &amp; Visualization</h2>
<section id="star-rnaseq-dds" class="level3">
<h3 class="anchored" data-anchor-id="star-rnaseq-dds">STAR-RNAseq-dds</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>rnaseq_dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> star_rnaseq,</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> rnaseq_description,</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> Colonies <span class="sc">+</span> Genotype )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>rnaseq_dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(rnaseq_dds, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"Droso_"</span>, <span class="fu">rownames</span>(star_rnaseq)))</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>rnaseq_dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(rnaseq_dds[<span class="fu">grepl</span>(<span class="st">"^ENS"</span>, <span class="fu">rownames</span>(star_rnaseq))])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(rnaseq_dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"          "Colonies_C2_vs_C1"  "Genotype_LOW_vs_WT"</code></pre>
</div>
</div>
</section>
<section id="salmon-rnaseq-dds-mus-genes" class="level3">
<h3 class="anchored" data-anchor-id="salmon-rnaseq-dds-mus-genes">Salmon-RNAseq-dds-Mus-Genes</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>ddsTxi <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_tx,</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">colData =</span> rnaseq_description,</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">design =</span> <span class="sc">~</span> Colonies <span class="sc">+</span> Genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using counts and average transcript lengths from tximport</code></pre>
</div>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>ddsTxi <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(ddsTxi, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"^FBtr|^PreMrnaDroso"</span>, <span class="fu">rownames</span>(ddsTxi)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'avgTxLength' from assays(dds), correcting for library size</code></pre>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>ddsTxi <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ddsTxi[<span class="fu">grepl</span>(<span class="st">"^ENS|^PreMrnaENS"</span>, <span class="fu">rownames</span>(ddsTxi))])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing normalization factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(ddsTxi)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"          "Colonies_C2_vs_C1"  "Genotype_LOW_vs_WT"</code></pre>
</div>
</div>
</section>
<section id="salmon-rnaseq-dds-mus-premrna" class="level3">
<h3 class="anchored" data-anchor-id="salmon-rnaseq-dds-mus-premrna">Salmon-RNAseq-dds-Mus-PremRNA</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>ddsTxiTT <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_tx,</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">colData =</span> rnaseq_description,</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">design =</span> <span class="sc">~</span> Colonies <span class="sc">+</span> Genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using counts and average transcript lengths from tximport</code></pre>
</div>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>ddsTxiTT <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(ddsTxiTT, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"^FBtr|^PreMrnaDroso"</span>, <span class="fu">rownames</span>(ddsTxiTT)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'avgTxLength' from assays(dds), correcting for library size</code></pre>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>premrna_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ddsTxiTT[<span class="fu">grepl</span>(<span class="st">"^PreMrnaENS"</span>, <span class="fu">rownames</span>(ddsTxiTT))])</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>premrna_id <span class="ot">&lt;-</span> <span class="fu">c</span>(premrna_id, mono_exon_id)</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>ddsTxiTT <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ddsTxiTT[<span class="fu">rownames</span>(ddsTxiTT) <span class="sc">%in%</span> premrna_id,])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing normalization factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(ddsTxiTT)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"          "Colonies_C2_vs_C1"  "Genotype_LOW_vs_WT"</code></pre>
</div>
</div>
</section>
<section id="star-ttseq-dds" class="level3">
<h3 class="anchored" data-anchor-id="star-ttseq-dds">STAR-TTseq-dds</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>ttseq_dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> star_ttseq,</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> ttseq_description,</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span> Colonies <span class="sc">+</span> Genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>ttseq_dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(ttseq_dds, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"^Droso"</span>, <span class="fu">rownames</span>(star_ttseq)))</span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>ttseq_dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ttseq_dds[<span class="fu">grepl</span>(<span class="st">"^ENS"</span>, <span class="fu">rownames</span>(star_ttseq))])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(ttseq_dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"          "Colonies_C2_vs_C1"  "Genotype_LOW_vs_WT"</code></pre>
</div>
</div>
</section>
<section id="bowtie2-ttseq-dds" class="level3">
<h3 class="anchored" data-anchor-id="bowtie2-ttseq-dds">Bowtie2-TTseq-dds</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>ttseq_dds_Bowtie <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> bowtie_ttseq,</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> ttseq_description,</span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span> Colonies <span class="sc">+</span> Genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>ttseq_dds_Bowtie <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(ttseq_dds_Bowtie, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"Droso_"</span>, <span class="fu">rownames</span>(bowtie_ttseq)))</span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>ttseq_dds_Bowtie <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ttseq_dds_Bowtie[<span class="fu">grepl</span>(<span class="st">"^ENS"</span>, <span class="fu">rownames</span>(bowtie_ttseq))])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(ttseq_dds_Bowtie)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"          "Colonies_C2_vs_C1"  "Genotype_LOW_vs_WT"</code></pre>
</div>
</div>
</section>
<section id="results-instances-low-vs-wt" class="level3">
<h3 class="anchored" data-anchor-id="results-instances-low-vs-wt">Results Instances LOW vs WT</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a>RnaRes <span class="ot">&lt;-</span> <span class="fu">results</span>(rnaseq_dds, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>TtRes <span class="ot">&lt;-</span> <span class="fu">results</span>(ttseq_dds, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>TtResBowtie <span class="ot">&lt;-</span> <span class="fu">results</span>(ttseq_dds_Bowtie, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a>TxiRes <span class="ot">&lt;-</span> <span class="fu">results</span>(ddsTxi, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a>TxiResTT <span class="ot">&lt;-</span> <span class="fu">results</span>(ddsTxiTT, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb145-6"><a href="#cb145-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-7"><a href="#cb145-7" aria-hidden="true" tabindex="-1"></a>TxiTTResNoNa <span class="ot">&lt;-</span> TxiResTT[<span class="sc">!</span><span class="fu">is.na</span>(TxiResTT<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(TxiResTT<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb145-8"><a href="#cb145-8" aria-hidden="true" tabindex="-1"></a>TxiResNoNa <span class="ot">&lt;-</span> TxiRes[<span class="sc">!</span><span class="fu">is.na</span>(TxiRes<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(TxiRes<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb145-9"><a href="#cb145-9" aria-hidden="true" tabindex="-1"></a>RnaResNoNa <span class="ot">&lt;-</span> RnaRes[<span class="sc">!</span><span class="fu">is.na</span>(RnaRes<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(RnaRes<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb145-10"><a href="#cb145-10" aria-hidden="true" tabindex="-1"></a>TTResNoNa <span class="ot">&lt;-</span> TtRes[<span class="sc">!</span><span class="fu">is.na</span>(TtRes<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(TtRes<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb145-11"><a href="#cb145-11" aria-hidden="true" tabindex="-1"></a>TTResBowtieNoNa <span class="ot">&lt;-</span> TtResBowtie[<span class="sc">!</span><span class="fu">is.na</span>(TtResBowtie<span class="sc">$</span>padj) <span class="sc">&amp;</span> </span>
<span id="cb145-12"><a href="#cb145-12" aria-hidden="true" tabindex="-1"></a>                              <span class="sc">!</span><span class="fu">is.na</span>(TtResBowtie<span class="sc">$</span>log2FoldChange), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ma-plots" class="level3">
<h3 class="anchored" data-anchor-id="ma-plots">MA plots</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(RnaRes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(TxiResNoNa)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-16-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(TxiTTResNoNa)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-16-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(TtRes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-16-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(TtResBowtie)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-16-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="manual-ma-plots" class="level3">
<h3 class="anchored" data-anchor-id="manual-ma-plots">Manual MA plots</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>foldchange_threeshold <span class="ot">&lt;-</span> <span class="fl">0.6</span></span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>padj_threeshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a>RnaResNoNa<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a>RnaResNoNa<span class="sc">$</span>significant[RnaResNoNa<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-7"><a href="#cb151-7" aria-hidden="true" tabindex="-1"></a>TTResNoNa<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb151-8"><a href="#cb151-8" aria-hidden="true" tabindex="-1"></a>TTResNoNa<span class="sc">$</span>significant[TTResNoNa<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span>
<span id="cb151-9"><a href="#cb151-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-10"><a href="#cb151-10" aria-hidden="true" tabindex="-1"></a>TTResBowtieNoNa<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb151-11"><a href="#cb151-11" aria-hidden="true" tabindex="-1"></a>TTResBowtieNoNa<span class="sc">$</span>significant[TTResBowtieNoNa<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span>
<span id="cb151-12"><a href="#cb151-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-13"><a href="#cb151-13" aria-hidden="true" tabindex="-1"></a>TxiResNoNa<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb151-14"><a href="#cb151-14" aria-hidden="true" tabindex="-1"></a>TxiResNoNa<span class="sc">$</span>significant[TxiResNoNa<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span>
<span id="cb151-15"><a href="#cb151-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-16"><a href="#cb151-16" aria-hidden="true" tabindex="-1"></a>TxiTTResNoNa<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb151-17"><a href="#cb151-17" aria-hidden="true" tabindex="-1"></a>TxiTTResNoNa<span class="sc">$</span>significant[TxiTTResNoNa<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span>
<span id="cb151-18"><a href="#cb151-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-19"><a href="#cb151-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(RnaResNoNa, <span class="fu">aes</span>(<span class="at">x =</span>RnaResNoNa<span class="sc">$</span>baseMean, </span>
<span id="cb151-20"><a href="#cb151-20" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> RnaResNoNa<span class="sc">$</span>log2FoldChange, </span>
<span id="cb151-21"><a href="#cb151-21" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb151-22"><a href="#cb151-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb151-23"><a href="#cb151-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb151-24"><a href="#cb151-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from STAR RNA-seq"</span>,</span>
<span id="cb151-25"><a href="#cb151-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb151-26"><a href="#cb151-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb151-27"><a href="#cb151-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb151-28"><a href="#cb151-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb151-29"><a href="#cb151-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb151-30"><a href="#cb151-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb151-31"><a href="#cb151-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb151-32"><a href="#cb151-32" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb151-33"><a href="#cb151-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TxiResNoNa, <span class="fu">aes</span>(<span class="at">x =</span>TxiResNoNa<span class="sc">$</span>baseMean, </span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> TxiResNoNa<span class="sc">$</span>log2FoldChange, </span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb152-4"><a href="#cb152-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb152-5"><a href="#cb152-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb152-6"><a href="#cb152-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from Salmon RNA-seq"</span>,</span>
<span id="cb152-7"><a href="#cb152-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb152-8"><a href="#cb152-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb152-9"><a href="#cb152-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb152-10"><a href="#cb152-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb152-11"><a href="#cb152-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb152-12"><a href="#cb152-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb152-13"><a href="#cb152-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb152-14"><a href="#cb152-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb152-15"><a href="#cb152-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-17-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TxiTTResNoNa, <span class="fu">aes</span>(<span class="at">x =</span>TxiTTResNoNa<span class="sc">$</span>baseMean, </span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> TxiTTResNoNa<span class="sc">$</span>log2FoldChange, </span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb153-6"><a href="#cb153-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from Salmon RNA-seq Only PreMrna"</span>,</span>
<span id="cb153-7"><a href="#cb153-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb153-8"><a href="#cb153-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb153-9"><a href="#cb153-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb153-10"><a href="#cb153-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb153-11"><a href="#cb153-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb153-12"><a href="#cb153-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb153-13"><a href="#cb153-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb153-14"><a href="#cb153-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb153-15"><a href="#cb153-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-17-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TTResNoNa, <span class="fu">aes</span>(<span class="at">x =</span>TTResNoNa<span class="sc">$</span>baseMean, </span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> TTResNoNa<span class="sc">$</span>log2FoldChange, </span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from STAR TT-seq"</span>,</span>
<span id="cb154-7"><a href="#cb154-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb154-8"><a href="#cb154-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb154-9"><a href="#cb154-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb154-10"><a href="#cb154-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb154-11"><a href="#cb154-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb154-12"><a href="#cb154-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb154-13"><a href="#cb154-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb154-14"><a href="#cb154-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb154-15"><a href="#cb154-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-17-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TTResBowtieNoNa, <span class="fu">aes</span>(<span class="at">x =</span>TTResBowtieNoNa<span class="sc">$</span>baseMean, </span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> TTResBowtieNoNa<span class="sc">$</span>log2FoldChange, </span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_point(data=TTResBowtieNoNa["ENSMUSG00000069049.11",], shape=23, color="darkgreen", size=5, aes(x = baseMean, </span></span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#y = log2FoldChange, </span></span>
<span id="cb155-7"><a href="#cb155-7" aria-hidden="true" tabindex="-1"></a>                       <span class="co">#col = significant)) +</span></span>
<span id="cb155-8"><a href="#cb155-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb155-9"><a href="#cb155-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from Bowtie2 TT-seq"</span>,</span>
<span id="cb155-10"><a href="#cb155-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb155-11"><a href="#cb155-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb155-12"><a href="#cb155-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb155-13"><a href="#cb155-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb155-14"><a href="#cb155-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb155-15"><a href="#cb155-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb155-16"><a href="#cb155-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb155-17"><a href="#cb155-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb155-18"><a href="#cb155-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-17-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="qq-plot" class="level3">
<h3 class="anchored" data-anchor-id="qq-plot">QQ plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a>RnaResNoNa<span class="sc">$</span>PvalueSorted <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">sort</span>(RnaResNoNa<span class="sc">$</span>pvalue))</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a>RnaResNoNa<span class="sc">$</span>PvalueExpected <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">ppoints</span>(<span class="fu">length</span>(RnaResNoNa<span class="sc">$</span>PvalueSorted)))</span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-4"><a href="#cb156-4" aria-hidden="true" tabindex="-1"></a>TxiResNoNa<span class="sc">$</span>PvalueSorted <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">sort</span>(TxiResNoNa<span class="sc">$</span>pvalue))</span>
<span id="cb156-5"><a href="#cb156-5" aria-hidden="true" tabindex="-1"></a>TxiResNoNa<span class="sc">$</span>PvalueExpected <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">ppoints</span>(<span class="fu">length</span>(TxiResNoNa<span class="sc">$</span>PvalueSorted)))</span>
<span id="cb156-6"><a href="#cb156-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-7"><a href="#cb156-7" aria-hidden="true" tabindex="-1"></a>TxiTTResNoNa<span class="sc">$</span>PvalueSorted <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">sort</span>(TxiTTResNoNa<span class="sc">$</span>pvalue))</span>
<span id="cb156-8"><a href="#cb156-8" aria-hidden="true" tabindex="-1"></a>TxiTTResNoNa<span class="sc">$</span>PvalueExpected <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">ppoints</span>(<span class="fu">length</span>(TxiTTResNoNa<span class="sc">$</span>PvalueSorted)))</span>
<span id="cb156-9"><a href="#cb156-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-10"><a href="#cb156-10" aria-hidden="true" tabindex="-1"></a>TTResNoNa<span class="sc">$</span>PvalueSorted <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">sort</span>(TTResNoNa<span class="sc">$</span>pvalue))</span>
<span id="cb156-11"><a href="#cb156-11" aria-hidden="true" tabindex="-1"></a>TTResNoNa<span class="sc">$</span>PvalueExpected <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">ppoints</span>(<span class="fu">length</span>(TTResNoNa<span class="sc">$</span>PvalueSorted)))</span>
<span id="cb156-12"><a href="#cb156-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-13"><a href="#cb156-13" aria-hidden="true" tabindex="-1"></a>TTResBowtieNoNa<span class="sc">$</span>PvalueSorted <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">sort</span>(TTResBowtieNoNa<span class="sc">$</span>pvalue))</span>
<span id="cb156-14"><a href="#cb156-14" aria-hidden="true" tabindex="-1"></a>TTResBowtieNoNa<span class="sc">$</span>PvalueExpected <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">ppoints</span>(<span class="fu">length</span>(TTResBowtieNoNa<span class="sc">$</span>PvalueSorted)))</span>
<span id="cb156-15"><a href="#cb156-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-16"><a href="#cb156-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-17"><a href="#cb156-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-18"><a href="#cb156-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(RnaResNoNa, <span class="fu">aes</span>(PvalueSorted, PvalueExpected)) <span class="sc">+</span></span>
<span id="cb156-19"><a href="#cb156-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb156-20"><a href="#cb156-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb156-21"><a href="#cb156-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"QQ Plot of RNA-seq"</span>,</span>
<span id="cb156-22"><a href="#cb156-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Observed -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>), </span>
<span id="cb156-23"><a href="#cb156-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Expected -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>)) <span class="sc">+</span> </span>
<span id="cb156-24"><a href="#cb156-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb156-25"><a href="#cb156-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb156-26"><a href="#cb156-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TxiResNoNa, <span class="fu">aes</span>(PvalueSorted, PvalueExpected)) <span class="sc">+</span></span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb157-4"><a href="#cb157-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"QQ Plot of Salmon RNA-seq"</span>,</span>
<span id="cb157-5"><a href="#cb157-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Observed -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>), </span>
<span id="cb157-6"><a href="#cb157-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Expected -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>)) <span class="sc">+</span> </span>
<span id="cb157-7"><a href="#cb157-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb157-8"><a href="#cb157-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb157-9"><a href="#cb157-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-18-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TxiTTResNoNa, <span class="fu">aes</span>(PvalueSorted, PvalueExpected)) <span class="sc">+</span></span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"QQ Plot of Salmon RNA-seq Only PreMrna"</span>,</span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Observed -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>), </span>
<span id="cb158-6"><a href="#cb158-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Expected -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>)) <span class="sc">+</span> </span>
<span id="cb158-7"><a href="#cb158-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb158-8"><a href="#cb158-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb158-9"><a href="#cb158-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-18-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TTResNoNa, <span class="fu">aes</span>(PvalueSorted, PvalueExpected)) <span class="sc">+</span></span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb159-4"><a href="#cb159-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"QQ Plot of STAR TT-seq"</span>,</span>
<span id="cb159-5"><a href="#cb159-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Observed -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>), </span>
<span id="cb159-6"><a href="#cb159-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Expected -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>)) <span class="sc">+</span> </span>
<span id="cb159-7"><a href="#cb159-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb159-8"><a href="#cb159-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb159-9"><a href="#cb159-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-18-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TTResBowtieNoNa, <span class="fu">aes</span>(PvalueSorted, PvalueExpected)) <span class="sc">+</span></span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"QQ Plot of Bowtie2 TT-seq"</span>,</span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Observed -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>), </span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Expected -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>)) <span class="sc">+</span> </span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb160-9"><a href="#cb160-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-18-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="rna-seq-analysis" class="level2">
<h2 class="anchored" data-anchor-id="rna-seq-analysis">RNA-Seq Analysis</h2>
<section id="star" class="level3">
<h3 class="anchored" data-anchor-id="star">STAR</h3>
<section id="volcano-plot" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot">Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a>RnaResVocPlot <span class="ot">&lt;-</span> RnaResNoNa[<span class="fu">order</span>(RnaResNoNa<span class="sc">$</span>padj),]</span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>RnaResVocPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb161-3"><a href="#cb161-3" aria-hidden="true" tabindex="-1"></a>RnaResVocPlot<span class="sc">$</span>diffexpressed[RnaResVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> RnaResVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb161-4"><a href="#cb161-4" aria-hidden="true" tabindex="-1"></a>RnaResVocPlot<span class="sc">$</span>diffexpressed[RnaResVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> RnaResVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb161-5"><a href="#cb161-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-6"><a href="#cb161-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> RnaResVocPlot, </span>
<span id="cb161-7"><a href="#cb161-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb161-8"><a href="#cb161-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb161-9"><a href="#cb161-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb161-10"><a href="#cb161-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of RNA-seq"</span>,</span>
<span id="cb161-11"><a href="#cb161-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb161-12"><a href="#cb161-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb161-13"><a href="#cb161-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb161-14"><a href="#cb161-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb161-15"><a href="#cb161-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb161-16"><a href="#cb161-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb161-17"><a href="#cb161-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb161-18"><a href="#cb161-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="degs-list-up-down" class="level4">
<h4 class="anchored" data-anchor-id="degs-list-up-down">Degs-List-Up-Down</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a>RnaUpReg <span class="ot">&lt;-</span> RnaResVocPlot[RnaResVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb162-2"><a href="#cb162-2" aria-hidden="true" tabindex="-1"></a>RnaDownReg <span class="ot">&lt;-</span> RnaResVocPlot[RnaResVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb162-3"><a href="#cb162-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-4"><a href="#cb162-4" aria-hidden="true" tabindex="-1"></a>RnaUpReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(RnaUpReg<span class="sc">$</span>padj)</span>
<span id="cb162-5"><a href="#cb162-5" aria-hidden="true" tabindex="-1"></a>RnaDownReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(RnaDownReg<span class="sc">$</span>padj)</span>
<span id="cb162-6"><a href="#cb162-6" aria-hidden="true" tabindex="-1"></a>RnaUpRegSorted <span class="ot">&lt;-</span> RnaUpReg[<span class="fu">order</span>(<span class="sc">-</span>RnaUpReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>RnaUpReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb162-7"><a href="#cb162-7" aria-hidden="true" tabindex="-1"></a>RnaDownRegSorted <span class="ot">&lt;-</span> RnaDownReg[<span class="fu">order</span>(RnaDownReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>RnaDownReg<span class="sc">$</span>neg_log10_padj),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bar-plot-top-20-degs" class="level4">
<h4 class="anchored" data-anchor-id="bar-plot-top-20-degs">Bar Plot Top 20 Degs</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a>top_upregRNA <span class="ot">&lt;-</span> <span class="fu">head</span>(RnaUpRegSorted, <span class="dv">20</span>)</span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>top_downregRNA <span class="ot">&lt;-</span> <span class="fu">head</span>(RnaDownRegSorted, <span class="dv">20</span>)</span>
<span id="cb163-3"><a href="#cb163-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb163-4"><a href="#cb163-4" aria-hidden="true" tabindex="-1"></a>top_upregRNA<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregRNA), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb163-5"><a href="#cb163-5" aria-hidden="true" tabindex="-1"></a>top_upregRNA<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregRNA<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregRNA<span class="sc">$</span>gene))</span>
<span id="cb163-6"><a href="#cb163-6" aria-hidden="true" tabindex="-1"></a>top_downregRNA<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregRNA), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb163-7"><a href="#cb163-7" aria-hidden="true" tabindex="-1"></a>top_downregRNA<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregRNA<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregRNA<span class="sc">$</span>gene))</span>
<span id="cb163-8"><a href="#cb163-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb163-9"><a href="#cb163-9" aria-hidden="true" tabindex="-1"></a>TopUpDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregRNA)</span>
<span id="cb163-10"><a href="#cb163-10" aria-hidden="true" tabindex="-1"></a>TopDownDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregRNA)</span>
<span id="cb163-11"><a href="#cb163-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb163-12"><a href="#cb163-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregRNA, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb163-13"><a href="#cb163-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb163-14"><a href="#cb163-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb163-15"><a href="#cb163-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [RNAseq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb163-16"><a href="#cb163-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregRNA, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb164-2"><a href="#cb164-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>   <span class="co"># reddish color for downregulated</span></span>
<span id="cb164-3"><a href="#cb164-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb164-4"><a href="#cb164-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [RNAseq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb164-5"><a href="#cb164-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-21-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>####Scatter Plot Mean Expression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalized <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">counts</span>(rnaseq_dds, <span class="at">normalize=</span><span class="cn">TRUE</span>))</span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalized<span class="sc">$</span>WT_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(RnaDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb165-3"><a href="#cb165-3" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalized<span class="sc">$</span>LOW_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(RnaDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb165-4"><a href="#cb165-4" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalized<span class="sc">$</span>WT_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(RnaDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb165-5"><a href="#cb165-5" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalized<span class="sc">$</span>LOW_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(RnaDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb165-6"><a href="#cb165-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-7"><a href="#cb165-7" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalizedFiltered <span class="ot">&lt;-</span> RnaDdsNormalized <span class="sc">%&gt;%</span> <span class="fu">filter</span>(WT_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb165-8"><a href="#cb165-8" aria-hidden="true" tabindex="-1"></a>                                                        (WT_Mean <span class="sc">-</span> WT_std <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb165-9"><a href="#cb165-9" aria-hidden="true" tabindex="-1"></a>                                                        LOW_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb165-10"><a href="#cb165-10" aria-hidden="true" tabindex="-1"></a>                                                        (LOW_Mean <span class="sc">-</span> LOW_std <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb165-11"><a href="#cb165-11" aria-hidden="true" tabindex="-1"></a>RnaResNoNa<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(RnaResNoNa)</span>
<span id="cb165-12"><a href="#cb165-12" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalizedFiltered<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(RnaDdsNormalizedFiltered)</span>
<span id="cb165-13"><a href="#cb165-13" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalizedFiltered <span class="ot">&lt;-</span> <span class="fu">merge</span>(RnaDdsNormalizedFiltered, <span class="fu">data.frame</span>(RnaResNoNa[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">10</span>)]), <span class="at">by=</span><span class="st">"index"</span>)</span>
<span id="cb165-14"><a href="#cb165-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(RnaDdsNormalizedFiltered) <span class="ot">&lt;-</span> RnaDdsNormalizedFiltered<span class="sc">$</span>index</span>
<span id="cb165-15"><a href="#cb165-15" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalizedFiltered <span class="ot">&lt;-</span> RnaDdsNormalizedFiltered[,<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(RnaDdsNormalizedFiltered)]</span>
<span id="cb165-16"><a href="#cb165-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-17"><a href="#cb165-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-18"><a href="#cb165-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(RnaDdsNormalizedFiltered, <span class="fu">aes</span>(<span class="at">x=</span> WT_Mean, <span class="at">y=</span> LOW_Mean, <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb165-19"><a href="#cb165-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb165-20"><a href="#cb165-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb165-21"><a href="#cb165-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb165-22"><a href="#cb165-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"Scatter plot of RNA-seq"</span>,</span>
<span id="cb165-23"><a href="#cb165-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb165-24"><a href="#cb165-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in WT"</span>), </span>
<span id="cb165-25"><a href="#cb165-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in LOW"</span>)) <span class="sc">+</span> </span>
<span id="cb165-26"><a href="#cb165-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb165-27"><a href="#cb165-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"blue"</span>),</span>
<span id="cb165-28"><a href="#cb165-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb165-29"><a href="#cb165-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbar(data = RnaDdsNormalizedFiltered[1:50,] ,aes(ymin = LOW_Mean - LOW_std, ymax = LOW_Mean + LOW_std), width = 0.1, alpha = 0.2) +</span></span>
<span id="cb165-30"><a href="#cb165-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbarh(aes(xmin = WT_Mean - WT_std, xmax = WT_Mean + WT_std), height = 0.1, alpha = 0.2) +</span></span>
<span id="cb165-31"><a href="#cb165-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="correlation-matrix-ssample" class="level4">
<h4 class="anchored" data-anchor-id="correlation-matrix-ssample">Correlation Matrix SSample</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Genotype =</span> rnaseq_description<span class="sc">$</span>Genotype)</span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(rnaseq_description)</span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb166-4"><a href="#cb166-4" aria-hidden="true" tabindex="-1"></a>spearman_correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(RnaDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(RnaDdsNormalized))], <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb166-5"><a href="#cb166-5" aria-hidden="true" tabindex="-1"></a>dist_from_spearman <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>spearman_correlation_matrix)</span>
<span id="cb166-6"><a href="#cb166-6" aria-hidden="true" tabindex="-1"></a>dist_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(dist_from_spearman)</span>
<span id="cb166-7"><a href="#cb166-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb166-8"><a href="#cb166-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(dist_matrix, </span>
<span id="cb166-9"><a href="#cb166-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> dist_from_spearman,</span>
<span id="cb166-10"><a href="#cb166-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> dist_from_spearman,</span>
<span id="cb166-11"><a href="#cb166-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> annotation,</span>
<span id="cb166-12"><a href="#cb166-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> annotation,</span>
<span id="cb166-13"><a href="#cb166-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">plasma</span>(<span class="dv">256</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalized<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(RnaDdsNormalized)</span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalized.long <span class="ot">&lt;-</span> </span>
<span id="cb167-3"><a href="#cb167-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(RnaDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(RnaDdsNormalized)) <span class="sc">|</span> <span class="fu">colnames</span>(RnaDdsNormalized) <span class="sc">==</span> <span class="st">"gene_name"</span>],</span>
<span id="cb167-4"><a href="#cb167-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">id.vars =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb167-5"><a href="#cb167-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">variable_name =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb167-6"><a href="#cb167-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(RnaDdsNormalized.long)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"sample_name"</span></span>
<span id="cb167-7"><a href="#cb167-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(RnaDdsNormalized.long)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"normalized_count"</span></span>
<span id="cb167-8"><a href="#cb167-8" aria-hidden="true" tabindex="-1"></a>RnaDdsNormalized.long <span class="ot">&lt;-</span> <span class="fu">merge</span>(RnaDdsNormalized.long, rnaseq_description, <span class="at">by =</span><span class="st">"sample_name"</span>)</span>
<span id="cb167-9"><a href="#cb167-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(RnaDdsNormalized.long)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     sample_name             gene_name normalized_count
1 RNA_S11589Nr53  ENSMUSG00000000001.4      42608.89815
2 RNA_S11589Nr53 ENSMUSG00000000003.15          0.00000
3 RNA_S11589Nr53 ENSMUSG00000000028.15      20282.47898
4 RNA_S11589Nr53 ENSMUSG00000000031.16       6948.76579
5 RNA_S11589Nr53 ENSMUSG00000000037.17       1558.97753
6 RNA_S11589Nr53 ENSMUSG00000000049.11         68.60447
                        sample                                     fastq_1
1 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
2 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
3 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
4 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
5 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
6 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
                                      fastq_2 strandedness Genotype Colonies
1 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
2 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
3 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
4 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
5 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
6 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
  Method Replicate Treatment
1 RNAseq      Rep1      &lt;NA&gt;
2 RNAseq      Rep1      &lt;NA&gt;
3 RNAseq      Rep1      &lt;NA&gt;
4 RNAseq      Rep1      &lt;NA&gt;
5 RNAseq      Rep1      &lt;NA&gt;
6 RNAseq      Rep1      &lt;NA&gt;</code></pre>
</div>
</div>
</section>
<section id="top-10-up-degs-bar-plot" class="level4">
<h4 class="anchored" data-anchor-id="top-10-up-degs-bar-plot">Top 10 Up Degs Bar Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a>top10_upregRna <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregRNA[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb169-2"><a href="#cb169-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-3"><a href="#cb169-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_upregRna) {</span>
<span id="cb169-4"><a href="#cb169-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(RnaDdsNormalized.long[RnaDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb169-5"><a href="#cb169-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb169-6"><a href="#cb169-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb169-7"><a href="#cb169-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb169-8"><a href="#cb169-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb169-9"><a href="#cb169-9" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb169-10"><a href="#cb169-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb169-11"><a href="#cb169-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb169-12"><a href="#cb169-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb169-13"><a href="#cb169-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb169-14"><a href="#cb169-14" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-25-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-down-degs-bar-plot" class="level4">
<h4 class="anchored" data-anchor-id="top-10-down-degs-bar-plot">Top 10 Down Degs Bar Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb170"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a>top10_downregRna <span class="ot">&lt;-</span> <span class="fu">rownames</span>(<span class="fu">head</span>(top_downregRNA,<span class="dv">10</span>))</span>
<span id="cb170-2"><a href="#cb170-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-3"><a href="#cb170-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_downregRna) {</span>
<span id="cb170-4"><a href="#cb170-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(RnaDdsNormalized.long[RnaDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb170-5"><a href="#cb170-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb170-6"><a href="#cb170-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb170-7"><a href="#cb170-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb170-8"><a href="#cb170-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb170-9"><a href="#cb170-9" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb170-10"><a href="#cb170-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb170-11"><a href="#cb170-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb170-12"><a href="#cb170-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb170-13"><a href="#cb170-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb170-14"><a href="#cb170-14" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-26-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="salmon" class="level3">
<h3 class="anchored" data-anchor-id="salmon">Salmon</h3>
<section id="gene-pre-mrna" class="level4">
<h4 class="anchored" data-anchor-id="gene-pre-mrna">Gene &amp; pre-mRNA</h4>
<section id="volcano-plot-1" class="level5">
<h5 class="anchored" data-anchor-id="volcano-plot-1">Volcano Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a>TxiResVocPlot <span class="ot">&lt;-</span> TxiResNoNa[<span class="fu">order</span>(TxiResNoNa<span class="sc">$</span>padj),]</span>
<span id="cb171-2"><a href="#cb171-2" aria-hidden="true" tabindex="-1"></a>TxiResVocPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb171-3"><a href="#cb171-3" aria-hidden="true" tabindex="-1"></a>TxiResVocPlot<span class="sc">$</span>diffexpressed[TxiResVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> TxiResVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb171-4"><a href="#cb171-4" aria-hidden="true" tabindex="-1"></a>TxiResVocPlot<span class="sc">$</span>diffexpressed[TxiResVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> TxiResVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb171-5"><a href="#cb171-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-6"><a href="#cb171-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> TxiResVocPlot, </span>
<span id="cb171-7"><a href="#cb171-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb171-8"><a href="#cb171-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb171-9"><a href="#cb171-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb171-10"><a href="#cb171-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of RNA-seq"</span>,</span>
<span id="cb171-11"><a href="#cb171-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb171-12"><a href="#cb171-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb171-13"><a href="#cb171-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb171-14"><a href="#cb171-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb171-15"><a href="#cb171-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb171-16"><a href="#cb171-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb171-17"><a href="#cb171-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb171-18"><a href="#cb171-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bar-plot-top-20-degs-1" class="level5">
<h5 class="anchored" data-anchor-id="bar-plot-top-20-degs-1">Bar Plot Top 20 Degs</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a>TxiUpReg <span class="ot">&lt;-</span> TxiResVocPlot[TxiResVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb172-2"><a href="#cb172-2" aria-hidden="true" tabindex="-1"></a>TxiDownReg <span class="ot">&lt;-</span> TxiResVocPlot[TxiResVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb172-3"><a href="#cb172-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb172-4"><a href="#cb172-4" aria-hidden="true" tabindex="-1"></a>TxiUpReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TxiUpReg<span class="sc">$</span>padj)</span>
<span id="cb172-5"><a href="#cb172-5" aria-hidden="true" tabindex="-1"></a>TxiDownReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TxiDownReg<span class="sc">$</span>padj)</span>
<span id="cb172-6"><a href="#cb172-6" aria-hidden="true" tabindex="-1"></a>TxiUpRegSorted <span class="ot">&lt;-</span> TxiUpReg[<span class="fu">order</span>(<span class="sc">-</span>TxiUpReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TxiUpReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb172-7"><a href="#cb172-7" aria-hidden="true" tabindex="-1"></a>TxiDownRegSorted <span class="ot">&lt;-</span> TxiDownReg[<span class="fu">order</span>(TxiDownReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TxiDownReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb172-8"><a href="#cb172-8" aria-hidden="true" tabindex="-1"></a>top_upregTxi <span class="ot">&lt;-</span> <span class="fu">head</span>(TxiUpRegSorted, <span class="dv">20</span>)</span>
<span id="cb172-9"><a href="#cb172-9" aria-hidden="true" tabindex="-1"></a>top_downregTxi <span class="ot">&lt;-</span> <span class="fu">head</span>(TxiDownRegSorted, <span class="dv">20</span>)</span>
<span id="cb172-10"><a href="#cb172-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb172-11"><a href="#cb172-11" aria-hidden="true" tabindex="-1"></a>top_upregTxi<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTxi)</span>
<span id="cb172-12"><a href="#cb172-12" aria-hidden="true" tabindex="-1"></a>top_upregTxi<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregTxi<span class="sc">$</span>transcript_id, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregTxi<span class="sc">$</span>transcript_id))</span>
<span id="cb172-13"><a href="#cb172-13" aria-hidden="true" tabindex="-1"></a>top_upregTxi<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tx2gene<span class="sc">$</span>GENEID[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregTxi), tx2gene<span class="sc">$</span>TXNAME)]</span>
<span id="cb172-14"><a href="#cb172-14" aria-hidden="true" tabindex="-1"></a>top_upregTxi<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(top_upregTxi<span class="sc">$</span>gene_id, MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb172-15"><a href="#cb172-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb172-16"><a href="#cb172-16" aria-hidden="true" tabindex="-1"></a>top_downregTxi<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTxi)</span>
<span id="cb172-17"><a href="#cb172-17" aria-hidden="true" tabindex="-1"></a>top_downregTxi<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregTxi<span class="sc">$</span>transcript_id, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregTxi<span class="sc">$</span>transcript_id))</span>
<span id="cb172-18"><a href="#cb172-18" aria-hidden="true" tabindex="-1"></a>top_downregTxi<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tx2gene<span class="sc">$</span>GENEID[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregTxi), tx2gene<span class="sc">$</span>TXNAME)]</span>
<span id="cb172-19"><a href="#cb172-19" aria-hidden="true" tabindex="-1"></a>top_downregTxi<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(top_downregTxi<span class="sc">$</span>gene_id, MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb172-20"><a href="#cb172-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb172-21"><a href="#cb172-21" aria-hidden="true" tabindex="-1"></a>TopUpDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTxi)</span>
<span id="cb172-22"><a href="#cb172-22" aria-hidden="true" tabindex="-1"></a>TopDownDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTxi)</span>
<span id="cb172-23"><a href="#cb172-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb172-24"><a href="#cb172-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregTxi, <span class="fu">aes</span>(<span class="at">x =</span> transcript_id, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb172-25"><a href="#cb172-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb172-26"><a href="#cb172-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb172-27"><a href="#cb172-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> top_upregTxi<span class="sc">$</span>gene_name) <span class="sc">+</span></span>
<span id="cb172-28"><a href="#cb172-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [Salmon RNAseq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb172-29"><a href="#cb172-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb173"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregTxi, <span class="fu">aes</span>(<span class="at">x =</span> transcript_id, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb173-2"><a href="#cb173-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>   <span class="co"># reddish color for downregulated</span></span>
<span id="cb173-3"><a href="#cb173-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb173-4"><a href="#cb173-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> top_downregTxi<span class="sc">$</span>gene_name) <span class="sc">+</span></span>
<span id="cb173-5"><a href="#cb173-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [Salmon RNAseq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb173-6"><a href="#cb173-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-28-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>#####Scatter Plot Mean Expression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalized <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">counts</span>(ddsTxi, <span class="at">normalize=</span><span class="cn">TRUE</span>))</span>
<span id="cb174-2"><a href="#cb174-2" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalized<span class="sc">$</span>WT_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TxiDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb174-3"><a href="#cb174-3" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalized<span class="sc">$</span>LOW_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TxiDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb174-4"><a href="#cb174-4" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalized<span class="sc">$</span>WT_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TxiDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb174-5"><a href="#cb174-5" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalized<span class="sc">$</span>LOW_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TxiDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb174-6"><a href="#cb174-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-7"><a href="#cb174-7" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalizedFiltered <span class="ot">&lt;-</span> TxiDdsNormalized <span class="sc">%&gt;%</span> <span class="fu">filter</span>(WT_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb174-8"><a href="#cb174-8" aria-hidden="true" tabindex="-1"></a>                                                        (WT_Mean <span class="sc">-</span> WT_std <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb174-9"><a href="#cb174-9" aria-hidden="true" tabindex="-1"></a>                                                        LOW_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb174-10"><a href="#cb174-10" aria-hidden="true" tabindex="-1"></a>                                                        (LOW_Mean <span class="sc">-</span> LOW_std <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb174-11"><a href="#cb174-11" aria-hidden="true" tabindex="-1"></a>TxiResNoNa<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TxiResNoNa)</span>
<span id="cb174-12"><a href="#cb174-12" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalizedFiltered<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TxiDdsNormalizedFiltered)</span>
<span id="cb174-13"><a href="#cb174-13" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalizedFiltered <span class="ot">&lt;-</span> <span class="fu">merge</span>(TxiDdsNormalizedFiltered, <span class="fu">data.frame</span>(TxiResNoNa[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">10</span>)]), <span class="at">by=</span><span class="st">"index"</span>)</span>
<span id="cb174-14"><a href="#cb174-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TxiDdsNormalizedFiltered) <span class="ot">&lt;-</span> TxiDdsNormalizedFiltered<span class="sc">$</span>index</span>
<span id="cb174-15"><a href="#cb174-15" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalizedFiltered <span class="ot">&lt;-</span> TxiDdsNormalizedFiltered[,<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(TxiDdsNormalizedFiltered)]</span>
<span id="cb174-16"><a href="#cb174-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-17"><a href="#cb174-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-18"><a href="#cb174-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TxiDdsNormalizedFiltered, <span class="fu">aes</span>(<span class="at">x=</span> WT_Mean, <span class="at">y=</span> LOW_Mean, <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb174-19"><a href="#cb174-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb174-20"><a href="#cb174-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb174-21"><a href="#cb174-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb174-22"><a href="#cb174-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"Scatter plot of Salmon RNA-seq"</span>,</span>
<span id="cb174-23"><a href="#cb174-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb174-24"><a href="#cb174-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in WT"</span>), </span>
<span id="cb174-25"><a href="#cb174-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in LOW"</span>)) <span class="sc">+</span> </span>
<span id="cb174-26"><a href="#cb174-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb174-27"><a href="#cb174-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"blue"</span>),</span>
<span id="cb174-28"><a href="#cb174-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb174-29"><a href="#cb174-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbar(data = TxiDdsNormalizedFiltered[1:50,] ,aes(ymin = LOW_Mean - LOW_std, ymax = LOW_Mean + LOW_std), width = 0.1, alpha = 0.2) +</span></span>
<span id="cb174-30"><a href="#cb174-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbarh(aes(xmin = WT_Mean - WT_std, xmax = WT_Mean + WT_std), height = 0.1, alpha = 0.2) +</span></span>
<span id="cb174-31"><a href="#cb174-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="correlation-matrix-ssample-1" class="level5">
<h5 class="anchored" data-anchor-id="correlation-matrix-ssample-1">Correlation Matrix SSample</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Genotype =</span> rnaseq_description<span class="sc">$</span>Genotype)</span>
<span id="cb175-2"><a href="#cb175-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(rnaseq_description)</span>
<span id="cb175-3"><a href="#cb175-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-4"><a href="#cb175-4" aria-hidden="true" tabindex="-1"></a>spearman_correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(TxiDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TxiDdsNormalized))], <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb175-5"><a href="#cb175-5" aria-hidden="true" tabindex="-1"></a>dist_from_spearman <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>spearman_correlation_matrix)</span>
<span id="cb175-6"><a href="#cb175-6" aria-hidden="true" tabindex="-1"></a>dist_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(dist_from_spearman)</span>
<span id="cb175-7"><a href="#cb175-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-8"><a href="#cb175-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(dist_matrix, </span>
<span id="cb175-9"><a href="#cb175-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> dist_from_spearman,</span>
<span id="cb175-10"><a href="#cb175-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> dist_from_spearman,</span>
<span id="cb175-11"><a href="#cb175-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> annotation,</span>
<span id="cb175-12"><a href="#cb175-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> annotation,</span>
<span id="cb175-13"><a href="#cb175-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">plasma</span>(<span class="dv">256</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalized<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TxiDdsNormalized)</span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalized.long <span class="ot">&lt;-</span> </span>
<span id="cb176-3"><a href="#cb176-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(TxiDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TxiDdsNormalized)) <span class="sc">|</span> <span class="fu">colnames</span>(TxiDdsNormalized) <span class="sc">==</span> <span class="st">"gene_name"</span>],</span>
<span id="cb176-4"><a href="#cb176-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">id.vars =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb176-5"><a href="#cb176-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">variable_name =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb176-6"><a href="#cb176-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TxiDdsNormalized.long)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"sample_name"</span></span>
<span id="cb176-7"><a href="#cb176-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TxiDdsNormalized.long)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"normalized_count"</span></span>
<span id="cb176-8"><a href="#cb176-8" aria-hidden="true" tabindex="-1"></a>TxiDdsNormalized.long <span class="ot">&lt;-</span> <span class="fu">merge</span>(TxiDdsNormalized.long, rnaseq_description, <span class="at">by=</span><span class="st">"sample_name"</span>)</span>
<span id="cb176-9"><a href="#cb176-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TxiDdsNormalized.long)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     sample_name                    gene_name normalized_count
1 RNA_S11589Nr53         ENSMUST00000000001.4     2.240250e+04
2 RNA_S11589Nr53  PreMrnaENSMUSG00000000001.4     8.728824e+02
3 RNA_S11589Nr53        ENSMUST00000000003.13     0.000000e+00
4 RNA_S11589Nr53         ENSMUST00000114041.2     0.000000e+00
5 RNA_S11589Nr53 PreMrnaENSMUSG00000000003.15     9.570479e-01
6 RNA_S11589Nr53        ENSMUST00000000028.13     7.598029e+03
                        sample                                     fastq_1
1 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
2 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
3 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
4 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
5 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
6 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
                                      fastq_2 strandedness Genotype Colonies
1 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
2 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
3 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
4 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
5 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
6 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
  Method Replicate Treatment
1 RNAseq      Rep1      &lt;NA&gt;
2 RNAseq      Rep1      &lt;NA&gt;
3 RNAseq      Rep1      &lt;NA&gt;
4 RNAseq      Rep1      &lt;NA&gt;
5 RNAseq      Rep1      &lt;NA&gt;
6 RNAseq      Rep1      &lt;NA&gt;</code></pre>
</div>
</div>
</section>
<section id="top-10-up-degs-bar-plot-1" class="level5">
<h5 class="anchored" data-anchor-id="top-10-up-degs-bar-plot-1">Top 10 Up Degs Bar Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>top10_upregTxi <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTxi[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_upregTxi) {</span>
<span id="cb178-3"><a href="#cb178-3" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TxiDdsNormalized.long[TxiDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb178-4"><a href="#cb178-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb178-5"><a href="#cb178-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb178-6"><a href="#cb178-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb178-7"><a href="#cb178-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb178-8"><a href="#cb178-8" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb178-9"><a href="#cb178-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb178-10"><a href="#cb178-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb178-11"><a href="#cb178-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb178-12"><a href="#cb178-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb178-13"><a href="#cb178-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-32-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-down-degs-bar-plot-1" class="level5">
<h5 class="anchored" data-anchor-id="top-10-down-degs-bar-plot-1">Top 10 Down Degs Bar Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a>top10_downregTxi <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTxi[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb179-2"><a href="#cb179-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-3"><a href="#cb179-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_downregTxi) {</span>
<span id="cb179-4"><a href="#cb179-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TxiDdsNormalized.long[TxiDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb179-5"><a href="#cb179-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb179-6"><a href="#cb179-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb179-7"><a href="#cb179-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb179-8"><a href="#cb179-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb179-9"><a href="#cb179-9" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb179-10"><a href="#cb179-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb179-11"><a href="#cb179-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb179-12"><a href="#cb179-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb179-13"><a href="#cb179-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb179-14"><a href="#cb179-14" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-33-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="pre-mrna-only" class="level4">
<h4 class="anchored" data-anchor-id="pre-mrna-only">pre-mRNA Only</h4>
<section id="volcano-plot-2" class="level5">
<h5 class="anchored" data-anchor-id="volcano-plot-2">Volcano Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a>TxiTTResVocPlot <span class="ot">&lt;-</span> TxiTTResNoNa[<span class="fu">order</span>(TxiTTResNoNa<span class="sc">$</span>padj),]</span>
<span id="cb180-2"><a href="#cb180-2" aria-hidden="true" tabindex="-1"></a>TxiTTResVocPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb180-3"><a href="#cb180-3" aria-hidden="true" tabindex="-1"></a>TxiTTResVocPlot<span class="sc">$</span>diffexpressed[TxiTTResVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> TxiTTResVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb180-4"><a href="#cb180-4" aria-hidden="true" tabindex="-1"></a>TxiTTResVocPlot<span class="sc">$</span>diffexpressed[TxiTTResVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> TxiTTResVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb180-5"><a href="#cb180-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-6"><a href="#cb180-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> TxiTTResVocPlot, </span>
<span id="cb180-7"><a href="#cb180-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb180-8"><a href="#cb180-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb180-9"><a href="#cb180-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb180-10"><a href="#cb180-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of RNA-seq"</span>,</span>
<span id="cb180-11"><a href="#cb180-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb180-12"><a href="#cb180-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb180-13"><a href="#cb180-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb180-14"><a href="#cb180-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb180-15"><a href="#cb180-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb180-16"><a href="#cb180-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb180-17"><a href="#cb180-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb180-18"><a href="#cb180-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bar-plot-top-20-degs-2" class="level5">
<h5 class="anchored" data-anchor-id="bar-plot-top-20-degs-2">Bar Plot Top 20 Degs</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a>TxiTTUpReg <span class="ot">&lt;-</span> TxiTTResVocPlot[TxiTTResVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb181-2"><a href="#cb181-2" aria-hidden="true" tabindex="-1"></a>TxiTTDownReg <span class="ot">&lt;-</span> TxiTTResVocPlot[TxiTTResVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb181-3"><a href="#cb181-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-4"><a href="#cb181-4" aria-hidden="true" tabindex="-1"></a>TxiTTUpReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TxiTTUpReg<span class="sc">$</span>padj)</span>
<span id="cb181-5"><a href="#cb181-5" aria-hidden="true" tabindex="-1"></a>TxiTTDownReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TxiTTDownReg<span class="sc">$</span>padj)</span>
<span id="cb181-6"><a href="#cb181-6" aria-hidden="true" tabindex="-1"></a>TxiTTUpRegSorted <span class="ot">&lt;-</span> TxiTTUpReg[<span class="fu">order</span>(<span class="sc">-</span>TxiTTUpReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TxiTTUpReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb181-7"><a href="#cb181-7" aria-hidden="true" tabindex="-1"></a>TxiTTDownRegSorted <span class="ot">&lt;-</span> TxiTTDownReg[<span class="fu">order</span>(TxiTTDownReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TxiTTDownReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb181-8"><a href="#cb181-8" aria-hidden="true" tabindex="-1"></a>top_upregTxiTT <span class="ot">&lt;-</span> <span class="fu">head</span>(TxiTTUpRegSorted, <span class="dv">20</span>)</span>
<span id="cb181-9"><a href="#cb181-9" aria-hidden="true" tabindex="-1"></a>top_downregTxiTT <span class="ot">&lt;-</span> <span class="fu">head</span>(TxiTTDownRegSorted, <span class="dv">20</span>)</span>
<span id="cb181-10"><a href="#cb181-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-11"><a href="#cb181-11" aria-hidden="true" tabindex="-1"></a>top_upregTxiTT<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTxiTT)</span>
<span id="cb181-12"><a href="#cb181-12" aria-hidden="true" tabindex="-1"></a>top_upregTxiTT<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregTxiTT<span class="sc">$</span>transcript_id, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregTxiTT<span class="sc">$</span>transcript_id))</span>
<span id="cb181-13"><a href="#cb181-13" aria-hidden="true" tabindex="-1"></a>top_upregTxiTT<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tx2gene<span class="sc">$</span>GENEID[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregTxiTT), tx2gene<span class="sc">$</span>TXNAME)]</span>
<span id="cb181-14"><a href="#cb181-14" aria-hidden="true" tabindex="-1"></a>top_upregTxiTT<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(top_upregTxiTT<span class="sc">$</span>gene_id, MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb181-15"><a href="#cb181-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-16"><a href="#cb181-16" aria-hidden="true" tabindex="-1"></a>top_downregTxiTT<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTxiTT)</span>
<span id="cb181-17"><a href="#cb181-17" aria-hidden="true" tabindex="-1"></a>top_downregTxiTT<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregTxiTT<span class="sc">$</span>transcript_id, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregTxiTT<span class="sc">$</span>transcript_id))</span>
<span id="cb181-18"><a href="#cb181-18" aria-hidden="true" tabindex="-1"></a>top_downregTxiTT<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tx2gene<span class="sc">$</span>GENEID[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregTxiTT), tx2gene<span class="sc">$</span>TXNAME)]</span>
<span id="cb181-19"><a href="#cb181-19" aria-hidden="true" tabindex="-1"></a>top_downregTxiTT<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(top_downregTxiTT<span class="sc">$</span>gene_id, MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb181-20"><a href="#cb181-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-21"><a href="#cb181-21" aria-hidden="true" tabindex="-1"></a>TopUpDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTxiTT)</span>
<span id="cb181-22"><a href="#cb181-22" aria-hidden="true" tabindex="-1"></a>TopDownDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTxiTT)</span>
<span id="cb181-23"><a href="#cb181-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-24"><a href="#cb181-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregTxiTT, <span class="fu">aes</span>(<span class="at">x =</span> transcript_id, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb181-25"><a href="#cb181-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb181-26"><a href="#cb181-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb181-27"><a href="#cb181-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> top_upregTxiTT<span class="sc">$</span>gene_name) <span class="sc">+</span></span>
<span id="cb181-28"><a href="#cb181-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [Salmon RNAseq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb181-29"><a href="#cb181-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregTxiTT, <span class="fu">aes</span>(<span class="at">x =</span> transcript_id, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb182-2"><a href="#cb182-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>   <span class="co"># reddish color for downregulated</span></span>
<span id="cb182-3"><a href="#cb182-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb182-4"><a href="#cb182-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> top_downregTxiTT<span class="sc">$</span>gene_name) <span class="sc">+</span></span>
<span id="cb182-5"><a href="#cb182-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [Salmon RNAseq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb182-6"><a href="#cb182-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-35-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>#####Scatter Plot Mean Expression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb183"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalized <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">counts</span>(ddsTxiTT, <span class="at">normalize=</span><span class="cn">TRUE</span>))</span>
<span id="cb183-2"><a href="#cb183-2" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalized<span class="sc">$</span>WT_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TxiTTDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb183-3"><a href="#cb183-3" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalized<span class="sc">$</span>LOW_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TxiTTDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb183-4"><a href="#cb183-4" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalized<span class="sc">$</span>WT_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TxiTTDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb183-5"><a href="#cb183-5" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalized<span class="sc">$</span>LOW_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TxiTTDdsNormalized[,<span class="fu">rownames</span>(rnaseq_description[rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb183-6"><a href="#cb183-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-7"><a href="#cb183-7" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalizedFiltered <span class="ot">&lt;-</span> TxiTTDdsNormalized <span class="sc">%&gt;%</span> <span class="fu">filter</span>(WT_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb183-8"><a href="#cb183-8" aria-hidden="true" tabindex="-1"></a>                                                        (WT_Mean <span class="sc">-</span> WT_std <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb183-9"><a href="#cb183-9" aria-hidden="true" tabindex="-1"></a>                                                        LOW_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb183-10"><a href="#cb183-10" aria-hidden="true" tabindex="-1"></a>                                                        (LOW_Mean <span class="sc">-</span> LOW_std <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb183-11"><a href="#cb183-11" aria-hidden="true" tabindex="-1"></a>TxiResNoNa<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TxiResNoNa)</span>
<span id="cb183-12"><a href="#cb183-12" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalizedFiltered<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TxiTTDdsNormalizedFiltered)</span>
<span id="cb183-13"><a href="#cb183-13" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalizedFiltered <span class="ot">&lt;-</span> <span class="fu">merge</span>(TxiTTDdsNormalizedFiltered, <span class="fu">data.frame</span>(TxiResNoNa[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">10</span>)]), <span class="at">by=</span><span class="st">"index"</span>)</span>
<span id="cb183-14"><a href="#cb183-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TxiTTDdsNormalizedFiltered) <span class="ot">&lt;-</span> TxiTTDdsNormalizedFiltered<span class="sc">$</span>index</span>
<span id="cb183-15"><a href="#cb183-15" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalizedFiltered <span class="ot">&lt;-</span> TxiTTDdsNormalizedFiltered[,<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(TxiTTDdsNormalizedFiltered)]</span>
<span id="cb183-16"><a href="#cb183-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-17"><a href="#cb183-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-18"><a href="#cb183-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TxiTTDdsNormalizedFiltered, <span class="fu">aes</span>(<span class="at">x=</span> WT_Mean, <span class="at">y=</span> LOW_Mean, <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb183-19"><a href="#cb183-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb183-20"><a href="#cb183-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb183-21"><a href="#cb183-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb183-22"><a href="#cb183-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"Scatter plot of Salmon RNA-seq"</span>,</span>
<span id="cb183-23"><a href="#cb183-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb183-24"><a href="#cb183-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in WT"</span>), </span>
<span id="cb183-25"><a href="#cb183-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in LOW"</span>)) <span class="sc">+</span> </span>
<span id="cb183-26"><a href="#cb183-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb183-27"><a href="#cb183-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"blue"</span>),</span>
<span id="cb183-28"><a href="#cb183-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb183-29"><a href="#cb183-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbar(data = TxiDdsNormalizedFiltered[1:50,] ,aes(ymin = LOW_Mean - LOW_std, ymax = LOW_Mean + LOW_std), width = 0.1, alpha = 0.2) +</span></span>
<span id="cb183-30"><a href="#cb183-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbarh(aes(xmin = WT_Mean - WT_std, xmax = WT_Mean + WT_std), height = 0.1, alpha = 0.2) +</span></span>
<span id="cb183-31"><a href="#cb183-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="correlation-matrix-ssample-2" class="level5">
<h5 class="anchored" data-anchor-id="correlation-matrix-ssample-2">Correlation Matrix SSample</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Genotype =</span> rnaseq_description<span class="sc">$</span>Genotype)</span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(rnaseq_description)</span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-4"><a href="#cb184-4" aria-hidden="true" tabindex="-1"></a>spearman_correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(TxiTTDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TxiTTDdsNormalized))], <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb184-5"><a href="#cb184-5" aria-hidden="true" tabindex="-1"></a>dist_from_spearman <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>spearman_correlation_matrix)</span>
<span id="cb184-6"><a href="#cb184-6" aria-hidden="true" tabindex="-1"></a>dist_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(dist_from_spearman)</span>
<span id="cb184-7"><a href="#cb184-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-8"><a href="#cb184-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(dist_matrix, </span>
<span id="cb184-9"><a href="#cb184-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> dist_from_spearman,</span>
<span id="cb184-10"><a href="#cb184-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> dist_from_spearman,</span>
<span id="cb184-11"><a href="#cb184-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> annotation,</span>
<span id="cb184-12"><a href="#cb184-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> annotation,</span>
<span id="cb184-13"><a href="#cb184-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">plasma</span>(<span class="dv">256</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalized<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TxiTTDdsNormalized)</span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalized.long <span class="ot">&lt;-</span> </span>
<span id="cb185-3"><a href="#cb185-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(TxiTTDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TxiTTDdsNormalized)) <span class="sc">|</span> <span class="fu">colnames</span>(TxiTTDdsNormalized) <span class="sc">==</span> <span class="st">"gene_name"</span>],</span>
<span id="cb185-4"><a href="#cb185-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">id.vars =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb185-5"><a href="#cb185-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">variable_name =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb185-6"><a href="#cb185-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TxiTTDdsNormalized.long)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"sample_name"</span></span>
<span id="cb185-7"><a href="#cb185-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TxiTTDdsNormalized.long)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"normalized_count"</span></span>
<span id="cb185-8"><a href="#cb185-8" aria-hidden="true" tabindex="-1"></a>TxiTTDdsNormalized.long <span class="ot">&lt;-</span> <span class="fu">merge</span>(TxiTTDdsNormalized.long, rnaseq_description, <span class="at">by=</span><span class="st">"sample_name"</span>)</span>
<span id="cb185-9"><a href="#cb185-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TxiTTDdsNormalized.long)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     sample_name                    gene_name normalized_count
1 RNA_S11589Nr53  PreMrnaENSMUSG00000000001.4      872.8824221
2 RNA_S11589Nr53 PreMrnaENSMUSG00000000003.15        0.9570479
3 RNA_S11589Nr53         ENSMUST00000231819.1        0.0000000
4 RNA_S11589Nr53 PreMrnaENSMUSG00000000028.15     1951.5936078
5 RNA_S11589Nr53 PreMrnaENSMUSG00000000031.16        3.9130132
6 RNA_S11589Nr53 PreMrnaENSMUSG00000000037.17      173.6732197
                        sample                                     fastq_1
1 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
2 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
3 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
4 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
5 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
6 H1_WT_C1_S2spike_RNAseq_Rep1 S11589_1/01_fastq/RNA_S11589Nr53.1.fastq.gz
                                      fastq_2 strandedness Genotype Colonies
1 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
2 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
3 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
4 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
5 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
6 S11589_1/01_fastq/RNA_S11589Nr53.2.fastq.gz         auto       WT       C1
  Method Replicate Treatment
1 RNAseq      Rep1      &lt;NA&gt;
2 RNAseq      Rep1      &lt;NA&gt;
3 RNAseq      Rep1      &lt;NA&gt;
4 RNAseq      Rep1      &lt;NA&gt;
5 RNAseq      Rep1      &lt;NA&gt;
6 RNAseq      Rep1      &lt;NA&gt;</code></pre>
</div>
</div>
</section>
<section id="top-10-up-degs-bar-plot-2" class="level5">
<h5 class="anchored" data-anchor-id="top-10-up-degs-bar-plot-2">Top 10 Up Degs Bar Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a>top10_upregTxiTT <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTxiTT[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_upregTxiTT) {</span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TxiTTDdsNormalized.long[TxiTTDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb187-4"><a href="#cb187-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb187-5"><a href="#cb187-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb187-6"><a href="#cb187-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb187-7"><a href="#cb187-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb187-8"><a href="#cb187-8" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb187-9"><a href="#cb187-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb187-10"><a href="#cb187-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb187-11"><a href="#cb187-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb187-12"><a href="#cb187-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb187-13"><a href="#cb187-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-39-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-down-degs-bar-plot-2" class="level5">
<h5 class="anchored" data-anchor-id="top-10-down-degs-bar-plot-2">Top 10 Down Degs Bar Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb188"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb188-1"><a href="#cb188-1" aria-hidden="true" tabindex="-1"></a>top10_downregTxiTT <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTxiTT[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb188-2"><a href="#cb188-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-3"><a href="#cb188-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_downregTxiTT) {</span>
<span id="cb188-4"><a href="#cb188-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TxiTTDdsNormalized.long[TxiTTDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb188-5"><a href="#cb188-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb188-6"><a href="#cb188-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb188-7"><a href="#cb188-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb188-8"><a href="#cb188-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb188-9"><a href="#cb188-9" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb188-10"><a href="#cb188-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb188-11"><a href="#cb188-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb188-12"><a href="#cb188-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb188-13"><a href="#cb188-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb188-14"><a href="#cb188-14" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-40-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="c1-removed-premrna" class="level4">
<h4 class="anchored" data-anchor-id="c1-removed-premrna">C1-Removed PremRNA</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a>negateC1 <span class="ot">&lt;-</span> <span class="sc">!</span>(rnaseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span> <span class="sc">&amp;</span> rnaseq_description<span class="sc">$</span>Colonies<span class="sc">==</span><span class="st">"C1"</span>)</span>
<span id="cb189-2"><a href="#cb189-2" aria-hidden="true" tabindex="-1"></a>rnaseq_description_C1_removed <span class="ot">&lt;-</span> rnaseq_description[negateC1,]</span>
<span id="cb189-3"><a href="#cb189-3" aria-hidden="true" tabindex="-1"></a>txi_C1Removed <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb189-4"><a href="#cb189-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">abundance =</span> txi_tx<span class="sc">$</span>abundance[,negateC1],</span>
<span id="cb189-5"><a href="#cb189-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> txi_tx<span class="sc">$</span>counts[,negateC1],</span>
<span id="cb189-6"><a href="#cb189-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">length =</span> txi_tx<span class="sc">$</span>length[,negateC1],</span>
<span id="cb189-7"><a href="#cb189-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">countsFromAbundance =</span> txi_tx<span class="sc">$</span>countsFromAbundance</span>
<span id="cb189-8"><a href="#cb189-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb189-9"><a href="#cb189-9" aria-hidden="true" tabindex="-1"></a>ddsTxi_C1Removed <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_C1Removed,</span>
<span id="cb189-10"><a href="#cb189-10" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">colData =</span> rnaseq_description_C1_removed,</span>
<span id="cb189-11"><a href="#cb189-11" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">design =</span> <span class="sc">~</span> Genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using counts and average transcript lengths from tximport</code></pre>
</div>
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a>ddsTxi_C1Removed <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(ddsTxi_C1Removed, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"^FBtr|^PreMrnaDroso"</span>, <span class="fu">rownames</span>(ddsTxi_C1Removed)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'avgTxLength' from assays(dds), correcting for library size</code></pre>
</div>
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a>premrna_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ddsTxi_C1Removed[<span class="fu">grepl</span>(<span class="st">"^PreMrnaENS"</span>, <span class="fu">rownames</span>(ddsTxi_C1Removed))])</span>
<span id="cb194-2"><a href="#cb194-2" aria-hidden="true" tabindex="-1"></a>premrna_id <span class="ot">&lt;-</span> <span class="fu">c</span>(premrna_id, mono_exon_id)</span>
<span id="cb194-3"><a href="#cb194-3" aria-hidden="true" tabindex="-1"></a>ddsTxi_C1Removed <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ddsTxi_C1Removed[<span class="fu">rownames</span>(ddsTxi_C1Removed) <span class="sc">%in%</span> premrna_id,])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing normalization factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a>TxiResC1Removed <span class="ot">&lt;-</span> <span class="fu">results</span>(ddsTxi_C1Removed, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a>TxiResNoNaC1Removed <span class="ot">&lt;-</span> TxiResC1Removed[<span class="sc">!</span><span class="fu">is.na</span>(TxiResC1Removed<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(TxiResC1Removed<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a>TxiResNoNaC1Removed<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb201-4"><a href="#cb201-4" aria-hidden="true" tabindex="-1"></a>TxiResNoNaC1Removed<span class="sc">$</span>significant[TxiResNoNaC1Removed<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="ma-plot" class="level5">
<h5 class="anchored" data-anchor-id="ma-plot">MA Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(TxiResNoNaC1Removed)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TxiResNoNaC1Removed, <span class="fu">aes</span>(<span class="at">x =</span>TxiResNoNaC1Removed<span class="sc">$</span>baseMean, </span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> TxiResNoNaC1Removed<span class="sc">$</span>log2FoldChange, </span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb203-4"><a href="#cb203-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb203-5"><a href="#cb203-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb203-6"><a href="#cb203-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from Salmon RNA-seq C1 Removed"</span>,</span>
<span id="cb203-7"><a href="#cb203-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb203-8"><a href="#cb203-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb203-9"><a href="#cb203-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb203-10"><a href="#cb203-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb203-11"><a href="#cb203-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb203-12"><a href="#cb203-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb203-13"><a href="#cb203-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb203-14"><a href="#cb203-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb203-15"><a href="#cb203-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-42-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="volcano-plot-3" class="level5">
<h5 class="anchored" data-anchor-id="volcano-plot-3">Volcano Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a>TxiTTC1ResVocPlot <span class="ot">&lt;-</span> TxiResNoNaC1Removed[<span class="fu">order</span>(TxiResNoNaC1Removed<span class="sc">$</span>padj),]</span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a>TxiTTC1ResVocPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb204-3"><a href="#cb204-3" aria-hidden="true" tabindex="-1"></a>TxiTTC1ResVocPlot<span class="sc">$</span>diffexpressed[TxiTTC1ResVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> TxiTTC1ResVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb204-4"><a href="#cb204-4" aria-hidden="true" tabindex="-1"></a>TxiTTC1ResVocPlot<span class="sc">$</span>diffexpressed[TxiTTC1ResVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> TxiTTC1ResVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb204-5"><a href="#cb204-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb204-6"><a href="#cb204-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> TxiTTC1ResVocPlot, </span>
<span id="cb204-7"><a href="#cb204-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb204-8"><a href="#cb204-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb204-9"><a href="#cb204-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb204-10"><a href="#cb204-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of Salmon RNA-seq C1 removed"</span>,</span>
<span id="cb204-11"><a href="#cb204-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb204-12"><a href="#cb204-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb204-13"><a href="#cb204-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb204-14"><a href="#cb204-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb204-15"><a href="#cb204-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb204-16"><a href="#cb204-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb204-17"><a href="#cb204-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb204-18"><a href="#cb204-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bar-plot-top-20-degs-3" class="level5">
<h5 class="anchored" data-anchor-id="bar-plot-top-20-degs-3">Bar Plot Top 20 Degs</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a>TxiTTUpRegC1 <span class="ot">&lt;-</span> TxiTTC1ResVocPlot[TxiTTC1ResVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a>TxiTTDownRegC1 <span class="ot">&lt;-</span> TxiTTC1ResVocPlot[TxiTTC1ResVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb205-3"><a href="#cb205-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-4"><a href="#cb205-4" aria-hidden="true" tabindex="-1"></a>TxiTTUpRegC1<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TxiTTUpRegC1<span class="sc">$</span>padj)</span>
<span id="cb205-5"><a href="#cb205-5" aria-hidden="true" tabindex="-1"></a>TxiTTDownRegC1<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TxiTTDownRegC1<span class="sc">$</span>padj)</span>
<span id="cb205-6"><a href="#cb205-6" aria-hidden="true" tabindex="-1"></a>TxiTTUpRegSortedC1 <span class="ot">&lt;-</span> TxiTTUpRegC1[<span class="fu">order</span>(<span class="sc">-</span>TxiTTUpRegC1<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TxiTTUpRegC1<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb205-7"><a href="#cb205-7" aria-hidden="true" tabindex="-1"></a>TxiTTDownRegSortedC1 <span class="ot">&lt;-</span> TxiTTDownRegC1[<span class="fu">order</span>(TxiTTDownRegC1<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TxiTTDownRegC1<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb205-8"><a href="#cb205-8" aria-hidden="true" tabindex="-1"></a>top_upregTxiTTC1 <span class="ot">&lt;-</span> <span class="fu">head</span>(TxiTTUpRegSortedC1, <span class="dv">20</span>)</span>
<span id="cb205-9"><a href="#cb205-9" aria-hidden="true" tabindex="-1"></a>top_downregTxiTTC1 <span class="ot">&lt;-</span> <span class="fu">head</span>(TxiTTDownRegSortedC1, <span class="dv">20</span>)</span>
<span id="cb205-10"><a href="#cb205-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-11"><a href="#cb205-11" aria-hidden="true" tabindex="-1"></a>top_upregTxiTTC1<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTxiTTC1)</span>
<span id="cb205-12"><a href="#cb205-12" aria-hidden="true" tabindex="-1"></a>top_upregTxiTTC1<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregTxiTTC1<span class="sc">$</span>transcript_id, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregTxiTTC1<span class="sc">$</span>transcript_id))</span>
<span id="cb205-13"><a href="#cb205-13" aria-hidden="true" tabindex="-1"></a>top_upregTxiTTC1<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tx2gene<span class="sc">$</span>GENEID[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregTxiTTC1), tx2gene<span class="sc">$</span>TXNAME)]</span>
<span id="cb205-14"><a href="#cb205-14" aria-hidden="true" tabindex="-1"></a>top_upregTxiTTC1<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(top_upregTxiTTC1<span class="sc">$</span>gene_id, MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb205-15"><a href="#cb205-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-16"><a href="#cb205-16" aria-hidden="true" tabindex="-1"></a>top_downregTxiTTC1<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTxiTTC1)</span>
<span id="cb205-17"><a href="#cb205-17" aria-hidden="true" tabindex="-1"></a>top_downregTxiTTC1<span class="sc">$</span>transcript_id <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregTxiTTC1<span class="sc">$</span>transcript_id, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregTxiTTC1<span class="sc">$</span>transcript_id))</span>
<span id="cb205-18"><a href="#cb205-18" aria-hidden="true" tabindex="-1"></a>top_downregTxiTTC1<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tx2gene<span class="sc">$</span>GENEID[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregTxiTTC1), tx2gene<span class="sc">$</span>TXNAME)]</span>
<span id="cb205-19"><a href="#cb205-19" aria-hidden="true" tabindex="-1"></a>top_downregTxiTTC1<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(top_downregTxiTTC1<span class="sc">$</span>gene_id, MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb205-20"><a href="#cb205-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-21"><a href="#cb205-21" aria-hidden="true" tabindex="-1"></a>TopUpDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTxiTTC1)</span>
<span id="cb205-22"><a href="#cb205-22" aria-hidden="true" tabindex="-1"></a>TopDownDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTxiTTC1)</span>
<span id="cb205-23"><a href="#cb205-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-24"><a href="#cb205-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregTxiTTC1, <span class="fu">aes</span>(<span class="at">x =</span> transcript_id, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb205-25"><a href="#cb205-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb205-26"><a href="#cb205-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb205-27"><a href="#cb205-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> top_upregTxiTTC1<span class="sc">$</span>gene_name) <span class="sc">+</span></span>
<span id="cb205-28"><a href="#cb205-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [Salmon RNAseq C1 removed]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb205-29"><a href="#cb205-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregTxiTTC1, <span class="fu">aes</span>(<span class="at">x =</span> transcript_id, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>   <span class="co"># reddish color for downregulated</span></span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb206-4"><a href="#cb206-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> top_downregTxiTTC1<span class="sc">$</span>gene_name) <span class="sc">+</span></span>
<span id="cb206-5"><a href="#cb206-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [Salmon RNAseq C1 removed]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb206-6"><a href="#cb206-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-44-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="c1-removed" class="level3">
<h3 class="anchored" data-anchor-id="c1-removed">C1-Removed</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a>rnaseq_C1_removed <span class="ot">&lt;-</span> star_rnaseq[,<span class="fu">rownames</span>(rnaseq_description_C1_removed)]</span>
<span id="cb207-2"><a href="#cb207-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb207-3"><a href="#cb207-3" aria-hidden="true" tabindex="-1"></a>rnaseq_dds_C1Removed <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> rnaseq_C1_removed,</span>
<span id="cb207-4"><a href="#cb207-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> rnaseq_description_C1_removed,</span>
<span id="cb207-5"><a href="#cb207-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span> Genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a>rnaseq_dds_C1Removed <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(rnaseq_dds_C1Removed, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"Droso_"</span>, <span class="fu">rownames</span>(rnaseq_C1_removed)))</span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a>rnaseq_dds_C1Removed <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(rnaseq_dds_C1Removed[<span class="fu">grepl</span>(<span class="st">"^ENS"</span>, <span class="fu">rownames</span>(rnaseq_C1_removed))])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a>RnaResC1Removed <span class="ot">&lt;-</span> <span class="fu">results</span>(rnaseq_dds_C1Removed, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a>RnaResNoNaC1Removed <span class="ot">&lt;-</span> RnaResC1Removed[<span class="sc">!</span><span class="fu">is.na</span>(RnaResC1Removed<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(RnaResC1Removed<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb216-3"><a href="#cb216-3" aria-hidden="true" tabindex="-1"></a>RnaResNoNaC1Removed<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb216-4"><a href="#cb216-4" aria-hidden="true" tabindex="-1"></a>RnaResNoNaC1Removed<span class="sc">$</span>significant[RnaResNoNaC1Removed<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="ma-plot-1" class="level4">
<h4 class="anchored" data-anchor-id="ma-plot-1">MA Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb217"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb217-1"><a href="#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(RnaResNoNaC1Removed)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb218"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb218-1"><a href="#cb218-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(RnaResNoNaC1Removed, <span class="fu">aes</span>(<span class="at">x =</span>RnaResNoNaC1Removed<span class="sc">$</span>baseMean, </span>
<span id="cb218-2"><a href="#cb218-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> RnaResNoNaC1Removed<span class="sc">$</span>log2FoldChange, </span>
<span id="cb218-3"><a href="#cb218-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb218-4"><a href="#cb218-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb218-5"><a href="#cb218-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb218-6"><a href="#cb218-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from Star RNA-seq C1 Removed"</span>,</span>
<span id="cb218-7"><a href="#cb218-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb218-8"><a href="#cb218-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb218-9"><a href="#cb218-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb218-10"><a href="#cb218-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb218-11"><a href="#cb218-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb218-12"><a href="#cb218-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb218-13"><a href="#cb218-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb218-14"><a href="#cb218-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb218-15"><a href="#cb218-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-46-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="volcano-plot-4" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-4">Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb219"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb219-1"><a href="#cb219-1" aria-hidden="true" tabindex="-1"></a>RnaResNoNaC1RemovedVocPlot <span class="ot">&lt;-</span> RnaResNoNaC1Removed[<span class="fu">order</span>(RnaResNoNaC1Removed<span class="sc">$</span>padj),]</span>
<span id="cb219-2"><a href="#cb219-2" aria-hidden="true" tabindex="-1"></a>RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb219-3"><a href="#cb219-3" aria-hidden="true" tabindex="-1"></a>RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>diffexpressed[RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb219-4"><a href="#cb219-4" aria-hidden="true" tabindex="-1"></a>RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>diffexpressed[RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb219-5"><a href="#cb219-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-6"><a href="#cb219-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> RnaResNoNaC1RemovedVocPlot, </span>
<span id="cb219-7"><a href="#cb219-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb219-8"><a href="#cb219-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb219-9"><a href="#cb219-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb219-10"><a href="#cb219-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of Star RNA-seq C1 removed"</span>,</span>
<span id="cb219-11"><a href="#cb219-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb219-12"><a href="#cb219-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb219-13"><a href="#cb219-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb219-14"><a href="#cb219-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb219-15"><a href="#cb219-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb219-16"><a href="#cb219-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb219-17"><a href="#cb219-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb219-18"><a href="#cb219-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bar-plot-top-20-degs-4" class="level4">
<h4 class="anchored" data-anchor-id="bar-plot-top-20-degs-4">Bar Plot Top 20 Degs</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb220"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb220-1"><a href="#cb220-1" aria-hidden="true" tabindex="-1"></a>RnaUpRegC1 <span class="ot">&lt;-</span> RnaResNoNaC1RemovedVocPlot[RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb220-2"><a href="#cb220-2" aria-hidden="true" tabindex="-1"></a>RnaDownRegC1 <span class="ot">&lt;-</span> RnaResNoNaC1RemovedVocPlot[RnaResNoNaC1RemovedVocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb220-3"><a href="#cb220-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb220-4"><a href="#cb220-4" aria-hidden="true" tabindex="-1"></a>RnaUpRegC1<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(RnaUpRegC1<span class="sc">$</span>padj)</span>
<span id="cb220-5"><a href="#cb220-5" aria-hidden="true" tabindex="-1"></a>RnaDownRegC1<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(RnaDownRegC1<span class="sc">$</span>padj)</span>
<span id="cb220-6"><a href="#cb220-6" aria-hidden="true" tabindex="-1"></a>RnaUpRegSortedC1 <span class="ot">&lt;-</span> RnaUpRegC1[<span class="fu">order</span>(<span class="sc">-</span>RnaUpRegC1<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>RnaUpRegC1<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb220-7"><a href="#cb220-7" aria-hidden="true" tabindex="-1"></a>RnaDownRegSortedC1 <span class="ot">&lt;-</span> RnaDownRegC1[<span class="fu">order</span>(RnaDownRegC1<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>RnaDownRegC1<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb220-8"><a href="#cb220-8" aria-hidden="true" tabindex="-1"></a>top_upregRnaC1 <span class="ot">&lt;-</span> <span class="fu">head</span>(RnaUpRegSortedC1, <span class="dv">20</span>)</span>
<span id="cb220-9"><a href="#cb220-9" aria-hidden="true" tabindex="-1"></a>top_downregRnaC1 <span class="ot">&lt;-</span> <span class="fu">head</span>(RnaDownRegSortedC1, <span class="dv">20</span>)</span>
<span id="cb220-10"><a href="#cb220-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb220-11"><a href="#cb220-11" aria-hidden="true" tabindex="-1"></a>top_upregRnaC1<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregRnaC1), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb220-12"><a href="#cb220-12" aria-hidden="true" tabindex="-1"></a>top_upregRnaC1<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregRnaC1<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregRnaC1<span class="sc">$</span>gene))</span>
<span id="cb220-13"><a href="#cb220-13" aria-hidden="true" tabindex="-1"></a>top_downregRnaC1<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregRnaC1), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb220-14"><a href="#cb220-14" aria-hidden="true" tabindex="-1"></a>top_downregRnaC1<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregRnaC1<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregRnaC1<span class="sc">$</span>gene))</span>
<span id="cb220-15"><a href="#cb220-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb220-16"><a href="#cb220-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregRnaC1, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb220-17"><a href="#cb220-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb220-18"><a href="#cb220-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb220-19"><a href="#cb220-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [STAR RNAseq C1 removed]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb220-20"><a href="#cb220-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb221"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb221-1"><a href="#cb221-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregRnaC1, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb221-2"><a href="#cb221-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>   <span class="co"># reddish color for downregulated</span></span>
<span id="cb221-3"><a href="#cb221-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb221-4"><a href="#cb221-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [STAR RNAseq C1 removed]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb221-5"><a href="#cb221-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-48-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="tt-seq-analysis" class="level2">
<h2 class="anchored" data-anchor-id="tt-seq-analysis">TT-Seq Analysis</h2>
<section id="star-1" class="level3">
<h3 class="anchored" data-anchor-id="star-1">STAR</h3>
<section id="volcano-plot-5" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-5">Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb222"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb222-1"><a href="#cb222-1" aria-hidden="true" tabindex="-1"></a>TTresVolcPlot <span class="ot">&lt;-</span> TTResNoNa[<span class="fu">order</span>(TTResNoNa<span class="sc">$</span>padj),]</span>
<span id="cb222-2"><a href="#cb222-2" aria-hidden="true" tabindex="-1"></a>TTresVolcPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb222-3"><a href="#cb222-3" aria-hidden="true" tabindex="-1"></a>TTresVolcPlot<span class="sc">$</span>diffexpressed[TTresVolcPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> TTresVolcPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb222-4"><a href="#cb222-4" aria-hidden="true" tabindex="-1"></a>TTresVolcPlot<span class="sc">$</span>diffexpressed[TTresVolcPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> TTresVolcPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb222-5"><a href="#cb222-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb222-6"><a href="#cb222-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> TTresVolcPlot, </span>
<span id="cb222-7"><a href="#cb222-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb222-8"><a href="#cb222-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb222-9"><a href="#cb222-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb222-10"><a href="#cb222-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of STAR TT-seq"</span>,</span>
<span id="cb222-11"><a href="#cb222-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb222-12"><a href="#cb222-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb222-13"><a href="#cb222-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb222-14"><a href="#cb222-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb222-15"><a href="#cb222-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb222-16"><a href="#cb222-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb222-17"><a href="#cb222-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb222-18"><a href="#cb222-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-49-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bar-plot-top-20-degs-5" class="level4">
<h4 class="anchored" data-anchor-id="bar-plot-top-20-degs-5">Bar Plot Top 20 Degs</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb223"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb223-1"><a href="#cb223-1" aria-hidden="true" tabindex="-1"></a>TTUpReg <span class="ot">&lt;-</span> TTresVolcPlot[TTresVolcPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb223-2"><a href="#cb223-2" aria-hidden="true" tabindex="-1"></a>TTDownReg <span class="ot">&lt;-</span> TTresVolcPlot[TTresVolcPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb223-3"><a href="#cb223-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb223-4"><a href="#cb223-4" aria-hidden="true" tabindex="-1"></a>TTUpReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TTUpReg<span class="sc">$</span>padj)</span>
<span id="cb223-5"><a href="#cb223-5" aria-hidden="true" tabindex="-1"></a>TTDownReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TTDownReg<span class="sc">$</span>padj)</span>
<span id="cb223-6"><a href="#cb223-6" aria-hidden="true" tabindex="-1"></a>TTUpRegSorted <span class="ot">&lt;-</span> TTUpReg[<span class="fu">order</span>(<span class="sc">-</span>TTUpReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TTUpReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb223-7"><a href="#cb223-7" aria-hidden="true" tabindex="-1"></a>TTDownRegSorted <span class="ot">&lt;-</span> TTDownReg[<span class="fu">order</span>(TTDownReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TTDownReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb223-8"><a href="#cb223-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb223-9"><a href="#cb223-9" aria-hidden="true" tabindex="-1"></a>top_upregTT <span class="ot">&lt;-</span> <span class="fu">head</span>(TTUpRegSorted, <span class="dv">20</span>)</span>
<span id="cb223-10"><a href="#cb223-10" aria-hidden="true" tabindex="-1"></a>top_downregTT <span class="ot">&lt;-</span> <span class="fu">head</span>(TTDownRegSorted, <span class="dv">20</span>)</span>
<span id="cb223-11"><a href="#cb223-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb223-12"><a href="#cb223-12" aria-hidden="true" tabindex="-1"></a>top_upregTT<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregTT), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb223-13"><a href="#cb223-13" aria-hidden="true" tabindex="-1"></a>top_upregTT<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregTT<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregTT<span class="sc">$</span>gene))</span>
<span id="cb223-14"><a href="#cb223-14" aria-hidden="true" tabindex="-1"></a>top_downregTT<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregTT), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb223-15"><a href="#cb223-15" aria-hidden="true" tabindex="-1"></a>top_downregTT<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregTT<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregTT<span class="sc">$</span>gene))</span>
<span id="cb223-16"><a href="#cb223-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb223-17"><a href="#cb223-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregTT, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb223-18"><a href="#cb223-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb223-19"><a href="#cb223-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb223-20"><a href="#cb223-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [STAR TT-seq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb223-21"><a href="#cb223-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb224"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb224-1"><a href="#cb224-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregTT, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb224-2"><a href="#cb224-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>   </span>
<span id="cb224-3"><a href="#cb224-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb224-4"><a href="#cb224-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [STAR TT-seq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb224-5"><a href="#cb224-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-50-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>####Scatter Plot Mean Expression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb225"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb225-1"><a href="#cb225-1" aria-hidden="true" tabindex="-1"></a>TTDdsNormalized <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">counts</span>(ttseq_dds, <span class="at">normalize=</span><span class="cn">TRUE</span>))</span>
<span id="cb225-2"><a href="#cb225-2" aria-hidden="true" tabindex="-1"></a>TTDdsNormalized<span class="sc">$</span>WT_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTDdsNormalized[,<span class="fu">rownames</span>(ttseq_description[ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb225-3"><a href="#cb225-3" aria-hidden="true" tabindex="-1"></a>TTDdsNormalized<span class="sc">$</span>LOW_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTDdsNormalized[,<span class="fu">rownames</span>(ttseq_description[ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb225-4"><a href="#cb225-4" aria-hidden="true" tabindex="-1"></a>TTDdsNormalized<span class="sc">$</span>WT_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTDdsNormalized[,<span class="fu">rownames</span>(ttseq_description[ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb225-5"><a href="#cb225-5" aria-hidden="true" tabindex="-1"></a>TTDdsNormalized<span class="sc">$</span>LOW_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTDdsNormalized[,<span class="fu">rownames</span>(ttseq_description[ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb225-6"><a href="#cb225-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb225-7"><a href="#cb225-7" aria-hidden="true" tabindex="-1"></a>TTDdsNormalizedFiltered <span class="ot">&lt;-</span> TTDdsNormalized <span class="sc">%&gt;%</span> <span class="fu">filter</span>(WT_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb225-8"><a href="#cb225-8" aria-hidden="true" tabindex="-1"></a>                                                        (WT_Mean <span class="sc">-</span> WT_std <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb225-9"><a href="#cb225-9" aria-hidden="true" tabindex="-1"></a>                                                        LOW_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb225-10"><a href="#cb225-10" aria-hidden="true" tabindex="-1"></a>                                                        (LOW_Mean <span class="sc">-</span> LOW_std <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb225-11"><a href="#cb225-11" aria-hidden="true" tabindex="-1"></a>TTResNoNa<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTResNoNa)</span>
<span id="cb225-12"><a href="#cb225-12" aria-hidden="true" tabindex="-1"></a>TTDdsNormalizedFiltered<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTDdsNormalizedFiltered)</span>
<span id="cb225-13"><a href="#cb225-13" aria-hidden="true" tabindex="-1"></a>TTDdsNormalizedFiltered <span class="ot">&lt;-</span> <span class="fu">merge</span>(TTDdsNormalizedFiltered, <span class="fu">data.frame</span>(TTResNoNa[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">10</span>)]), <span class="at">by=</span><span class="st">"index"</span>)</span>
<span id="cb225-14"><a href="#cb225-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TTDdsNormalizedFiltered) <span class="ot">&lt;-</span> TTDdsNormalizedFiltered<span class="sc">$</span>index</span>
<span id="cb225-15"><a href="#cb225-15" aria-hidden="true" tabindex="-1"></a>TTDdsNormalizedFiltered <span class="ot">&lt;-</span> TTDdsNormalizedFiltered[,<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(TTDdsNormalizedFiltered)]</span>
<span id="cb225-16"><a href="#cb225-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb225-17"><a href="#cb225-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb225-18"><a href="#cb225-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TTDdsNormalizedFiltered, <span class="fu">aes</span>(<span class="at">x=</span> WT_Mean, <span class="at">y=</span> LOW_Mean, <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb225-19"><a href="#cb225-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb225-20"><a href="#cb225-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb225-21"><a href="#cb225-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb225-22"><a href="#cb225-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"Scatter plot of STAR TT-seq"</span>,</span>
<span id="cb225-23"><a href="#cb225-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb225-24"><a href="#cb225-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in WT"</span>), </span>
<span id="cb225-25"><a href="#cb225-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in LOW"</span>)) <span class="sc">+</span> </span>
<span id="cb225-26"><a href="#cb225-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb225-27"><a href="#cb225-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"blue"</span>),</span>
<span id="cb225-28"><a href="#cb225-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb225-29"><a href="#cb225-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbar(data = RnaDdsNormalizedFiltered[1:50,] ,aes(ymin = LOW_Mean - LOW_std, ymax = LOW_Mean + LOW_std), width = 0.1, alpha = 0.2) +</span></span>
<span id="cb225-30"><a href="#cb225-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbarh(aes(xmin = WT_Mean - WT_std, xmax = WT_Mean + WT_std), height = 0.1, alpha = 0.2) +</span></span>
<span id="cb225-31"><a href="#cb225-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-51-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="correlation-matrix-ssample-3" class="level4">
<h4 class="anchored" data-anchor-id="correlation-matrix-ssample-3">Correlation Matrix SSample</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb226"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb226-1"><a href="#cb226-1" aria-hidden="true" tabindex="-1"></a>TTannotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Genotype =</span> ttseq_description<span class="sc">$</span>Genotype)</span>
<span id="cb226-2"><a href="#cb226-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TTannotation) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ttseq_description)</span>
<span id="cb226-3"><a href="#cb226-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb226-4"><a href="#cb226-4" aria-hidden="true" tabindex="-1"></a>TTspearman_correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(TTDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TTDdsNormalized))], <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb226-5"><a href="#cb226-5" aria-hidden="true" tabindex="-1"></a>TTdist_from_spearman <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>TTspearman_correlation_matrix)</span>
<span id="cb226-6"><a href="#cb226-6" aria-hidden="true" tabindex="-1"></a>TTdist_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(TTdist_from_spearman)</span>
<span id="cb226-7"><a href="#cb226-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb226-8"><a href="#cb226-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(TTdist_matrix, </span>
<span id="cb226-9"><a href="#cb226-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> TTdist_from_spearman,</span>
<span id="cb226-10"><a href="#cb226-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> TTdist_from_spearman,</span>
<span id="cb226-11"><a href="#cb226-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> TTannotation,</span>
<span id="cb226-12"><a href="#cb226-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> TTannotation,</span>
<span id="cb226-13"><a href="#cb226-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">plasma</span>(<span class="dv">256</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-up-degs-bar-plot-3" class="level4">
<h4 class="anchored" data-anchor-id="top-10-up-degs-bar-plot-3">Top 10 Up Degs Bar Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb227"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb227-1"><a href="#cb227-1" aria-hidden="true" tabindex="-1"></a>TTDdsNormalized<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTDdsNormalized)</span>
<span id="cb227-2"><a href="#cb227-2" aria-hidden="true" tabindex="-1"></a>TTDdsNormalized.long <span class="ot">&lt;-</span> </span>
<span id="cb227-3"><a href="#cb227-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(TTDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TTDdsNormalized)) <span class="sc">|</span> <span class="fu">colnames</span>(TTDdsNormalized) <span class="sc">==</span> <span class="st">"gene_name"</span>],</span>
<span id="cb227-4"><a href="#cb227-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">id.vars =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb227-5"><a href="#cb227-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">variable_name =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb227-6"><a href="#cb227-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TTDdsNormalized.long)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"sample_name"</span></span>
<span id="cb227-7"><a href="#cb227-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TTDdsNormalized.long)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"normalized_count"</span></span>
<span id="cb227-8"><a href="#cb227-8" aria-hidden="true" tabindex="-1"></a>TTDdsNormalized.long <span class="ot">&lt;-</span> <span class="fu">merge</span>(TTDdsNormalized.long, ttseq_description, <span class="at">by=</span><span class="st">"sample_name"</span>)</span>
<span id="cb227-9"><a href="#cb227-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TTDdsNormalized.long)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     sample_name             gene_name normalized_count
1 RNA_S11589Nr69  ENSMUSG00000000001.4      4419.691073
2 RNA_S11589Nr69 ENSMUSG00000000003.15         0.000000
3 RNA_S11589Nr69 ENSMUSG00000000028.15      3379.993681
4 RNA_S11589Nr69 ENSMUSG00000000031.16        23.451821
5 RNA_S11589Nr69 ENSMUSG00000000037.17       725.052128
6 RNA_S11589Nr69 ENSMUSG00000000049.11         2.931478
                       sample                                     fastq_1
1 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
2 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
3 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
4 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
5 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
6 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
                                      fastq_2 strandedness Genotype Colonies
1 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
2 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
3 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
4 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
5 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
6 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
  Method Replicate Treatment
1  TTseq      Rep1      &lt;NA&gt;
2  TTseq      Rep1      &lt;NA&gt;
3  TTseq      Rep1      &lt;NA&gt;
4  TTseq      Rep1      &lt;NA&gt;
5  TTseq      Rep1      &lt;NA&gt;
6  TTseq      Rep1      &lt;NA&gt;</code></pre>
</div>
</div>
</section>
<section id="top-10-down-degs-bar-plot-3" class="level4">
<h4 class="anchored" data-anchor-id="top-10-down-degs-bar-plot-3">Top 10 Down Degs Bar Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb229"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb229-1"><a href="#cb229-1" aria-hidden="true" tabindex="-1"></a>top10_upregTT <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTT[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb229-2"><a href="#cb229-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb229-3"><a href="#cb229-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_upregTT) {</span>
<span id="cb229-4"><a href="#cb229-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTDdsNormalized.long[TTDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb229-5"><a href="#cb229-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb229-6"><a href="#cb229-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb229-7"><a href="#cb229-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb229-8"><a href="#cb229-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb229-9"><a href="#cb229-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb229-10"><a href="#cb229-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb229-11"><a href="#cb229-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb229-12"><a href="#cb229-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb229-13"><a href="#cb229-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-54-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a>top10_downregTT <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTT[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb230-2"><a href="#cb230-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb230-3"><a href="#cb230-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_downregTT) {</span>
<span id="cb230-4"><a href="#cb230-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTDdsNormalized.long[TTDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb230-5"><a href="#cb230-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb230-6"><a href="#cb230-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb230-7"><a href="#cb230-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb230-8"><a href="#cb230-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb230-9"><a href="#cb230-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb230-10"><a href="#cb230-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb230-11"><a href="#cb230-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb230-12"><a href="#cb230-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb230-13"><a href="#cb230-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-55-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="bowtie2" class="level3">
<h3 class="anchored" data-anchor-id="bowtie2">Bowtie2</h3>
<section id="volcano-plot-6" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot-6">Volcano Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb231"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb231-1"><a href="#cb231-1" aria-hidden="true" tabindex="-1"></a>TTBowtieResVolcPlot <span class="ot">&lt;-</span> TTResBowtieNoNa[<span class="fu">order</span>(TTResBowtieNoNa<span class="sc">$</span>padj),]</span>
<span id="cb231-2"><a href="#cb231-2" aria-hidden="true" tabindex="-1"></a>TTBowtieResVolcPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb231-3"><a href="#cb231-3" aria-hidden="true" tabindex="-1"></a>TTBowtieResVolcPlot<span class="sc">$</span>diffexpressed[TTBowtieResVolcPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> TTBowtieResVolcPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb231-4"><a href="#cb231-4" aria-hidden="true" tabindex="-1"></a>TTBowtieResVolcPlot<span class="sc">$</span>diffexpressed[TTBowtieResVolcPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> TTBowtieResVolcPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb231-5"><a href="#cb231-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb231-6"><a href="#cb231-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> TTBowtieResVolcPlot, </span>
<span id="cb231-7"><a href="#cb231-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb231-8"><a href="#cb231-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb231-9"><a href="#cb231-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb231-10"><a href="#cb231-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of Bowtie2 TT-seq"</span>,</span>
<span id="cb231-11"><a href="#cb231-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb231-12"><a href="#cb231-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb231-13"><a href="#cb231-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb231-14"><a href="#cb231-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb231-15"><a href="#cb231-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb231-16"><a href="#cb231-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb231-17"><a href="#cb231-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb231-18"><a href="#cb231-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb232"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb232-1"><a href="#cb232-1" aria-hidden="true" tabindex="-1"></a>top_upregTxiTT<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tx2gene<span class="sc">$</span>GENEID[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregTxiTT), tx2gene<span class="sc">$</span>TXNAME)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bar-plot-top-20-degs-6" class="level4">
<h4 class="anchored" data-anchor-id="bar-plot-top-20-degs-6">Bar Plot Top 20 Degs</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a>TTBowtieUpReg <span class="ot">&lt;-</span> TTBowtieResVolcPlot[TTBowtieResVolcPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb233-2"><a href="#cb233-2" aria-hidden="true" tabindex="-1"></a>TTBowtieDownReg <span class="ot">&lt;-</span> TTBowtieResVolcPlot[TTBowtieResVolcPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb233-3"><a href="#cb233-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-4"><a href="#cb233-4" aria-hidden="true" tabindex="-1"></a>TTBowtieUpReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TTBowtieUpReg<span class="sc">$</span>padj)</span>
<span id="cb233-5"><a href="#cb233-5" aria-hidden="true" tabindex="-1"></a>TTBowtieDownReg<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TTBowtieDownReg<span class="sc">$</span>padj)</span>
<span id="cb233-6"><a href="#cb233-6" aria-hidden="true" tabindex="-1"></a>TTBowtieUpRegSorted <span class="ot">&lt;-</span> TTBowtieUpReg[<span class="fu">order</span>(<span class="sc">-</span>TTBowtieUpReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TTBowtieUpReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb233-7"><a href="#cb233-7" aria-hidden="true" tabindex="-1"></a>TTBowtieDownRegSorted <span class="ot">&lt;-</span> TTBowtieDownReg[<span class="fu">order</span>(TTBowtieDownReg<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TTBowtieDownReg<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb233-8"><a href="#cb233-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-9"><a href="#cb233-9" aria-hidden="true" tabindex="-1"></a>top_upregTTBowtie <span class="ot">&lt;-</span> <span class="fu">head</span>(TTBowtieUpRegSorted, <span class="dv">20</span>)</span>
<span id="cb233-10"><a href="#cb233-10" aria-hidden="true" tabindex="-1"></a>top_downregTTBowtie <span class="ot">&lt;-</span> <span class="fu">head</span>(TTBowtieDownRegSorted, <span class="dv">20</span>)</span>
<span id="cb233-11"><a href="#cb233-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-12"><a href="#cb233-12" aria-hidden="true" tabindex="-1"></a>top_upregTTBowtie<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregTTBowtie), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb233-13"><a href="#cb233-13" aria-hidden="true" tabindex="-1"></a>top_upregTTBowtie<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregTTBowtie<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregTTBowtie<span class="sc">$</span>gene))</span>
<span id="cb233-14"><a href="#cb233-14" aria-hidden="true" tabindex="-1"></a>top_downregTTBowtie<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregTTBowtie), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb233-15"><a href="#cb233-15" aria-hidden="true" tabindex="-1"></a>top_downregTTBowtie<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregTTBowtie<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregTTBowtie<span class="sc">$</span>gene))</span>
<span id="cb233-16"><a href="#cb233-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-17"><a href="#cb233-17" aria-hidden="true" tabindex="-1"></a>TTBowtieUpDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTTBowtie)</span>
<span id="cb233-18"><a href="#cb233-18" aria-hidden="true" tabindex="-1"></a>TTBowtieDownDegsList <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTTBowtie)</span>
<span id="cb233-19"><a href="#cb233-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-20"><a href="#cb233-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregTTBowtie, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb233-21"><a href="#cb233-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb233-22"><a href="#cb233-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb233-23"><a href="#cb233-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [Bowtie2 TT-seq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb233-24"><a href="#cb233-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregTT, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb234-2"><a href="#cb234-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>   </span>
<span id="cb234-3"><a href="#cb234-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb234-4"><a href="#cb234-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [Bowtie2 TT-seq]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb234-5"><a href="#cb234-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-57-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>####Scatter Plot Mean Expression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb235"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb235-1"><a href="#cb235-1" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalized <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">counts</span>(ttseq_dds, <span class="at">normalize=</span><span class="cn">TRUE</span>))</span>
<span id="cb235-2"><a href="#cb235-2" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalized<span class="sc">$</span>WT_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTBowtieDdsNormalized[,<span class="fu">rownames</span>(ttseq_description[ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb235-3"><a href="#cb235-3" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalized<span class="sc">$</span>LOW_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTBowtieDdsNormalized[,<span class="fu">rownames</span>(ttseq_description[ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb235-4"><a href="#cb235-4" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalized<span class="sc">$</span>WT_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTBowtieDdsNormalized[,<span class="fu">rownames</span>(ttseq_description[ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb235-5"><a href="#cb235-5" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalized<span class="sc">$</span>LOW_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTBowtieDdsNormalized[,<span class="fu">rownames</span>(ttseq_description[ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb235-6"><a href="#cb235-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-7"><a href="#cb235-7" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalizedFiltered <span class="ot">&lt;-</span> TTBowtieDdsNormalized <span class="sc">%&gt;%</span> <span class="fu">filter</span>(WT_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb235-8"><a href="#cb235-8" aria-hidden="true" tabindex="-1"></a>                                                        (WT_Mean <span class="sc">-</span> WT_std <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb235-9"><a href="#cb235-9" aria-hidden="true" tabindex="-1"></a>                                                        LOW_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb235-10"><a href="#cb235-10" aria-hidden="true" tabindex="-1"></a>                                                        (LOW_Mean <span class="sc">-</span> LOW_std <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb235-11"><a href="#cb235-11" aria-hidden="true" tabindex="-1"></a>TTResBowtieNoNa<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTResBowtieNoNa)</span>
<span id="cb235-12"><a href="#cb235-12" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalizedFiltered<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTBowtieDdsNormalizedFiltered)</span>
<span id="cb235-13"><a href="#cb235-13" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalizedFiltered <span class="ot">&lt;-</span> <span class="fu">merge</span>(TTBowtieDdsNormalizedFiltered, <span class="fu">data.frame</span>(TTResBowtieNoNa[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">10</span>)]), <span class="at">by=</span><span class="st">"index"</span>)</span>
<span id="cb235-14"><a href="#cb235-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TTBowtieDdsNormalizedFiltered) <span class="ot">&lt;-</span> TTBowtieDdsNormalizedFiltered<span class="sc">$</span>index</span>
<span id="cb235-15"><a href="#cb235-15" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalizedFiltered <span class="ot">&lt;-</span> TTBowtieDdsNormalizedFiltered[,<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(TTBowtieDdsNormalizedFiltered)]</span>
<span id="cb235-16"><a href="#cb235-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-17"><a href="#cb235-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-18"><a href="#cb235-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TTBowtieDdsNormalizedFiltered, <span class="fu">aes</span>(<span class="at">x=</span> WT_Mean, <span class="at">y=</span> LOW_Mean, <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb235-19"><a href="#cb235-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb235-20"><a href="#cb235-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb235-21"><a href="#cb235-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb235-22"><a href="#cb235-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"Scatter plot of STAR TT-seq"</span>,</span>
<span id="cb235-23"><a href="#cb235-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb235-24"><a href="#cb235-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in WT"</span>), </span>
<span id="cb235-25"><a href="#cb235-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in LOW"</span>)) <span class="sc">+</span> </span>
<span id="cb235-26"><a href="#cb235-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb235-27"><a href="#cb235-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"blue"</span>),</span>
<span id="cb235-28"><a href="#cb235-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb235-29"><a href="#cb235-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbar(data = RnaDdsNormalizedFiltered[1:50,] ,aes(ymin = LOW_Mean - LOW_std, ymax = LOW_Mean + LOW_std), width = 0.1, alpha = 0.2) +</span></span>
<span id="cb235-30"><a href="#cb235-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbarh(aes(xmin = WT_Mean - WT_std, xmax = WT_Mean + WT_std), height = 0.1, alpha = 0.2) +</span></span>
<span id="cb235-31"><a href="#cb235-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="correlation-matrix-ssample-4" class="level4">
<h4 class="anchored" data-anchor-id="correlation-matrix-ssample-4">Correlation Matrix SSample</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a>TTBowtieannotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Genotype =</span> ttseq_description<span class="sc">$</span>Genotype)</span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TTBowtieannotation) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ttseq_description)</span>
<span id="cb236-3"><a href="#cb236-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-4"><a href="#cb236-4" aria-hidden="true" tabindex="-1"></a>TTBowtieSpearman_correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(TTBowtieDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TTBowtieDdsNormalized))], <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb236-5"><a href="#cb236-5" aria-hidden="true" tabindex="-1"></a>TTBowtiedist_from_spearman <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>TTBowtieSpearman_correlation_matrix)</span>
<span id="cb236-6"><a href="#cb236-6" aria-hidden="true" tabindex="-1"></a>TTBowtiedist_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(TTBowtiedist_from_spearman)</span>
<span id="cb236-7"><a href="#cb236-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-8"><a href="#cb236-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(TTBowtiedist_matrix, </span>
<span id="cb236-9"><a href="#cb236-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> TTBowtiedist_from_spearman,</span>
<span id="cb236-10"><a href="#cb236-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> TTBowtiedist_from_spearman,</span>
<span id="cb236-11"><a href="#cb236-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> TTBowtieannotation,</span>
<span id="cb236-12"><a href="#cb236-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> TTBowtieannotation,</span>
<span id="cb236-13"><a href="#cb236-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">plasma</span>(<span class="dv">256</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-59-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-up-degs-bar-plot-4" class="level4">
<h4 class="anchored" data-anchor-id="top-10-up-degs-bar-plot-4">Top 10 Up Degs Bar Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb237"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb237-1"><a href="#cb237-1" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalized<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTBowtieDdsNormalized)</span>
<span id="cb237-2"><a href="#cb237-2" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalized.long <span class="ot">&lt;-</span> </span>
<span id="cb237-3"><a href="#cb237-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(TTDdsNormalized[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TTBowtieDdsNormalized)) <span class="sc">|</span> <span class="fu">colnames</span>(TTBowtieDdsNormalized) <span class="sc">==</span> <span class="st">"gene_name"</span>],</span>
<span id="cb237-4"><a href="#cb237-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">id.vars =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb237-5"><a href="#cb237-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">variable_name =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb237-6"><a href="#cb237-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TTBowtieDdsNormalized.long)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"sample_name"</span></span>
<span id="cb237-7"><a href="#cb237-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TTBowtieDdsNormalized.long)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"normalized_count"</span></span>
<span id="cb237-8"><a href="#cb237-8" aria-hidden="true" tabindex="-1"></a>TTBowtieDdsNormalized.long <span class="ot">&lt;-</span> <span class="fu">merge</span>(TTBowtieDdsNormalized.long, ttseq_description, <span class="at">by=</span><span class="st">"sample_name"</span>)</span>
<span id="cb237-9"><a href="#cb237-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TTBowtieDdsNormalized.long)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     sample_name             gene_name normalized_count
1 RNA_S11589Nr69  ENSMUSG00000000001.4      4419.691073
2 RNA_S11589Nr69 ENSMUSG00000000003.15         0.000000
3 RNA_S11589Nr69 ENSMUSG00000000028.15      3379.993681
4 RNA_S11589Nr69 ENSMUSG00000000031.16        23.451821
5 RNA_S11589Nr69 ENSMUSG00000000037.17       725.052128
6 RNA_S11589Nr69 ENSMUSG00000000049.11         2.931478
                       sample                                     fastq_1
1 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
2 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
3 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
4 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
5 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
6 H1_WT_C1_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr69.1.fastq.gz
                                      fastq_2 strandedness Genotype Colonies
1 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
2 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
3 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
4 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
5 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
6 S11589_2/01_fastq/RNA_S11589Nr69.2.fastq.gz         auto       WT       C1
  Method Replicate Treatment
1  TTseq      Rep1      &lt;NA&gt;
2  TTseq      Rep1      &lt;NA&gt;
3  TTseq      Rep1      &lt;NA&gt;
4  TTseq      Rep1      &lt;NA&gt;
5  TTseq      Rep1      &lt;NA&gt;
6  TTseq      Rep1      &lt;NA&gt;</code></pre>
</div>
</div>
</section>
<section id="top-10-up-degs-bar-plot-5" class="level4">
<h4 class="anchored" data-anchor-id="top-10-up-degs-bar-plot-5">Top 10 Up Degs Bar Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a>top10_upregTTBowtie <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTTBowtie[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb239-2"><a href="#cb239-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-3"><a href="#cb239-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_upregTTBowtie) {</span>
<span id="cb239-4"><a href="#cb239-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTBowtieDdsNormalized.long[TTBowtieDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb239-5"><a href="#cb239-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb239-6"><a href="#cb239-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb239-7"><a href="#cb239-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb239-8"><a href="#cb239-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb239-9"><a href="#cb239-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb239-10"><a href="#cb239-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb239-11"><a href="#cb239-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb239-12"><a href="#cb239-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb239-13"><a href="#cb239-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-61-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-down-degs-bar-plot-4" class="level4">
<h4 class="anchored" data-anchor-id="top-10-down-degs-bar-plot-4">Top 10 Down Degs Bar Plot</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a>top10_downregTTBowtie <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_downregTTBowtie[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_downregTTBowtie) {</span>
<span id="cb240-4"><a href="#cb240-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTBowtieDdsNormalized.long[TTBowtieDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb240-5"><a href="#cb240-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb240-6"><a href="#cb240-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb240-7"><a href="#cb240-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb240-8"><a href="#cb240-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb240-9"><a href="#cb240-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb240-10"><a href="#cb240-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb240-11"><a href="#cb240-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb240-12"><a href="#cb240-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb240-13"><a href="#cb240-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-62-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTBowtieDdsNormalized.long[TTBowtieDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> top10_downregTTBowtie[<span class="dv">1</span>], ],</span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb241-3"><a href="#cb241-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb241-4"><a href="#cb241-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb241-5"><a href="#cb241-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Bowtie2 TT-seq of"</span>, MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> top10_downregTTBowtie[<span class="dv">1</span>]]),</span>
<span id="cb241-6"><a href="#cb241-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb241-7"><a href="#cb241-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb241-8"><a href="#cb241-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb241-9"><a href="#cb241-9" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTDdsNormalized.long[TTDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> top10_downregTTBowtie[<span class="dv">1</span>], ],</span>
<span id="cb241-10"><a href="#cb241-10" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb241-11"><a href="#cb241-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb241-12"><a href="#cb241-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb241-13"><a href="#cb241-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"STAR TT-seq of"</span>, MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> top10_downregTTBowtie[<span class="dv">1</span>]]),</span>
<span id="cb241-14"><a href="#cb241-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb241-15"><a href="#cb241-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb241-16"><a href="#cb241-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb241-17"><a href="#cb241-17" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(RnaDdsNormalized.long[RnaDdsNormalized.long<span class="sc">$</span>gene_name <span class="sc">==</span> top10_downregTTBowtie[<span class="dv">1</span>], ],</span>
<span id="cb241-18"><a href="#cb241-18" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb241-19"><a href="#cb241-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb241-20"><a href="#cb241-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb241-21"><a href="#cb241-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"RNA-seq of"</span>, MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> top10_downregTTBowtie[<span class="dv">1</span>]]),</span>
<span id="cb241-22"><a href="#cb241-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb241-23"><a href="#cb241-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb241-24"><a href="#cb241-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb241-25"><a href="#cb241-25" aria-hidden="true" tabindex="-1"></a>p<span class="sc">+</span>p1<span class="sc">+</span>p2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-63-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="c1-removed-bowtie2" class="level3">
<h3 class="anchored" data-anchor-id="c1-removed-bowtie2">C1-Removed-Bowtie2</h3>
<section id="deseq-genotype" class="level4">
<h4 class="anchored" data-anchor-id="deseq-genotype">DESeq-Genotype</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a>ttseq_description_C1_removed <span class="ot">&lt;-</span> ttseq_description[<span class="sc">!</span>(ttseq_description<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span> <span class="sc">&amp;</span> ttseq_description<span class="sc">$</span>Colonies<span class="sc">==</span><span class="st">"C1"</span>),]</span>
<span id="cb242-2"><a href="#cb242-2" aria-hidden="true" tabindex="-1"></a>bowtie_ttseq_C1_removed <span class="ot">&lt;-</span> bowtie_ttseq[,<span class="fu">rownames</span>(ttseq_description_C1_removed)]</span>
<span id="cb242-3"><a href="#cb242-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb242-4"><a href="#cb242-4" aria-hidden="true" tabindex="-1"></a>ttseq_dds_c1_removed.G <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> bowtie_ttseq_C1_removed,</span>
<span id="cb242-5"><a href="#cb242-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> ttseq_description_C1_removed,</span>
<span id="cb242-6"><a href="#cb242-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span> Genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="sourceCode cell-code" id="cb244"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1" aria-hidden="true" tabindex="-1"></a>ttseq_dds_c1_removed.G <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(ttseq_dds_c1_removed.G, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"Droso_"</span>, <span class="fu">rownames</span>(bowtie_ttseq_C1_removed)))</span>
<span id="cb244-2"><a href="#cb244-2" aria-hidden="true" tabindex="-1"></a>ttseq_dds_c1_removed.G <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ttseq_dds_c1_removed.G[<span class="fu">grepl</span>(<span class="st">"^ENS"</span>, <span class="fu">rownames</span>(bowtie_ttseq_C1_removed))])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a>TtResC1Removed.G <span class="ot">&lt;-</span> <span class="fu">results</span>(ttseq_dds_c1_removed.G, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb251-2"><a href="#cb251-2" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.G <span class="ot">&lt;-</span> TtResC1Removed.G[<span class="sc">!</span><span class="fu">is.na</span>(TtResC1Removed.G<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(TtResC1Removed.G<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb251-3"><a href="#cb251-3" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.G<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb251-4"><a href="#cb251-4" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.G<span class="sc">$</span>significant[TtResNoNaC1Removed.G<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="ma-plot-2" class="level5">
<h5 class="anchored" data-anchor-id="ma-plot-2">MA Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb252"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb252-1"><a href="#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(TtResNoNaC1Removed.G)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-65-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TtResNoNaC1Removed.G, <span class="fu">aes</span>(<span class="at">x =</span>TtResNoNaC1Removed.G<span class="sc">$</span>baseMean, </span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> TtResNoNaC1Removed.G<span class="sc">$</span>log2FoldChange, </span>
<span id="cb253-3"><a href="#cb253-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb253-4"><a href="#cb253-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb253-5"><a href="#cb253-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb253-6"><a href="#cb253-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from Bowtie2 TT-seq C1 Removed"</span>,</span>
<span id="cb253-7"><a href="#cb253-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb253-8"><a href="#cb253-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb253-9"><a href="#cb253-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb253-10"><a href="#cb253-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb253-11"><a href="#cb253-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb253-12"><a href="#cb253-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb253-13"><a href="#cb253-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb253-14"><a href="#cb253-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb253-15"><a href="#cb253-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-65-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="qq-plot-1" class="level5">
<h5 class="anchored" data-anchor-id="qq-plot-1">QQ Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb254"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb254-1"><a href="#cb254-1" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.G<span class="sc">$</span>PvalueSorted <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">sort</span>(TtResNoNaC1Removed.G<span class="sc">$</span>pvalue))</span>
<span id="cb254-2"><a href="#cb254-2" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.G<span class="sc">$</span>PvalueExpected <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">ppoints</span>(<span class="fu">length</span>(TtResNoNaC1Removed.G<span class="sc">$</span>PvalueSorted)))</span>
<span id="cb254-3"><a href="#cb254-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-4"><a href="#cb254-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-5"><a href="#cb254-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TtResNoNaC1Removed.G, <span class="fu">aes</span>(PvalueSorted, PvalueExpected)) <span class="sc">+</span></span>
<span id="cb254-6"><a href="#cb254-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb254-7"><a href="#cb254-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb254-8"><a href="#cb254-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"QQ Plot of Bowtie2 TT-seq C1 Removed"</span>,</span>
<span id="cb254-9"><a href="#cb254-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Observed -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>), </span>
<span id="cb254-10"><a href="#cb254-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Expected -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>)) <span class="sc">+</span> </span>
<span id="cb254-11"><a href="#cb254-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb254-12"><a href="#cb254-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb254-13"><a href="#cb254-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-66-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="volcano-plot-7" class="level5">
<h5 class="anchored" data-anchor-id="volcano-plot-7">Volcano Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a>TTseqC1Removed.G_VocPlot <span class="ot">&lt;-</span> TtResNoNaC1Removed.G[<span class="fu">order</span>(TtResNoNaC1Removed.G<span class="sc">$</span>padj),]</span>
<span id="cb255-2"><a href="#cb255-2" aria-hidden="true" tabindex="-1"></a>TTseqC1Removed.G_VocPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb255-3"><a href="#cb255-3" aria-hidden="true" tabindex="-1"></a>TTseqC1Removed.G_VocPlot<span class="sc">$</span>diffexpressed[TTseqC1Removed.G_VocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> TTseqC1Removed.G_VocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb255-4"><a href="#cb255-4" aria-hidden="true" tabindex="-1"></a>TTseqC1Removed.G_VocPlot<span class="sc">$</span>diffexpressed[TTseqC1Removed.G_VocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> TTseqC1Removed.G_VocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb255-5"><a href="#cb255-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-6"><a href="#cb255-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> TTseqC1Removed.G_VocPlot, </span>
<span id="cb255-7"><a href="#cb255-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb255-8"><a href="#cb255-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb255-9"><a href="#cb255-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb255-10"><a href="#cb255-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of Bowtie2 TT-seq C1 Removed"</span>,</span>
<span id="cb255-11"><a href="#cb255-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb255-12"><a href="#cb255-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb255-13"><a href="#cb255-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb255-14"><a href="#cb255-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb255-15"><a href="#cb255-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb255-16"><a href="#cb255-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb255-17"><a href="#cb255-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb255-18"><a href="#cb255-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-67-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bar-plot-top-20-degs-7" class="level5">
<h5 class="anchored" data-anchor-id="bar-plot-top-20-degs-7">Bar Plot Top 20 Degs</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb256"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb256-1"><a href="#cb256-1" aria-hidden="true" tabindex="-1"></a>TTC1RemovedUpReg.G <span class="ot">&lt;-</span> TTseqC1Removed.G_VocPlot[TTseqC1Removed.G_VocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb256-2"><a href="#cb256-2" aria-hidden="true" tabindex="-1"></a>TTC1RemovedDownReg.G <span class="ot">&lt;-</span> TTseqC1Removed.G_VocPlot[TTseqC1Removed.G_VocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb256-3"><a href="#cb256-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb256-4"><a href="#cb256-4" aria-hidden="true" tabindex="-1"></a>TTC1RemovedUpReg.G<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TTC1RemovedUpReg.G<span class="sc">$</span>padj)</span>
<span id="cb256-5"><a href="#cb256-5" aria-hidden="true" tabindex="-1"></a>TTC1RemovedDownReg.G<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TTC1RemovedDownReg.G<span class="sc">$</span>padj)</span>
<span id="cb256-6"><a href="#cb256-6" aria-hidden="true" tabindex="-1"></a>TTC1RemovedUpRegSorted.G <span class="ot">&lt;-</span> TTC1RemovedUpReg.G[<span class="fu">order</span>(<span class="sc">-</span>TTC1RemovedUpReg.G<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TTC1RemovedUpReg.G<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb256-7"><a href="#cb256-7" aria-hidden="true" tabindex="-1"></a>TTC1RemovedDownRegSorted.G <span class="ot">&lt;-</span> TTC1RemovedDownReg.G[<span class="fu">order</span>(TTC1RemovedDownReg.G<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TTC1RemovedDownReg.G<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb256-8"><a href="#cb256-8" aria-hidden="true" tabindex="-1"></a>top_upregTTC1Removed.G <span class="ot">&lt;-</span> <span class="fu">head</span>(TTC1RemovedUpRegSorted.G, <span class="dv">50</span>)</span>
<span id="cb256-9"><a href="#cb256-9" aria-hidden="true" tabindex="-1"></a>top_downregTTC1Removed.G <span class="ot">&lt;-</span> <span class="fu">head</span>(TTC1RemovedDownRegSorted.G, <span class="dv">50</span>)</span>
<span id="cb256-10"><a href="#cb256-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb256-11"><a href="#cb256-11" aria-hidden="true" tabindex="-1"></a>top_upregTTC1Removed.G<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregTTC1Removed.G), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb256-12"><a href="#cb256-12" aria-hidden="true" tabindex="-1"></a>top_upregTTC1Removed.G<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregTTC1Removed.G<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregTTC1Removed.G<span class="sc">$</span>gene))</span>
<span id="cb256-13"><a href="#cb256-13" aria-hidden="true" tabindex="-1"></a>top_downregTTC1Removed.G<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregTTC1Removed.G), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb256-14"><a href="#cb256-14" aria-hidden="true" tabindex="-1"></a>top_downregTTC1Removed.G<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregTTC1Removed.G<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregTTC1Removed.G<span class="sc">$</span>gene))</span>
<span id="cb256-15"><a href="#cb256-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb256-16"><a href="#cb256-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb256-17"><a href="#cb256-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregTTC1Removed.G, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb256-18"><a href="#cb256-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb256-19"><a href="#cb256-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb256-20"><a href="#cb256-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [Bowtie TTseq C1 Removed]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb256-21"><a href="#cb256-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-68-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb257"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb257-1"><a href="#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregTTC1Removed.G, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb257-2"><a href="#cb257-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>  </span>
<span id="cb257-3"><a href="#cb257-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb257-4"><a href="#cb257-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [Bowtie TTseq C1 Removed]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb257-5"><a href="#cb257-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-68-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="scatter-plot-mean-expression" class="level5">
<h5 class="anchored" data-anchor-id="scatter-plot-mean-expression">Scatter Plot Mean Expression</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb258"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb258-1"><a href="#cb258-1" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.G <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">counts</span>(ttseq_dds_c1_removed.G, <span class="at">normalize=</span><span class="cn">TRUE</span>))</span>
<span id="cb258-2"><a href="#cb258-2" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.G<span class="sc">$</span>WT_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTseqC1RemovedNormalized.G[,<span class="fu">rownames</span>(ttseq_description_C1_removed[ttseq_description_C1_removed<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb258-3"><a href="#cb258-3" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.G<span class="sc">$</span>LOW_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTseqC1RemovedNormalized.G[,<span class="fu">rownames</span>(ttseq_description_C1_removed[ttseq_description_C1_removed<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb258-4"><a href="#cb258-4" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.G<span class="sc">$</span>WT_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTseqC1RemovedNormalized.G[,<span class="fu">rownames</span>(ttseq_description_C1_removed[ttseq_description_C1_removed<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb258-5"><a href="#cb258-5" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.G<span class="sc">$</span>LOW_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTseqC1RemovedNormalized.G[,<span class="fu">rownames</span>(ttseq_description_C1_removed[ttseq_description_C1_removed<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb258-6"><a href="#cb258-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-7"><a href="#cb258-7" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedFiltered.G <span class="ot">&lt;-</span> TTseqC1RemovedNormalized.G <span class="sc">%&gt;%</span> <span class="fu">filter</span>(WT_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb258-8"><a href="#cb258-8" aria-hidden="true" tabindex="-1"></a>                                                        (WT_Mean <span class="sc">-</span> WT_std <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb258-9"><a href="#cb258-9" aria-hidden="true" tabindex="-1"></a>                                                        LOW_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb258-10"><a href="#cb258-10" aria-hidden="true" tabindex="-1"></a>                                                        (LOW_Mean <span class="sc">-</span> LOW_std <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb258-11"><a href="#cb258-11" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.G<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TtResNoNaC1Removed.G)</span>
<span id="cb258-12"><a href="#cb258-12" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedFiltered.G<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTseqC1RemovedNormalizedFiltered.G)</span>
<span id="cb258-13"><a href="#cb258-13" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedFiltered.G <span class="ot">&lt;-</span> <span class="fu">merge</span>(TTseqC1RemovedNormalizedFiltered.G, <span class="fu">data.frame</span>(TtResNoNaC1Removed.G[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">10</span>)]), <span class="at">by=</span><span class="st">"index"</span>)</span>
<span id="cb258-14"><a href="#cb258-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TTseqC1RemovedNormalizedFiltered.G) <span class="ot">&lt;-</span> TTseqC1RemovedNormalizedFiltered.G<span class="sc">$</span>index</span>
<span id="cb258-15"><a href="#cb258-15" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedFiltered.G <span class="ot">&lt;-</span> TTseqC1RemovedNormalizedFiltered.G[,<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(TTseqC1RemovedNormalizedFiltered.G)]</span>
<span id="cb258-16"><a href="#cb258-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-17"><a href="#cb258-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-18"><a href="#cb258-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TTseqC1RemovedNormalizedFiltered.G, <span class="fu">aes</span>(<span class="at">x=</span> WT_Mean, <span class="at">y=</span> LOW_Mean, <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb258-19"><a href="#cb258-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb258-20"><a href="#cb258-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb258-21"><a href="#cb258-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb258-22"><a href="#cb258-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"Scatter plot of Bowtie2 TT-seq C1 Removed"</span>,</span>
<span id="cb258-23"><a href="#cb258-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb258-24"><a href="#cb258-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in WT"</span>), </span>
<span id="cb258-25"><a href="#cb258-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in LOW"</span>)) <span class="sc">+</span> </span>
<span id="cb258-26"><a href="#cb258-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb258-27"><a href="#cb258-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"blue"</span>),</span>
<span id="cb258-28"><a href="#cb258-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb258-29"><a href="#cb258-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbar(data = TTseqC1RemovedNormalizedFiltered.G[1:50,] ,aes(ymin = LOW_Mean - LOW_std, ymax = LOW_Mean + LOW_std), width = 0.1, alpha = 0.2) +</span></span>
<span id="cb258-30"><a href="#cb258-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbarh(aes(xmin = WT_Mean - WT_std, xmax = WT_Mean + WT_std), height = 0.1, alpha = 0.2) +</span></span>
<span id="cb258-31"><a href="#cb258-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-69-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="correlation-matrix-ssample-5" class="level5">
<h5 class="anchored" data-anchor-id="correlation-matrix-ssample-5">Correlation Matrix SSample</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Genotype =</span> ttseq_description_C1_removed<span class="sc">$</span>Genotype)</span>
<span id="cb259-2"><a href="#cb259-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ttseq_description_C1_removed)</span>
<span id="cb259-3"><a href="#cb259-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-4"><a href="#cb259-4" aria-hidden="true" tabindex="-1"></a>spearman_correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(TTseqC1RemovedNormalized.G[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TTseqC1RemovedNormalized.G))], <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb259-5"><a href="#cb259-5" aria-hidden="true" tabindex="-1"></a>dist_from_spearman <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>spearman_correlation_matrix)</span>
<span id="cb259-6"><a href="#cb259-6" aria-hidden="true" tabindex="-1"></a>dist_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(dist_from_spearman)</span>
<span id="cb259-7"><a href="#cb259-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-8"><a href="#cb259-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(dist_matrix, </span>
<span id="cb259-9"><a href="#cb259-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> dist_from_spearman,</span>
<span id="cb259-10"><a href="#cb259-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> dist_from_spearman,</span>
<span id="cb259-11"><a href="#cb259-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> annotation,</span>
<span id="cb259-12"><a href="#cb259-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> annotation,</span>
<span id="cb259-13"><a href="#cb259-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">plasma</span>(<span class="dv">256</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-70-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb260"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.G<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTseqC1RemovedNormalized.G)</span>
<span id="cb260-2"><a href="#cb260-2" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedLong.G <span class="ot">&lt;-</span> </span>
<span id="cb260-3"><a href="#cb260-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(TTseqC1RemovedNormalized.G[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TTseqC1RemovedNormalized.G)) <span class="sc">|</span> <span class="fu">colnames</span>(TTseqC1RemovedNormalized.G) <span class="sc">==</span> <span class="st">"gene_name"</span>],</span>
<span id="cb260-4"><a href="#cb260-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">id.vars =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb260-5"><a href="#cb260-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">variable_name =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb260-6"><a href="#cb260-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TTseqC1RemovedNormalizedLong.G)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"sample_name"</span></span>
<span id="cb260-7"><a href="#cb260-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TTseqC1RemovedNormalizedLong.G)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"normalized_count"</span></span>
<span id="cb260-8"><a href="#cb260-8" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedLong.G <span class="ot">&lt;-</span> <span class="fu">merge</span>(TTseqC1RemovedNormalizedLong.G, ttseq_description_C1_removed, <span class="at">by=</span><span class="st">"sample_name"</span>)</span>
<span id="cb260-9"><a href="#cb260-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TTseqC1RemovedNormalizedLong.G)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     sample_name             gene_name normalized_count
1 RNA_S11589Nr70  ENSMUSG00000000001.4      6774.331066
2 RNA_S11589Nr70 ENSMUSG00000000003.15         0.000000
3 RNA_S11589Nr70 ENSMUSG00000000028.15      4729.202526
4 RNA_S11589Nr70 ENSMUSG00000000031.16        69.177165
5 RNA_S11589Nr70 ENSMUSG00000000037.17      1734.460182
6 RNA_S11589Nr70 ENSMUSG00000000049.11         2.515533
                       sample                                     fastq_1
1 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
2 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
3 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
4 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
5 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
6 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
                                      fastq_2 strandedness Genotype Colonies
1 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
2 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
3 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
4 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
5 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
6 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
  Method Replicate Treatment
1  TTseq      Rep1      &lt;NA&gt;
2  TTseq      Rep1      &lt;NA&gt;
3  TTseq      Rep1      &lt;NA&gt;
4  TTseq      Rep1      &lt;NA&gt;
5  TTseq      Rep1      &lt;NA&gt;
6  TTseq      Rep1      &lt;NA&gt;</code></pre>
</div>
</div>
</section>
<section id="top-10-up-degs-bar-plot-6" class="level5">
<h5 class="anchored" data-anchor-id="top-10-up-degs-bar-plot-6">Top 10 Up Degs Bar Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb262"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb262-1"><a href="#cb262-1" aria-hidden="true" tabindex="-1"></a>top10_upregTTseqC1Removed.G <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTTC1Removed.G[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb262-2"><a href="#cb262-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_upregTTseqC1Removed.G) {</span>
<span id="cb262-3"><a href="#cb262-3" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTseqC1RemovedNormalizedLong.G[TTseqC1RemovedNormalizedLong.G<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb262-4"><a href="#cb262-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb262-5"><a href="#cb262-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb262-6"><a href="#cb262-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb262-7"><a href="#cb262-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb262-8"><a href="#cb262-8" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb262-9"><a href="#cb262-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb262-10"><a href="#cb262-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb262-11"><a href="#cb262-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb262-12"><a href="#cb262-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb262-13"><a href="#cb262-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-72-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-down-degs-bar-plot-5" class="level5">
<h5 class="anchored" data-anchor-id="top-10-down-degs-bar-plot-5">Top 10 Down Degs Bar Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a>top10_downregTTseqC1Removed.G <span class="ot">&lt;-</span> <span class="fu">rownames</span>(<span class="fu">head</span>(top_downregTTC1Removed.G, <span class="dv">10</span>))</span>
<span id="cb263-2"><a href="#cb263-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb263-3"><a href="#cb263-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_downregTTseqC1Removed.G) {</span>
<span id="cb263-4"><a href="#cb263-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTseqC1RemovedNormalizedLong.G[TTseqC1RemovedNormalizedLong.G<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb263-5"><a href="#cb263-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb263-6"><a href="#cb263-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb263-7"><a href="#cb263-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Colonies), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb263-8"><a href="#cb263-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb263-9"><a href="#cb263-9" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb263-10"><a href="#cb263-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb263-11"><a href="#cb263-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb263-12"><a href="#cb263-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb263-13"><a href="#cb263-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb263-14"><a href="#cb263-14" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-73-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="deseq-genotype-replicate" class="level4">
<h4 class="anchored" data-anchor-id="deseq-genotype-replicate">DESeq-Genotype-Replicate</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb264"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb264-1"><a href="#cb264-1" aria-hidden="true" tabindex="-1"></a>ttseq_dds_c1_removed.GR <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> bowtie_ttseq_C1_removed,</span>
<span id="cb264-2"><a href="#cb264-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> ttseq_description_C1_removed,</span>
<span id="cb264-3"><a href="#cb264-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span> Replicate <span class="sc">+</span> Genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="sourceCode cell-code" id="cb267"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb267-1"><a href="#cb267-1" aria-hidden="true" tabindex="-1"></a>ttseq_dds_c1_removed.GR <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(ttseq_dds_c1_removed.GR, <span class="at">controlGenes =</span> <span class="fu">grepl</span>(<span class="st">"Droso_"</span>, <span class="fu">rownames</span>(bowtie_ttseq_C1_removed)))</span>
<span id="cb267-2"><a href="#cb267-2" aria-hidden="true" tabindex="-1"></a>ttseq_dds_c1_removed.GR <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ttseq_dds_c1_removed.GR[<span class="fu">grepl</span>(<span class="st">"^ENS"</span>, <span class="fu">rownames</span>(bowtie_ttseq_C1_removed))])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using pre-existing size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb274"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb274-1"><a href="#cb274-1" aria-hidden="true" tabindex="-1"></a>TtResC1Removed.GR <span class="ot">&lt;-</span> <span class="fu">results</span>(ttseq_dds_c1_removed.GR, <span class="at">name=</span><span class="st">"Genotype_WT_vs_LOW"</span>, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">"Genotype"</span>,<span class="st">"LOW"</span>,<span class="st">"WT"</span>))</span>
<span id="cb274-2"><a href="#cb274-2" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.GR <span class="ot">&lt;-</span> TtResC1Removed.GR[<span class="sc">!</span><span class="fu">is.na</span>(TtResC1Removed.GR<span class="sc">$</span>padj) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(TtResC1Removed.GR<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb274-3"><a href="#cb274-3" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.GR<span class="sc">$</span>significant <span class="ot">&lt;-</span> <span class="st">"Not Significant"</span></span>
<span id="cb274-4"><a href="#cb274-4" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.GR<span class="sc">$</span>significant[TtResNoNaC1Removed.GR<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"Significant"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="ma-plot-3" class="level5">
<h5 class="anchored" data-anchor-id="ma-plot-3">MA Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb275"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb275-1"><a href="#cb275-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(TtResNoNaC1Removed.GR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-75-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb276"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb276-1"><a href="#cb276-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TtResNoNaC1Removed.GR, <span class="fu">aes</span>(<span class="at">x =</span>TtResNoNaC1Removed.GR<span class="sc">$</span>baseMean, </span>
<span id="cb276-2"><a href="#cb276-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">y =</span> TtResNoNaC1Removed.GR<span class="sc">$</span>log2FoldChange, </span>
<span id="cb276-3"><a href="#cb276-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb276-4"><a href="#cb276-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb276-5"><a href="#cb276-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb276-6"><a href="#cb276-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"MA Plot from Bowtie2 TT-seq C1 Removed - Genotype + Replicate"</span>,</span>
<span id="cb276-7"><a href="#cb276-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb276-8"><a href="#cb276-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"BaseMean Of Normalized Counts"</span>), </span>
<span id="cb276-9"><a href="#cb276-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">" Fold Change"</span>)) <span class="sc">+</span> </span>
<span id="cb276-10"><a href="#cb276-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb276-11"><a href="#cb276-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb276-12"><a href="#cb276-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb276-13"><a href="#cb276-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkgrey"</span>, <span class="st">"blue"</span>),</span>
<span id="cb276-14"><a href="#cb276-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-75-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb277"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb277-1"><a href="#cb277-1" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_density2d(col="red")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="qq-plot-2" class="level5">
<h5 class="anchored" data-anchor-id="qq-plot-2">QQ Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb278"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb278-1"><a href="#cb278-1" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.GR<span class="sc">$</span>PvalueSorted <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">sort</span>(TtResNoNaC1Removed.GR<span class="sc">$</span>pvalue))</span>
<span id="cb278-2"><a href="#cb278-2" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.GR<span class="sc">$</span>PvalueExpected <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">ppoints</span>(<span class="fu">length</span>(TtResNoNaC1Removed.GR<span class="sc">$</span>PvalueSorted)))</span>
<span id="cb278-3"><a href="#cb278-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-4"><a href="#cb278-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-5"><a href="#cb278-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TtResNoNaC1Removed.GR, <span class="fu">aes</span>(PvalueSorted, PvalueExpected)) <span class="sc">+</span></span>
<span id="cb278-6"><a href="#cb278-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb278-7"><a href="#cb278-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb278-8"><a href="#cb278-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"QQ Plot of Bowtie2 TT-seq C1 Removed - Genotype + Replicate"</span>,</span>
<span id="cb278-9"><a href="#cb278-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Observed -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>), </span>
<span id="cb278-10"><a href="#cb278-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Expected -Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">" pvalue"</span>)) <span class="sc">+</span> </span>
<span id="cb278-11"><a href="#cb278-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb278-12"><a href="#cb278-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb278-13"><a href="#cb278-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-76-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="volcano-plot-8" class="level5">
<h5 class="anchored" data-anchor-id="volcano-plot-8">Volcano Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb279"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb279-1"><a href="#cb279-1" aria-hidden="true" tabindex="-1"></a>TTseqC1Removed.GR_VocPlot <span class="ot">&lt;-</span> TtResNoNaC1Removed.GR[<span class="fu">order</span>(TtResNoNaC1Removed.GR<span class="sc">$</span>padj),]</span>
<span id="cb279-2"><a href="#cb279-2" aria-hidden="true" tabindex="-1"></a>TTseqC1Removed.GR_VocPlot<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb279-3"><a href="#cb279-3" aria-hidden="true" tabindex="-1"></a>TTseqC1Removed.GR_VocPlot<span class="sc">$</span>diffexpressed[TTseqC1Removed.GR_VocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> foldchange_threeshold <span class="sc">&amp;</span> TTseqC1Removed.GR_VocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb279-4"><a href="#cb279-4" aria-hidden="true" tabindex="-1"></a>TTseqC1Removed.GR_VocPlot<span class="sc">$</span>diffexpressed[TTseqC1Removed.GR_VocPlot<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>foldchange_threeshold <span class="sc">&amp;</span> TTseqC1Removed.GR_VocPlot<span class="sc">$</span>padj <span class="sc">&lt;</span> padj_threeshold] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb279-5"><a href="#cb279-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb279-6"><a href="#cb279-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> TTseqC1Removed.GR_VocPlot, </span>
<span id="cb279-7"><a href="#cb279-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb279-8"><a href="#cb279-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>foldchange_threeshold, foldchange_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb279-9"><a href="#cb279-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(padj_threeshold), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb279-10"><a href="#cb279-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">"Volcano Plot Of Bowtie2 TT-seq C1 Removed - Genotype + Replicate"</span>,</span>
<span id="cb279-11"><a href="#cb279-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Category'</span>,  </span>
<span id="cb279-12"><a href="#cb279-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"(FC)"</span>), </span>
<span id="cb279-13"><a href="#cb279-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(padj)"</span>)) <span class="sc">+</span> </span>
<span id="cb279-14"><a href="#cb279-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">14</span>, <span class="dv">14</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb279-15"><a href="#cb279-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb279-16"><a href="#cb279-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#FFDB6D"</span>),</span>
<span id="cb279-17"><a href="#cb279-17" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>))<span class="sc">+</span></span>
<span id="cb279-18"><a href="#cb279-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-77-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bar-plot-top-20-degs-8" class="level5">
<h5 class="anchored" data-anchor-id="bar-plot-top-20-degs-8">Bar Plot Top 20 Degs</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb280"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb280-1"><a href="#cb280-1" aria-hidden="true" tabindex="-1"></a>TTC1RemovedUpReg.GR <span class="ot">&lt;-</span> TTseqC1Removed.GR_VocPlot[TTseqC1Removed.GR_VocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"UP"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb280-2"><a href="#cb280-2" aria-hidden="true" tabindex="-1"></a>TTC1RemovedDownReg.GR <span class="ot">&lt;-</span> TTseqC1Removed.GR_VocPlot[TTseqC1Removed.GR_VocPlot<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">"DOWN"</span>,][,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb280-3"><a href="#cb280-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb280-4"><a href="#cb280-4" aria-hidden="true" tabindex="-1"></a>TTC1RemovedUpReg.GR<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TTC1RemovedUpReg.GR<span class="sc">$</span>padj)</span>
<span id="cb280-5"><a href="#cb280-5" aria-hidden="true" tabindex="-1"></a>TTC1RemovedDownReg.GR<span class="sc">$</span>neg_log10_padj <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(TTC1RemovedDownReg.GR<span class="sc">$</span>padj)</span>
<span id="cb280-6"><a href="#cb280-6" aria-hidden="true" tabindex="-1"></a>TTC1RemovedUpRegSorted.GR <span class="ot">&lt;-</span> TTC1RemovedUpReg.GR[<span class="fu">order</span>(<span class="sc">-</span>TTC1RemovedUpReg.GR<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TTC1RemovedUpReg.GR<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb280-7"><a href="#cb280-7" aria-hidden="true" tabindex="-1"></a>TTC1RemovedDownRegSorted.GR <span class="ot">&lt;-</span> TTC1RemovedDownReg.GR[<span class="fu">order</span>(TTC1RemovedDownReg.GR<span class="sc">$</span>log2FoldChange, <span class="sc">-</span>TTC1RemovedDownReg.GR<span class="sc">$</span>neg_log10_padj),]</span>
<span id="cb280-8"><a href="#cb280-8" aria-hidden="true" tabindex="-1"></a>top_upregTTC1Removed.GR <span class="ot">&lt;-</span> <span class="fu">head</span>(TTC1RemovedUpRegSorted.GR, <span class="dv">50</span>)</span>
<span id="cb280-9"><a href="#cb280-9" aria-hidden="true" tabindex="-1"></a>top_downregTTC1Removed.GR <span class="ot">&lt;-</span> <span class="fu">head</span>(TTC1RemovedDownRegSorted.GR, <span class="dv">50</span>)</span>
<span id="cb280-10"><a href="#cb280-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb280-11"><a href="#cb280-11" aria-hidden="true" tabindex="-1"></a>top_upregTTC1Removed.GR<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_upregTTC1Removed.GR), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb280-12"><a href="#cb280-12" aria-hidden="true" tabindex="-1"></a>top_upregTTC1Removed.GR<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_upregTTC1Removed.GR<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_upregTTC1Removed.GR<span class="sc">$</span>gene))</span>
<span id="cb280-13"><a href="#cb280-13" aria-hidden="true" tabindex="-1"></a>top_downregTTC1Removed.GR<span class="sc">$</span>gene <span class="ot">&lt;-</span> MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(top_downregTTC1Removed.GR), MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb280-14"><a href="#cb280-14" aria-hidden="true" tabindex="-1"></a>top_downregTTC1Removed.GR<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">factor</span>(top_downregTTC1Removed.GR<span class="sc">$</span>gene, <span class="at">levels =</span> <span class="fu">rev</span>(top_downregTTC1Removed.GR<span class="sc">$</span>gene))</span>
<span id="cb280-15"><a href="#cb280-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb280-16"><a href="#cb280-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb280-17"><a href="#cb280-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_upregTTC1Removed.GR, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb280-18"><a href="#cb280-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb280-19"><a href="#cb280-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb280-20"><a href="#cb280-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Upregulated Genes [Bowtie TTseq C1 Removed - Genotype + Replicate]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb280-21"><a href="#cb280-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-78-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb281"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb281-1"><a href="#cb281-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_downregTTC1Removed.GR, <span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span></span>
<span id="cb281-2"><a href="#cb281-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span>  </span>
<span id="cb281-3"><a href="#cb281-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb281-4"><a href="#cb281-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Most Downregulated Genes [Bowtie TTseq C1 Removed - Genotype + Replicate]"</span>, <span class="at">x =</span> <span class="st">"Gene"</span>, <span class="at">y =</span> <span class="st">"log2 Fold Change"</span>) <span class="sc">+</span></span>
<span id="cb281-5"><a href="#cb281-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-78-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>#####Scatter Plot Mean Expression</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb282"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb282-1"><a href="#cb282-1" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.GR <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">counts</span>(ttseq_dds_c1_removed.GR, <span class="at">normalize=</span><span class="cn">TRUE</span>))</span>
<span id="cb282-2"><a href="#cb282-2" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.GR<span class="sc">$</span>WT_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTseqC1RemovedNormalized.GR[,<span class="fu">rownames</span>(ttseq_description_C1_removed[ttseq_description_C1_removed<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb282-3"><a href="#cb282-3" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.GR<span class="sc">$</span>LOW_Mean <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTseqC1RemovedNormalized.GR[,<span class="fu">rownames</span>(ttseq_description_C1_removed[ttseq_description_C1_removed<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, mean))</span>
<span id="cb282-4"><a href="#cb282-4" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.GR<span class="sc">$</span>WT_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTseqC1RemovedNormalized.GR[,<span class="fu">rownames</span>(ttseq_description_C1_removed[ttseq_description_C1_removed<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"WT"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb282-5"><a href="#cb282-5" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.GR<span class="sc">$</span>LOW_std <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">apply</span>(TTseqC1RemovedNormalized.GR[,<span class="fu">rownames</span>(ttseq_description_C1_removed[ttseq_description_C1_removed<span class="sc">$</span>Genotype<span class="sc">==</span><span class="st">"LOW"</span>,])], <span class="dv">1</span>, sd))</span>
<span id="cb282-6"><a href="#cb282-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb282-7"><a href="#cb282-7" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedFiltered.GR <span class="ot">&lt;-</span> TTseqC1RemovedNormalized.GR <span class="sc">%&gt;%</span> <span class="fu">filter</span>(WT_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb282-8"><a href="#cb282-8" aria-hidden="true" tabindex="-1"></a>                                                        (WT_Mean <span class="sc">-</span> WT_std <span class="sc">&gt;</span> <span class="dv">0</span>),</span>
<span id="cb282-9"><a href="#cb282-9" aria-hidden="true" tabindex="-1"></a>                                                        LOW_Mean <span class="sc">&gt;</span> <span class="dv">0</span>,</span>
<span id="cb282-10"><a href="#cb282-10" aria-hidden="true" tabindex="-1"></a>                                                        (LOW_Mean <span class="sc">-</span> LOW_std <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb282-11"><a href="#cb282-11" aria-hidden="true" tabindex="-1"></a>TtResNoNaC1Removed.GR<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TtResNoNaC1Removed.GR)</span>
<span id="cb282-12"><a href="#cb282-12" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedFiltered.GR<span class="sc">$</span>index <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTseqC1RemovedNormalizedFiltered.GR)</span>
<span id="cb282-13"><a href="#cb282-13" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedFiltered.GR <span class="ot">&lt;-</span> <span class="fu">merge</span>(TTseqC1RemovedNormalizedFiltered.GR, <span class="fu">data.frame</span>(TtResNoNaC1Removed.GR[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">10</span>)]), <span class="at">by=</span><span class="st">"index"</span>)</span>
<span id="cb282-14"><a href="#cb282-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TTseqC1RemovedNormalizedFiltered.GR) <span class="ot">&lt;-</span> TTseqC1RemovedNormalizedFiltered.GR<span class="sc">$</span>index</span>
<span id="cb282-15"><a href="#cb282-15" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedFiltered.GR <span class="ot">&lt;-</span> TTseqC1RemovedNormalizedFiltered.GR[,<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(TTseqC1RemovedNormalizedFiltered.GR)]</span>
<span id="cb282-16"><a href="#cb282-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb282-17"><a href="#cb282-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb282-18"><a href="#cb282-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(TTseqC1RemovedNormalizedFiltered.GR, <span class="fu">aes</span>(<span class="at">x=</span> WT_Mean, <span class="at">y=</span> LOW_Mean, <span class="at">col =</span> significant)) <span class="sc">+</span></span>
<span id="cb282-19"><a href="#cb282-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb282-20"><a href="#cb282-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb282-21"><a href="#cb282-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb282-22"><a href="#cb282-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>( <span class="at">title =</span> <span class="st">"Scatter plot of Bowtie2 TT-seq C1 Removed - Genotype + Replicate"</span>,</span>
<span id="cb282-23"><a href="#cb282-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">'Category'</span>,</span>
<span id="cb282-24"><a href="#cb282-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in WT"</span>), </span>
<span id="cb282-25"><a href="#cb282-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"(meanExpression) in LOW"</span>)) <span class="sc">+</span> </span>
<span id="cb282-26"><a href="#cb282-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density2d</span>(<span class="at">col=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb282-27"><a href="#cb282-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"blue"</span>),</span>
<span id="cb282-28"><a href="#cb282-28" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Not Significant"</span>, <span class="st">"Significant"</span>))<span class="sc">+</span></span>
<span id="cb282-29"><a href="#cb282-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbar(data = TTseqC1RemovedNormalizedFiltered.G[1:50,] ,aes(ymin = LOW_Mean - LOW_std, ymax = LOW_Mean + LOW_std), width = 0.1, alpha = 0.2) +</span></span>
<span id="cb282-30"><a href="#cb282-30" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_errorbarh(aes(xmin = WT_Mean - WT_std, xmax = WT_Mean + WT_std), height = 0.1, alpha = 0.2) +</span></span>
<span id="cb282-31"><a href="#cb282-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-79-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="correlation-matrix-ssample-6" class="level5">
<h5 class="anchored" data-anchor-id="correlation-matrix-ssample-6">Correlation Matrix SSample</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb283"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb283-1"><a href="#cb283-1" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Genotype =</span> ttseq_description_C1_removed<span class="sc">$</span>Genotype)</span>
<span id="cb283-2"><a href="#cb283-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(annotation) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ttseq_description_C1_removed)</span>
<span id="cb283-3"><a href="#cb283-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-4"><a href="#cb283-4" aria-hidden="true" tabindex="-1"></a>spearman_correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(TTseqC1RemovedNormalized.GR[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TTseqC1RemovedNormalized.GR))], <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb283-5"><a href="#cb283-5" aria-hidden="true" tabindex="-1"></a>dist_from_spearman <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>spearman_correlation_matrix)</span>
<span id="cb283-6"><a href="#cb283-6" aria-hidden="true" tabindex="-1"></a>dist_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(dist_from_spearman)</span>
<span id="cb283-7"><a href="#cb283-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-8"><a href="#cb283-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(dist_matrix, </span>
<span id="cb283-9"><a href="#cb283-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> dist_from_spearman,</span>
<span id="cb283-10"><a href="#cb283-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> dist_from_spearman,</span>
<span id="cb283-11"><a href="#cb283-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_row =</span> annotation,</span>
<span id="cb283-12"><a href="#cb283-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> annotation,</span>
<span id="cb283-13"><a href="#cb283-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">plasma</span>(<span class="dv">256</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-80-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb284"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb284-1"><a href="#cb284-1" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalized.GR<span class="sc">$</span>gene_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(TTseqC1RemovedNormalized.GR)</span>
<span id="cb284-2"><a href="#cb284-2" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedLong.GR <span class="ot">&lt;-</span> </span>
<span id="cb284-3"><a href="#cb284-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(TTseqC1RemovedNormalized.GR[,<span class="fu">grepl</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(TTseqC1RemovedNormalized.GR)) <span class="sc">|</span> <span class="fu">colnames</span>(TTseqC1RemovedNormalized.GR) <span class="sc">==</span> <span class="st">"gene_name"</span>],</span>
<span id="cb284-4"><a href="#cb284-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">id.vars =</span> <span class="st">"gene_name"</span>,</span>
<span id="cb284-5"><a href="#cb284-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">variable_name =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb284-6"><a href="#cb284-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TTseqC1RemovedNormalizedLong.GR)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"sample_name"</span></span>
<span id="cb284-7"><a href="#cb284-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TTseqC1RemovedNormalizedLong.GR)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"normalized_count"</span></span>
<span id="cb284-8"><a href="#cb284-8" aria-hidden="true" tabindex="-1"></a>TTseqC1RemovedNormalizedLong.GR <span class="ot">&lt;-</span> <span class="fu">merge</span>(TTseqC1RemovedNormalizedLong.GR, ttseq_description_C1_removed, <span class="at">by=</span><span class="st">"sample_name"</span>)</span>
<span id="cb284-9"><a href="#cb284-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TTseqC1RemovedNormalizedLong.GR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     sample_name             gene_name normalized_count
1 RNA_S11589Nr70  ENSMUSG00000000001.4      6774.331066
2 RNA_S11589Nr70 ENSMUSG00000000003.15         0.000000
3 RNA_S11589Nr70 ENSMUSG00000000028.15      4729.202526
4 RNA_S11589Nr70 ENSMUSG00000000031.16        69.177165
5 RNA_S11589Nr70 ENSMUSG00000000037.17      1734.460182
6 RNA_S11589Nr70 ENSMUSG00000000049.11         2.515533
                       sample                                     fastq_1
1 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
2 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
3 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
4 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
5 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
6 H1_WT_C2_S2spike_TTseq_Rep1 S11589_2/01_fastq/RNA_S11589Nr70.1.fastq.gz
                                      fastq_2 strandedness Genotype Colonies
1 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
2 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
3 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
4 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
5 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
6 S11589_2/01_fastq/RNA_S11589Nr70.2.fastq.gz         auto       WT       C2
  Method Replicate Treatment
1  TTseq      Rep1      &lt;NA&gt;
2  TTseq      Rep1      &lt;NA&gt;
3  TTseq      Rep1      &lt;NA&gt;
4  TTseq      Rep1      &lt;NA&gt;
5  TTseq      Rep1      &lt;NA&gt;
6  TTseq      Rep1      &lt;NA&gt;</code></pre>
</div>
</div>
</section>
<section id="top-10-up-degs-bar-plot-7" class="level5">
<h5 class="anchored" data-anchor-id="top-10-up-degs-bar-plot-7">Top 10 Up Degs Bar Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb286"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb286-1"><a href="#cb286-1" aria-hidden="true" tabindex="-1"></a>top10_upregTTseqC1Removed.GR <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top_upregTTC1Removed.GR[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,])</span>
<span id="cb286-2"><a href="#cb286-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_upregTTseqC1Removed.GR) {</span>
<span id="cb286-3"><a href="#cb286-3" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTseqC1RemovedNormalizedLong.GR[TTseqC1RemovedNormalizedLong.GR<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb286-4"><a href="#cb286-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb286-5"><a href="#cb286-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb286-6"><a href="#cb286-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Replicate), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb286-7"><a href="#cb286-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb286-8"><a href="#cb286-8" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb286-9"><a href="#cb286-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb286-10"><a href="#cb286-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb286-11"><a href="#cb286-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb286-12"><a href="#cb286-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb286-13"><a href="#cb286-13" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-82-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="top-10-down-degs-bar-plot-6" class="level5">
<h5 class="anchored" data-anchor-id="top-10-down-degs-bar-plot-6">Top 10 Down Degs Bar Plot</h5>
<div class="cell">
<div class="sourceCode cell-code" id="cb287"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb287-1"><a href="#cb287-1" aria-hidden="true" tabindex="-1"></a>top10_downregTTseqC1Removed.GR <span class="ot">&lt;-</span> <span class="fu">rownames</span>(<span class="fu">head</span>(top_downregTTC1Removed.G, <span class="dv">10</span>))</span>
<span id="cb287-2"><a href="#cb287-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-3"><a href="#cb287-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> top10_downregTTseqC1Removed.GR) {</span>
<span id="cb287-4"><a href="#cb287-4" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(TTseqC1RemovedNormalizedLong.GR[TTseqC1RemovedNormalizedLong.GR<span class="sc">$</span>gene_name <span class="sc">==</span> gene, ],</span>
<span id="cb287-5"><a href="#cb287-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> normalized_count)) <span class="sc">+</span></span>
<span id="cb287-6"><a href="#cb287-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="fu">aes</span>(<span class="at">fill =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb287-7"><a href="#cb287-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Replicate), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb287-8"><a href="#cb287-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Expression of"</span>, </span>
<span id="cb287-9"><a href="#cb287-9" aria-hidden="true" tabindex="-1"></a>                       MusGeneNames<span class="sc">$</span>gene_name[MusGeneNames<span class="sc">$</span>gene_id <span class="sc">==</span> gene]),</span>
<span id="cb287-10"><a href="#cb287-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Normalized Counts"</span>),</span>
<span id="cb287-11"><a href="#cb287-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Genotype"</span>) <span class="sc">+</span></span>
<span id="cb287-12"><a href="#cb287-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb287-13"><a href="#cb287-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb287-14"><a href="#cb287-14" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-83-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
</section>
<section id="karyoploter" class="level2">
<h2 class="anchored" data-anchor-id="karyoploter">karyoploteR</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb288"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb288-1"><a href="#cb288-1" aria-hidden="true" tabindex="-1"></a>fai <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/home/franck/SCRATCH/fpan/Genome_Index/chR_index_files/cat_chR_MusDroso.fasta.fai"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb288-2"><a href="#cb288-2" aria-hidden="true" tabindex="-1"></a>session <span class="ot">&lt;-</span> <span class="fu">browserSession</span>(<span class="st">"UCSC"</span>)</span>
<span id="cb288-3"><a href="#cb288-3" aria-hidden="true" tabindex="-1"></a><span class="fu">genome</span>(session) <span class="ot">&lt;-</span> <span class="st">"mm10"</span></span>
<span id="cb288-4"><a href="#cb288-4" aria-hidden="true" tabindex="-1"></a>cyto <span class="ot">&lt;-</span> <span class="fu">track</span>(session, <span class="st">"cytoBand"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb289"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb289-1"><a href="#cb289-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(fai) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"length"</span>, <span class="st">"offset"</span>, <span class="st">"line_bases"</span>, <span class="st">"line_bytes"</span>)</span>
<span id="cb289-2"><a href="#cb289-2" aria-hidden="true" tabindex="-1"></a>fai <span class="ot">&lt;-</span> fai[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"^Droso_|^GL|^JH"</span>, fai<span class="sc">$</span>chr),]</span>
<span id="cb289-3"><a href="#cb289-3" aria-hidden="true" tabindex="-1"></a>custom.genome <span class="ot">&lt;-</span> <span class="fu">GRanges</span>(<span class="at">seqnames =</span> fai<span class="sc">$</span>chr,</span>
<span id="cb289-4"><a href="#cb289-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="dv">1</span>, <span class="at">end =</span> fai<span class="sc">$</span>length))</span>
<span id="cb289-5"><a href="#cb289-5" aria-hidden="true" tabindex="-1"></a>desired_order <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"chr"</span>, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">19</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>, <span class="st">"M"</span>, <span class="st">"R"</span>))</span>
<span id="cb289-6"><a href="#cb289-6" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(custom.genome) <span class="ot">&lt;-</span> desired_order</span>
<span id="cb289-7"><a href="#cb289-7" aria-hidden="true" tabindex="-1"></a>custom.genome <span class="ot">&lt;-</span> <span class="fu">sort</span>(custom.genome)</span>
<span id="cb289-8"><a href="#cb289-8" aria-hidden="true" tabindex="-1"></a>mus_chrs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"chr"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">19</span>), <span class="st">"chrX"</span>, <span class="st">"chrY"</span>)</span>
<span id="cb289-9"><a href="#cb289-9" aria-hidden="true" tabindex="-1"></a>filtered_gtf <span class="ot">&lt;-</span> latest_gtf[<span class="fu">seqnames</span>(latest_gtf) <span class="sc">%in%</span> mus_chrs]</span>
<span id="cb289-10"><a href="#cb289-10" aria-hidden="true" tabindex="-1"></a>gene_features <span class="ot">&lt;-</span> filtered_gtf[filtered_gtf<span class="sc">$</span>type <span class="sc">==</span> <span class="st">"gene"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb290"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb290-1"><a href="#cb290-1" aria-hidden="true" tabindex="-1"></a>cyto <span class="ot">&lt;-</span> <span class="fu">keepSeqlevels</span>(cyto, mus_chrs, <span class="at">pruning.mode=</span><span class="st">"coarse"</span>)</span>
<span id="cb290-2"><a href="#cb290-2" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">6</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb290-3"><a href="#cb290-3" aria-hidden="true" tabindex="-1"></a><span class="fu">kpDataBackground</span>(kp, <span class="at">color =</span> <span class="st">"#FFFFFFAA"</span>)</span>
<span id="cb290-4"><a href="#cb290-4" aria-hidden="true" tabindex="-1"></a><span class="fu">kpPlotDensity</span>(kp, gene_features, <span class="at">window.size =</span> <span class="fl">0.5e6</span>, <span class="at">data.panel=</span><span class="st">"ideogram"</span>, <span class="at">col=</span><span class="st">"red"</span>, <span class="at">border=</span><span class="st">"red"</span>, <span class="at">r0=</span><span class="fl">0.5</span>, <span class="at">r1=</span><span class="dv">1</span>)</span>
<span id="cb290-5"><a href="#cb290-5" aria-hidden="true" tabindex="-1"></a><span class="fu">kpPlotDensity</span>(kp, gene_features, <span class="at">window.size =</span> <span class="fl">0.5e6</span>, <span class="at">data.panel=</span><span class="st">"ideogram"</span>, <span class="at">col=</span><span class="st">"red"</span>, <span class="at">border=</span><span class="st">"red"</span>, <span class="at">r0=</span><span class="fl">0.5</span>, <span class="at">r1=</span><span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-86-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb291"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb291-1"><a href="#cb291-1" aria-hidden="true" tabindex="-1"></a>TPM <span class="ot">&lt;-</span> TTtxi_tx<span class="sc">$</span>abundance</span>
<span id="cb291-2"><a href="#cb291-2" aria-hidden="true" tabindex="-1"></a>TPM <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(TPM)</span>
<span id="cb291-3"><a href="#cb291-3" aria-hidden="true" tabindex="-1"></a>TPM <span class="ot">&lt;-</span> TPM[<span class="fu">grepl</span>(<span class="st">"^ENS|^PreMrnaENS"</span>, <span class="fu">rownames</span>(TPM)),]</span>
<span id="cb291-4"><a href="#cb291-4" aria-hidden="true" tabindex="-1"></a>TPM<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> tx2gene<span class="sc">$</span>GENEID[<span class="fu">match</span>(<span class="fu">rownames</span>(TPM), tx2gene<span class="sc">$</span>TXNAME)]</span>
<span id="cb291-5"><a href="#cb291-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TPM)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                             RNA_S11589Nr69 RNA_S11589Nr70 RNA_S11589Nr71
ENSMUST00000000001.4               5.244005       5.862483       6.926472
PreMrnaENSMUSG00000000001.4        3.996062       4.527589       5.576653
ENSMUST00000000003.13              0.000000       0.000000       0.000000
ENSMUST00000114041.2               0.000000       0.000000       0.000000
PreMrnaENSMUSG00000000003.15       0.024784       0.017782       0.004770
ENSMUST00000000028.13              2.304408       3.788864       2.805012
                             RNA_S11589Nr72 RNA_S11589Nr73 RNA_S11589Nr74
ENSMUST00000000001.4               5.628878       5.431469       5.598580
PreMrnaENSMUSG00000000001.4        5.113059       3.939640       2.785029
ENSMUST00000000003.13              0.000000       0.000000       0.000000
ENSMUST00000114041.2               0.000000       0.000000       0.000000
PreMrnaENSMUSG00000000003.15       0.016212       0.014236       0.000000
ENSMUST00000000028.13              1.949744       3.121570       2.466538
                             RNA_S11589Nr75 RNA_S11589Nr76 RNA_S11589Nr77
ENSMUST00000000001.4               6.644234       5.562029       5.186139
PreMrnaENSMUSG00000000001.4        5.273426       5.334046       3.705405
ENSMUST00000000003.13              0.000000       0.000000       0.000000
ENSMUST00000114041.2               0.000000       0.000000       0.000000
PreMrnaENSMUSG00000000003.15       0.016504       0.005454       0.019669
ENSMUST00000000028.13              2.953617       2.985884       2.762844
                             RNA_S11589Nr78 RNA_S11589Nr79 RNA_S11589Nr80
ENSMUST00000000001.4               6.595703       7.240502       4.863424
PreMrnaENSMUSG00000000001.4        4.860146       6.002424       6.518097
ENSMUST00000000003.13              0.000000       0.000000       0.000000
ENSMUST00000114041.2               0.000000       0.000000       0.000000
PreMrnaENSMUSG00000000003.15       0.004425       0.017906       0.066526
ENSMUST00000000028.13              2.637946       1.943336       2.214210
                                           gene_id
ENSMUST00000000001.4          ENSMUSG00000000001.4
PreMrnaENSMUSG00000000001.4   ENSMUSG00000000001.4
ENSMUST00000000003.13        ENSMUSG00000000003.15
ENSMUST00000114041.2         ENSMUSG00000000003.15
PreMrnaENSMUSG00000000003.15 ENSMUSG00000000003.15
ENSMUST00000000028.13        ENSMUSG00000000028.15</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb293"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb293-1"><a href="#cb293-1" aria-hidden="true" tabindex="-1"></a>transcript_density <span class="ot">&lt;-</span> TPM <span class="sc">%&gt;%</span></span>
<span id="cb293-2"><a href="#cb293-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb293-3"><a href="#cb293-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb293-4"><a href="#cb293-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb293-5"><a href="#cb293-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb293-6"><a href="#cb293-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(transcript_density) <span class="ot">&lt;-</span> transcript_density<span class="sc">$</span>gene_id</span>
<span id="cb293-7"><a href="#cb293-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(transcript_density)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"score"</span></span>
<span id="cb293-8"><a href="#cb293-8" aria-hidden="true" tabindex="-1"></a>transcript_density<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb293-9"><a href="#cb293-9" aria-hidden="true" tabindex="-1"></a>trdf <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(transcript_density)</span>
<span id="cb293-10"><a href="#cb293-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(trdf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 6 rows and 1 column
                          score
                      &lt;integer&gt;
ENSMUSG00000000001.4          2
ENSMUSG00000000003.15         3
ENSMUSG00000000028.15         5
ENSMUSG00000000031.16         8
ENSMUSG00000000037.17        10
ENSMUSG00000000049.11         5</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb295"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb295-1"><a href="#cb295-1" aria-hidden="true" tabindex="-1"></a>TPM_gene <span class="ot">&lt;-</span> TPM <span class="sc">%&gt;%</span></span>
<span id="cb295-2"><a href="#cb295-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb295-3"><a href="#cb295-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), \(x) <span class="fu">sum</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb295-4"><a href="#cb295-4" aria-hidden="true" tabindex="-1"></a>TPM_gene <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(TPM_gene)</span>
<span id="cb295-5"><a href="#cb295-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TPM_gene) <span class="ot">&lt;-</span> TPM_gene<span class="sc">$</span>gene_id</span>
<span id="cb295-6"><a href="#cb295-6" aria-hidden="true" tabindex="-1"></a>TPM_gene <span class="ot">&lt;-</span> TPM_gene[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb295-7"><a href="#cb295-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TPM_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      RNA_S11589Nr69 RNA_S11589Nr70 RNA_S11589Nr71
ENSMUSG00000000001.4        9.240067      10.390072      12.503125
ENSMUSG00000000003.15       0.024784       0.017782       0.004770
ENSMUSG00000000028.15      10.028642       8.211139       9.976762
ENSMUSG00000000031.16       0.656462       1.247191       1.847492
ENSMUSG00000000037.17       0.750342       0.607503       0.800054
ENSMUSG00000000049.11       0.009476       0.032136       0.031224
                      RNA_S11589Nr72 RNA_S11589Nr73 RNA_S11589Nr74
ENSMUSG00000000001.4       10.741937       9.371109       8.383609
ENSMUSG00000000003.15       0.016212       0.014236       0.000000
ENSMUSG00000000028.15      12.667579      10.694930       7.668825
ENSMUSG00000000031.16       0.066239       0.856605       0.757170
ENSMUSG00000000037.17       0.299384       0.752512       0.866711
ENSMUSG00000000049.11       0.286363       0.004023       0.015747
                      RNA_S11589Nr75 RNA_S11589Nr76 RNA_S11589Nr77
ENSMUSG00000000001.4       11.917660      10.896075       8.891544
ENSMUSG00000000003.15       0.016504       0.005454       0.019669
ENSMUSG00000000028.15      10.546855      11.269407       8.699807
ENSMUSG00000000031.16       0.963168       0.126466       0.713319
ENSMUSG00000000037.17       1.206909       0.550232       0.583609
ENSMUSG00000000049.11       0.023736       0.385340       0.006421
                      RNA_S11589Nr78 RNA_S11589Nr79 RNA_S11589Nr80
ENSMUSG00000000001.4       11.455849      13.242926      11.381521
ENSMUSG00000000003.15       0.004425       0.017906       0.066526
ENSMUSG00000000028.15       8.855206       9.481148      11.751154
ENSMUSG00000000031.16       0.927211       1.385282       0.025864
ENSMUSG00000000037.17       0.888402       1.315605       0.372120
ENSMUSG00000000049.11       0.031892       0.109004       0.246762</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb297"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb297-1"><a href="#cb297-1" aria-hidden="true" tabindex="-1"></a>gtf_tpm <span class="ot">&lt;-</span> gtf_genes[gtf_genes<span class="sc">$</span>gene_id <span class="sc">%in%</span> <span class="fu">unique</span>(MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb297-2"><a href="#cb297-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(gtf_tpm) <span class="ot">&lt;-</span> <span class="fu">mcols</span>(gtf_tpm)[, <span class="fu">c</span>(<span class="st">"gene_name"</span>,<span class="st">"gene_id"</span>), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb297-3"><a href="#cb297-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(gtf_tpm) <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">log2</span>(TPM_gene[gtf_tpm<span class="sc">$</span>gene_id, <span class="fu">colnames</span>(TPM_gene)] <span class="sc">+</span> <span class="dv">1</span>), gtf_tpm<span class="sc">$</span>gene_id, gtf_tpm<span class="sc">$</span>gene_name)</span>
<span id="cb297-4"><a href="#cb297-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gtf_tpm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 6 ranges and 14 metadata columns:
      seqnames              ranges strand | RNA_S11589Nr69 RNA_S11589Nr70
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;numeric&gt;      &lt;numeric&gt;
  [1]     chr3 108107280-108146146      - |      3.3561532      3.5097050
  [2]     chrX   77837901-77853623      - |      0.0353199      0.0254286
  [3]    chr16   18780447-18811987      - |      3.4631833      3.2033796
  [4]     chr7 142575529-142578143      - |      0.7281051      1.1681228
  [5]     chrX 161082525-161258213      + |      0.8076368      0.6848214
  [6]    chr11 108343354-108414396      + |      0.0136066      0.0456331
      RNA_S11589Nr71 RNA_S11589Nr72 RNA_S11589Nr73 RNA_S11589Nr74
           &lt;numeric&gt;      &lt;numeric&gt;      &lt;numeric&gt;      &lt;numeric&gt;
  [1]     3.75522142      3.5535985     3.37449827      3.2301429
  [2]     0.00686529      0.0232014     0.02039339      0.0000000
  [3]     3.45638064      3.7726858     3.54781132      3.1158365
  [4]     1.50969179      0.0925309     0.89266691      0.8132538
  [5]     0.84804019      0.3778278     0.80942432      0.9004986
  [6]     0.04435775      0.3632978     0.00579232      0.0225411
      RNA_S11589Nr75 RNA_S11589Nr76 RNA_S11589Nr77 RNA_S11589Nr78
           &lt;numeric&gt;      &lt;numeric&gt;      &lt;numeric&gt;      &lt;numeric&gt;
  [1]      3.6912728     3.57241374     3.30619573     3.63875145
  [2]      0.0236159     0.00784708     0.02810091     0.00636984
  [3]      3.5294281     3.61699362     3.27795604     3.30088603
  [4]      0.9731836     0.17180377     0.77679379     0.94651453
  [5]      1.1420271     0.63248414     0.66321617     0.91716592
  [6]      0.0338437     0.47024010     0.00923393     0.04529198
      RNA_S11589Nr79 RNA_S11589Nr80       gtf_tpm.gene_id gtf_tpm.gene_name
           &lt;numeric&gt;      &lt;numeric&gt;           &lt;character&gt;       &lt;character&gt;
  [1]      3.8321737      3.6301166  ENSMUSG00000000001.4             Gnai3
  [2]      0.0256043      0.0929191 ENSMUSG00000000003.15              Pbsn
  [3]      3.3897248      3.6725559 ENSMUSG00000000028.15             Cdc45
  [4]      1.2541598      0.0368395 ENSMUSG00000000031.16               H19
  [5]      1.2113892      0.4564067 ENSMUSG00000000037.17             Scml2
  [6]      0.1492646      0.3181861 ENSMUSG00000000049.11              Apoh
  -------
  seqinfo: 71 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb299"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb299-1"><a href="#cb299-1" aria-hidden="true" tabindex="-1"></a>gtf_covTr <span class="ot">&lt;-</span> gtf_genes[gtf_genes<span class="sc">$</span>gene_id <span class="sc">%in%</span> <span class="fu">unique</span>(MusGeneNames<span class="sc">$</span>gene_id)]</span>
<span id="cb299-2"><a href="#cb299-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(gtf_covTr) <span class="ot">&lt;-</span> <span class="fu">mcols</span>(gtf_covTr)[, <span class="st">"gene_id"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb299-3"><a href="#cb299-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(gtf_covTr) <span class="ot">&lt;-</span> trdf[gtf_covTr<span class="sc">$</span>gene_id, <span class="fu">colnames</span>(trdf)]</span>
<span id="cb299-4"><a href="#cb299-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gtf_covTr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>GRanges object with 6 ranges and 1 metadata column:
      seqnames              ranges strand |         X
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;
  [1]     chr3 108107280-108146146      - |         2
  [2]     chrX   77837901-77853623      - |         3
  [3]    chr16   18780447-18811987      - |         5
  [4]     chr7 142575529-142578143      - |         8
  [5]     chrX 161082525-161258213      + |        10
  [6]    chr11 108343354-108414396      + |         5
  -------
  seqinfo: 71 sequences from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<section id="compare_chr_ttsample" class="level3">
<h3 class="anchored" data-anchor-id="compare_chr_ttsample">Compare_Chr_TTsample</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb301"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb301-1"><a href="#cb301-1" aria-hidden="true" tabindex="-1"></a>height_per_sample <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">/</span> <span class="fu">length</span>(mus_chrs)</span>
<span id="cb301-2"><a href="#cb301-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(mus_chrs)){</span>
<span id="cb301-3"><a href="#cb301-3" aria-hidden="true" tabindex="-1"></a>  nth_chr <span class="ot">&lt;-</span> mus_chrs[i]</span>
<span id="cb301-4"><a href="#cb301-4" aria-hidden="true" tabindex="-1"></a>  chr_genes <span class="ot">&lt;-</span> gtf_tpm[<span class="fu">seqnames</span>(gtf_tpm)<span class="sc">==</span>nth_chr]</span>
<span id="cb301-5"><a href="#cb301-5" aria-hidden="true" tabindex="-1"></a>  bin_size <span class="ot">&lt;-</span> <span class="fl">1e6</span></span>
<span id="cb301-6"><a href="#cb301-6" aria-hidden="true" tabindex="-1"></a>  chr_length <span class="ot">&lt;-</span> <span class="fu">width</span>(custom.genome)[i]</span>
<span id="cb301-7"><a href="#cb301-7" aria-hidden="true" tabindex="-1"></a>  bins <span class="ot">&lt;-</span> <span class="fu">tileGenome</span>(<span class="at">seqlengths =</span> <span class="fu">setNames</span>(chr_length, nth_chr),</span>
<span id="cb301-8"><a href="#cb301-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tilewidth =</span> bin_size, </span>
<span id="cb301-9"><a href="#cb301-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">cut.last.tile.in.chrom =</span> <span class="cn">TRUE</span>)</span>
<span id="cb301-10"><a href="#cb301-10" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb301-11"><a href="#cb301-11" aria-hidden="true" tabindex="-1"></a>  kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">genome=</span>custom.genome, <span class="at">chromosomes=</span>nth_chr, <span class="at">cytobands=</span>cyto, <span class="at">plot.type=</span><span class="dv">4</span>)</span>
<span id="cb301-12"><a href="#cb301-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(sample <span class="cf">in</span> <span class="fu">grep</span>(<span class="st">"^RNA"</span>, <span class="fu">colnames</span>(<span class="fu">mcols</span>(chr_genes)), <span class="at">value =</span> <span class="cn">TRUE</span>)){</span>
<span id="cb301-13"><a href="#cb301-13" aria-hidden="true" tabindex="-1"></a>      hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(bins, chr_genes) </span>
<span id="cb301-14"><a href="#cb301-14" aria-hidden="true" tabindex="-1"></a>      sum_expr <span class="ot">&lt;-</span> <span class="fu">tapply</span>(<span class="fu">mcols</span>(chr_genes)[[sample]][<span class="fu">subjectHits</span>(hits)],</span>
<span id="cb301-15"><a href="#cb301-15" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">queryHits</span>(hits), sum, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb301-16"><a href="#cb301-16" aria-hidden="true" tabindex="-1"></a>      expr_bins <span class="ot">&lt;-</span> bins[<span class="fu">as.numeric</span>(<span class="fu">names</span>(sum_expr))]</span>
<span id="cb301-17"><a href="#cb301-17" aria-hidden="true" tabindex="-1"></a>      expr_bins<span class="sc">$</span>expr <span class="ot">&lt;-</span> sum_expr</span>
<span id="cb301-18"><a href="#cb301-18" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb301-19"><a href="#cb301-19" aria-hidden="true" tabindex="-1"></a>      <span class="co">#kpArea(kp, data=expr_bins, y=expr_bins$expr, ymax = max(expr_bins$expr) * 1.1, col = rgb(34/255, 139/255, 34/255, alpha = 0.6), border=NA)</span></span>
<span id="cb301-20"><a href="#cb301-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">kpLines</span>(kp, <span class="at">chr=</span>nth_chr, <span class="at">data =</span> expr_bins, <span class="at">y=</span>expr_bins<span class="sc">$</span>expr, <span class="at">ymin=</span> <span class="dv">0</span>, <span class="at">ymax =</span> <span class="fu">max</span>(expr_bins<span class="sc">$</span>expr) <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col=</span><span class="st">"forestgreen"</span>, <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">r0 =</span> r, </span>
<span id="cb301-21"><a href="#cb301-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">r1 =</span> r <span class="sc">+</span> height_per_sample)</span>
<span id="cb301-22"><a href="#cb301-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">kpPoints</span>(kp, <span class="at">data=</span>expr_bins, <span class="at">y=</span>expr_bins<span class="sc">$</span>expr, <span class="at">ymin=</span> <span class="dv">0</span>, <span class="at">ymax =</span> <span class="fu">max</span>(expr_bins<span class="sc">$</span>expr) <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col=</span><span class="st">"blue"</span>, <span class="at">cex=</span><span class="fl">0.5</span>, <span class="at">r0 =</span> r, <span class="at">r1 =</span> r <span class="sc">+</span> height_per_sample)</span>
<span id="cb301-23"><a href="#cb301-23" aria-hidden="true" tabindex="-1"></a>      <span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> <span class="fu">substr</span>(sample, <span class="fu">nchar</span>(sample)<span class="sc">-</span><span class="dv">1</span>, <span class="fu">nchar</span>(sample)), <span class="at">r0 =</span> r, <span class="at">r1 =</span> r <span class="sc">+</span> height_per_sample, </span>
<span id="cb301-24"><a href="#cb301-24" aria-hidden="true" tabindex="-1"></a>            <span class="at">cex =</span> <span class="fl">0.7</span>, <span class="at">srt =</span> <span class="dv">0</span>, <span class="at">label.margin =</span> <span class="fl">0.01</span>)</span>
<span id="cb301-25"><a href="#cb301-25" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb301-26"><a href="#cb301-26" aria-hidden="true" tabindex="-1"></a>      r <span class="ot">&lt;-</span> r <span class="sc">+</span> height_per_sample</span>
<span id="cb301-27"><a href="#cb301-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb301-28"><a href="#cb301-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-11.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-12.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-13.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-14.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-15.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-16.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-17.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-18.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-19.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-20.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-92-21.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="compare_chr_2sample" class="level3">
<h3 class="anchored" data-anchor-id="compare_chr_2sample">Compare_Chr_2sample</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb302"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb302-1"><a href="#cb302-1" aria-hidden="true" tabindex="-1"></a>interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"RNA_S11589Nr69"</span>,<span class="st">"RNA_S11589Nr71"</span>)</span>
<span id="cb302-2"><a href="#cb302-2" aria-hidden="true" tabindex="-1"></a>gap_ratio <span class="ot">&lt;-</span> <span class="fl">0.1</span> </span>
<span id="cb302-3"><a href="#cb302-3" aria-hidden="true" tabindex="-1"></a>n_tracks <span class="ot">&lt;-</span> <span class="fu">length</span>(interest)</span>
<span id="cb302-4"><a href="#cb302-4" aria-hidden="true" tabindex="-1"></a>height_per_sample <span class="ot">&lt;-</span> (<span class="dv">1</span> <span class="sc">-</span> (n_tracks <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">*</span> gap_ratio) <span class="sc">/</span> n_tracks</span>
<span id="cb302-5"><a href="#cb302-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(mus_chrs)){</span>
<span id="cb302-6"><a href="#cb302-6" aria-hidden="true" tabindex="-1"></a>  nth_chr <span class="ot">&lt;-</span> mus_chrs[i]</span>
<span id="cb302-7"><a href="#cb302-7" aria-hidden="true" tabindex="-1"></a>  chr_genes <span class="ot">&lt;-</span> gtf_tpm[<span class="fu">seqnames</span>(gtf_tpm)<span class="sc">==</span>nth_chr]</span>
<span id="cb302-8"><a href="#cb302-8" aria-hidden="true" tabindex="-1"></a>  bin_size <span class="ot">&lt;-</span> <span class="fl">1e6</span></span>
<span id="cb302-9"><a href="#cb302-9" aria-hidden="true" tabindex="-1"></a>  chr_length <span class="ot">&lt;-</span> <span class="fu">width</span>(custom.genome)[i]</span>
<span id="cb302-10"><a href="#cb302-10" aria-hidden="true" tabindex="-1"></a>  bins <span class="ot">&lt;-</span> <span class="fu">tileGenome</span>(<span class="at">seqlengths =</span> <span class="fu">setNames</span>(chr_length, nth_chr),</span>
<span id="cb302-11"><a href="#cb302-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tilewidth =</span> bin_size, </span>
<span id="cb302-12"><a href="#cb302-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">cut.last.tile.in.chrom =</span> <span class="cn">TRUE</span>)</span>
<span id="cb302-13"><a href="#cb302-13" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb302-14"><a href="#cb302-14" aria-hidden="true" tabindex="-1"></a>  kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">genome=</span>custom.genome, <span class="at">chromosomes=</span>nth_chr, <span class="at">cytobands=</span>cyto, <span class="at">plot.type=</span><span class="dv">4</span>)</span>
<span id="cb302-15"><a href="#cb302-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(sample <span class="cf">in</span> interest){</span>
<span id="cb302-16"><a href="#cb302-16" aria-hidden="true" tabindex="-1"></a>      hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(bins, chr_genes) </span>
<span id="cb302-17"><a href="#cb302-17" aria-hidden="true" tabindex="-1"></a>      sum_expr <span class="ot">&lt;-</span> <span class="fu">tapply</span>(<span class="fu">mcols</span>(chr_genes)[[sample]][<span class="fu">subjectHits</span>(hits)],</span>
<span id="cb302-18"><a href="#cb302-18" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">queryHits</span>(hits), sum, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb302-19"><a href="#cb302-19" aria-hidden="true" tabindex="-1"></a>      expr_bins <span class="ot">&lt;-</span> bins[<span class="fu">as.numeric</span>(<span class="fu">names</span>(sum_expr))]</span>
<span id="cb302-20"><a href="#cb302-20" aria-hidden="true" tabindex="-1"></a>      expr_bins<span class="sc">$</span>expr <span class="ot">&lt;-</span> sum_expr</span>
<span id="cb302-21"><a href="#cb302-21" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb302-22"><a href="#cb302-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">kpArea</span>(kp, <span class="at">data=</span>expr_bins, <span class="at">y=</span>expr_bins<span class="sc">$</span>expr, <span class="at">ymax =</span> <span class="fu">max</span>(expr_bins<span class="sc">$</span>expr) <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col =</span> <span class="fu">rgb</span>(<span class="dv">34</span><span class="sc">/</span><span class="dv">255</span>, <span class="dv">139</span><span class="sc">/</span><span class="dv">255</span>, <span class="dv">34</span><span class="sc">/</span><span class="dv">255</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>), </span>
<span id="cb302-23"><a href="#cb302-23" aria-hidden="true" tabindex="-1"></a>             <span class="at">border=</span><span class="cn">NA</span>, <span class="at">r0 =</span> r, <span class="at">r1 =</span> r <span class="sc">+</span> height_per_sample)</span>
<span id="cb302-24"><a href="#cb302-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">kpLines</span>(kp, <span class="at">chr=</span>nth_chr, <span class="at">data =</span> expr_bins, <span class="at">y=</span>expr_bins<span class="sc">$</span>expr, <span class="at">ymin=</span> <span class="dv">0</span>, <span class="at">ymax =</span> <span class="fu">max</span>(expr_bins<span class="sc">$</span>expr) <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col=</span><span class="st">"forestgreen"</span>, <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">r0 =</span> r, </span>
<span id="cb302-25"><a href="#cb302-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">r1 =</span> r <span class="sc">+</span> height_per_sample)</span>
<span id="cb302-26"><a href="#cb302-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">kpPoints</span>(kp, <span class="at">data=</span>expr_bins, <span class="at">y=</span>expr_bins<span class="sc">$</span>expr, <span class="at">ymin=</span> <span class="dv">0</span>, <span class="at">ymax =</span> <span class="fu">max</span>(expr_bins<span class="sc">$</span>expr) <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col=</span><span class="st">"blue"</span>, <span class="at">cex=</span><span class="fl">0.5</span>, </span>
<span id="cb302-27"><a href="#cb302-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">r0 =</span> r, <span class="at">r1 =</span> r <span class="sc">+</span> height_per_sample)</span>
<span id="cb302-28"><a href="#cb302-28" aria-hidden="true" tabindex="-1"></a>      <span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> <span class="fu">substr</span>(sample, <span class="fu">nchar</span>(sample)<span class="sc">-</span><span class="dv">1</span>, <span class="fu">nchar</span>(sample)), <span class="at">r0 =</span> r <span class="sc">-</span> height_per_sample <span class="sc">*</span> <span class="fl">0.1</span>,</span>
<span id="cb302-29"><a href="#cb302-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">r1 =</span> r , <span class="at">cex =</span> <span class="fl">0.7</span>, <span class="at">srt =</span> <span class="dv">0</span>, <span class="at">label.margin =</span> <span class="fl">0.01</span>, <span class="at">pos =</span> <span class="dv">1</span>)</span>
<span id="cb302-30"><a href="#cb302-30" aria-hidden="true" tabindex="-1"></a>      <span class="fu">kpAxis</span>(kp, <span class="at">ymin =</span> <span class="dv">0</span>, <span class="at">ymax =</span> <span class="fu">max</span>(expr_bins<span class="sc">$</span>expr) <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">r0 =</span> r, <span class="at">r1 =</span> r <span class="sc">+</span> height_per_sample, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">numticks =</span> <span class="dv">4</span>)</span>
<span id="cb302-31"><a href="#cb302-31" aria-hidden="true" tabindex="-1"></a>      r <span class="ot">&lt;-</span> r <span class="sc">+</span> height_per_sample <span class="sc">+</span> gap_ratio</span>
<span id="cb302-32"><a href="#cb302-32" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb302-33"><a href="#cb302-33" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-11.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-12.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-13.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-14.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-15.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-16.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-17.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-18.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-19.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-20.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-93-21.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="compare_chr_1max_2sample" class="level3">
<h3 class="anchored" data-anchor-id="compare_chr_1max_2sample">Compare_Chr_1max_2sample</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb303"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb303-1"><a href="#cb303-1" aria-hidden="true" tabindex="-1"></a>interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"RNA_S11589Nr69"</span>,<span class="st">"RNA_S11589Nr71"</span>)</span>
<span id="cb303-2"><a href="#cb303-2" aria-hidden="true" tabindex="-1"></a>height_per_sample <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">/</span> <span class="fu">length</span>(interest)</span>
<span id="cb303-3"><a href="#cb303-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(mus_chrs)){</span>
<span id="cb303-4"><a href="#cb303-4" aria-hidden="true" tabindex="-1"></a>  nth_chr <span class="ot">&lt;-</span> mus_chrs[i]</span>
<span id="cb303-5"><a href="#cb303-5" aria-hidden="true" tabindex="-1"></a>  chr_genes <span class="ot">&lt;-</span> gtf_tpm[<span class="fu">seqnames</span>(gtf_tpm)<span class="sc">==</span>nth_chr]</span>
<span id="cb303-6"><a href="#cb303-6" aria-hidden="true" tabindex="-1"></a>  bin_size <span class="ot">&lt;-</span> <span class="fl">1e6</span></span>
<span id="cb303-7"><a href="#cb303-7" aria-hidden="true" tabindex="-1"></a>  chr_length <span class="ot">&lt;-</span> <span class="fu">width</span>(custom.genome)[i]</span>
<span id="cb303-8"><a href="#cb303-8" aria-hidden="true" tabindex="-1"></a>  bins <span class="ot">&lt;-</span> <span class="fu">tileGenome</span>(<span class="at">seqlengths =</span> <span class="fu">setNames</span>(chr_length, nth_chr),</span>
<span id="cb303-9"><a href="#cb303-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">tilewidth =</span> bin_size, </span>
<span id="cb303-10"><a href="#cb303-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">cut.last.tile.in.chrom =</span> <span class="cn">TRUE</span>)</span>
<span id="cb303-11"><a href="#cb303-11" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb303-12"><a href="#cb303-12" aria-hidden="true" tabindex="-1"></a>  expr_list <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>()</span>
<span id="cb303-13"><a href="#cb303-13" aria-hidden="true" tabindex="-1"></a>  kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">genome=</span>custom.genome, <span class="at">chromosomes=</span>nth_chr, <span class="at">cytobands=</span>cyto, <span class="at">plot.type=</span><span class="dv">4</span>)</span>
<span id="cb303-14"><a href="#cb303-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(sample <span class="cf">in</span> interest){</span>
<span id="cb303-15"><a href="#cb303-15" aria-hidden="true" tabindex="-1"></a>      hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(bins, chr_genes) </span>
<span id="cb303-16"><a href="#cb303-16" aria-hidden="true" tabindex="-1"></a>      sum_expr <span class="ot">&lt;-</span> <span class="fu">tapply</span>(<span class="fu">mcols</span>(chr_genes)[[sample]][<span class="fu">subjectHits</span>(hits)],</span>
<span id="cb303-17"><a href="#cb303-17" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">queryHits</span>(hits), sum, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb303-18"><a href="#cb303-18" aria-hidden="true" tabindex="-1"></a>      expr_bins <span class="ot">&lt;-</span> bins[<span class="fu">as.numeric</span>(<span class="fu">names</span>(sum_expr))]</span>
<span id="cb303-19"><a href="#cb303-19" aria-hidden="true" tabindex="-1"></a>      expr_bins<span class="sc">$</span>expr <span class="ot">&lt;-</span> sum_expr</span>
<span id="cb303-20"><a href="#cb303-20" aria-hidden="true" tabindex="-1"></a>      expr_list[[sample]] <span class="ot">&lt;-</span> expr_bins</span>
<span id="cb303-21"><a href="#cb303-21" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb303-22"><a href="#cb303-22" aria-hidden="true" tabindex="-1"></a>  max_y <span class="ot">=</span> <span class="fu">max</span>(expr_list[[interest[<span class="dv">1</span>]]]<span class="sc">$</span>expr, expr_list[[interest[<span class="dv">2</span>]]]<span class="sc">$</span>expr)</span>
<span id="cb303-23"><a href="#cb303-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpArea</span>(kp, <span class="at">data=</span>expr_list[[interest[<span class="dv">1</span>]]], <span class="at">y=</span>expr_list[[interest[<span class="dv">1</span>]]]<span class="sc">$</span>expr, <span class="at">ymax =</span> max_y <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col =</span> <span class="fu">rgb</span>(<span class="dv">34</span><span class="sc">/</span><span class="dv">255</span>, <span class="dv">139</span><span class="sc">/</span><span class="dv">255</span>, <span class="dv">34</span><span class="sc">/</span><span class="dv">255</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>), </span>
<span id="cb303-24"><a href="#cb303-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">border=</span><span class="cn">NA</span>, <span class="at">r0 =</span> r, <span class="at">r1 =</span> r <span class="sc">+</span> <span class="fl">0.4</span>)</span>
<span id="cb303-25"><a href="#cb303-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpLines</span>(kp, <span class="at">chr=</span>nth_chr, <span class="at">data =</span> expr_list[[interest[<span class="dv">1</span>]]], <span class="at">y=</span>expr_list[[interest[<span class="dv">1</span>]]]<span class="sc">$</span>expr, <span class="at">ymin=</span> <span class="dv">0</span>, <span class="at">ymax =</span> max_y <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col=</span><span class="st">"forestgreen"</span>, </span>
<span id="cb303-26"><a href="#cb303-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">r0 =</span> r, </span>
<span id="cb303-27"><a href="#cb303-27" aria-hidden="true" tabindex="-1"></a>          <span class="at">r1 =</span> r <span class="sc">+</span> <span class="fl">0.4</span>)</span>
<span id="cb303-28"><a href="#cb303-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpPoints</span>(kp, <span class="at">data=</span>expr_list[[interest[<span class="dv">1</span>]]], <span class="at">y=</span>expr_list[[interest[<span class="dv">1</span>]]]<span class="sc">$</span>expr, <span class="at">ymin=</span> <span class="dv">0</span>, <span class="at">ymax =</span> max_y <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col=</span><span class="st">"blue"</span>, </span>
<span id="cb303-29"><a href="#cb303-29" aria-hidden="true" tabindex="-1"></a>           <span class="at">cex=</span><span class="fl">0.5</span>, <span class="at">r0 =</span> r, <span class="at">r1 =</span> r <span class="sc">+</span> <span class="fl">0.4</span>)</span>
<span id="cb303-30"><a href="#cb303-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> <span class="fu">substr</span>(interest[<span class="dv">1</span>], <span class="fu">nchar</span>(interest[<span class="dv">1</span>])<span class="sc">-</span><span class="dv">1</span>, <span class="fu">nchar</span>(interest[<span class="dv">1</span>])), <span class="at">r0 =</span> r, </span>
<span id="cb303-31"><a href="#cb303-31" aria-hidden="true" tabindex="-1"></a>              <span class="at">r1 =</span> r <span class="sc">+</span> <span class="fl">0.4</span>, <span class="at">cex =</span> <span class="fl">0.7</span>, <span class="at">srt =</span> <span class="dv">0</span>, <span class="at">label.margin =</span> <span class="fl">0.01</span>)</span>
<span id="cb303-32"><a href="#cb303-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb303-33"><a href="#cb303-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpArea</span>(kp, <span class="at">data=</span>expr_list[[interest[<span class="dv">2</span>]]], <span class="at">y=</span>expr_list[[interest[<span class="dv">2</span>]]]<span class="sc">$</span>expr, <span class="at">ymax =</span> max_y <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col =</span> <span class="fu">rgb</span>(<span class="dv">34</span><span class="sc">/</span><span class="dv">255</span>, <span class="dv">139</span><span class="sc">/</span><span class="dv">255</span>, <span class="dv">34</span><span class="sc">/</span><span class="dv">255</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>), </span>
<span id="cb303-34"><a href="#cb303-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">border=</span><span class="cn">NA</span>, <span class="at">r0 =</span> <span class="fl">0.6</span>, <span class="at">r1 =</span> <span class="dv">1</span>)</span>
<span id="cb303-35"><a href="#cb303-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpLines</span>(kp, <span class="at">chr=</span>nth_chr, <span class="at">data =</span> expr_list[[interest[<span class="dv">2</span>]]], <span class="at">y=</span>expr_list[[interest[<span class="dv">2</span>]]]<span class="sc">$</span>expr, <span class="at">ymin=</span> <span class="dv">0</span>, <span class="at">ymax =</span> max_y <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col=</span><span class="st">"forestgreen"</span>, </span>
<span id="cb303-36"><a href="#cb303-36" aria-hidden="true" tabindex="-1"></a>          <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">r0 =</span> <span class="fl">0.6</span>, </span>
<span id="cb303-37"><a href="#cb303-37" aria-hidden="true" tabindex="-1"></a>          <span class="at">r1 =</span> <span class="dv">1</span>)</span>
<span id="cb303-38"><a href="#cb303-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpPoints</span>(kp, <span class="at">data=</span>expr_list[[interest[<span class="dv">2</span>]]], <span class="at">y=</span>expr_list[[interest[<span class="dv">2</span>]]]<span class="sc">$</span>expr, <span class="at">ymin=</span> <span class="dv">0</span>, <span class="at">ymax =</span> max_y <span class="sc">*</span> <span class="fl">1.1</span>, <span class="at">col=</span><span class="st">"blue"</span>, </span>
<span id="cb303-39"><a href="#cb303-39" aria-hidden="true" tabindex="-1"></a>           <span class="at">cex=</span><span class="fl">0.5</span>, <span class="at">r0 =</span> <span class="fl">0.6</span>, <span class="at">r1 =</span> <span class="dv">1</span>)</span>
<span id="cb303-40"><a href="#cb303-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> <span class="fu">substr</span>(interest[<span class="dv">2</span>], <span class="fu">nchar</span>(interest[<span class="dv">2</span>])<span class="sc">-</span><span class="dv">1</span>, <span class="fu">nchar</span>(interest[<span class="dv">2</span>])), <span class="at">r0 =</span> <span class="fl">0.6</span>, </span>
<span id="cb303-41"><a href="#cb303-41" aria-hidden="true" tabindex="-1"></a>              <span class="at">r1 =</span> <span class="dv">1</span>, <span class="at">cex =</span> <span class="fl">0.7</span>, <span class="at">srt =</span> <span class="dv">0</span>, <span class="at">label.margin =</span> <span class="fl">0.01</span>)</span>
<span id="cb303-42"><a href="#cb303-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpAxis</span>(kp, <span class="at">ymin =</span> <span class="dv">0</span>, <span class="at">ymax =</span> max_y, <span class="at">r0 =</span> <span class="dv">0</span>, <span class="at">r1 =</span> <span class="fl">0.4</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">numticks =</span> <span class="dv">4</span>)</span>
<span id="cb303-43"><a href="#cb303-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpAxis</span>(kp, <span class="at">ymin =</span> <span class="dv">0</span>, <span class="at">ymax =</span> max_y, <span class="at">r0 =</span> <span class="fl">0.6</span>, <span class="at">r1 =</span> <span class="dv">1</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">numticks =</span> <span class="dv">4</span>)</span>
<span id="cb303-44"><a href="#cb303-44" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb303-45"><a href="#cb303-45" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-11.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-12.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-13.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-14.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-15.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-16.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-17.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-18.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-19.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-20.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-94-21.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="ttdegs-gene-density" class="level3">
<h3 class="anchored" data-anchor-id="ttdegs-gene-density">TTDegs &amp; Gene Density</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb304"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb304-1"><a href="#cb304-1" aria-hidden="true" tabindex="-1"></a>TTdegs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rownames</span>(TTBowtieDownRegSorted), <span class="fu">rownames</span>(TTBowtieUpRegSorted))</span>
<span id="cb304-2"><a href="#cb304-2" aria-hidden="true" tabindex="-1"></a>TT_genes <span class="ot">&lt;-</span> gene_features[gene_features<span class="sc">$</span>gene_id <span class="sc">%in%</span> TTdegs]</span>
<span id="cb304-3"><a href="#cb304-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb304-4"><a href="#cb304-4" aria-hidden="true" tabindex="-1"></a>kpTr <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb304-5"><a href="#cb304-5" aria-hidden="true" tabindex="-1"></a>kpTr <span class="ot">&lt;-</span> <span class="fu">kpPlotDensity</span>(kpTr, <span class="at">window.size =</span> <span class="fl">0.5e6</span>, <span class="at">data=</span> TT_genes, <span class="at">data.panel=</span><span class="dv">1</span>, <span class="at">col =</span> <span class="st">"forestgreen"</span>, <span class="at">border=</span><span class="st">"forestgreen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-95-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb305"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb305-1"><a href="#cb305-1" aria-hidden="true" tabindex="-1"></a>density_dataTr <span class="ot">&lt;-</span> kpTr<span class="sc">$</span>latest.plot<span class="sc">$</span>computed.values</span>
<span id="cb305-2"><a href="#cb305-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb305-3"><a href="#cb305-3" aria-hidden="true" tabindex="-1"></a>kpGene <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb305-4"><a href="#cb305-4" aria-hidden="true" tabindex="-1"></a>kpGene <span class="ot">&lt;-</span> <span class="fu">kpPlotDensity</span>(kpGene, <span class="at">window.size =</span> <span class="fl">0.5e6</span>, <span class="at">data=</span> gene_features, <span class="at">data.panel=</span><span class="dv">2</span>, <span class="at">col =</span> <span class="st">"steelblue"</span>, <span class="at">border=</span><span class="st">"steelblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-95-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb306"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb306-1"><a href="#cb306-1" aria-hidden="true" tabindex="-1"></a>density_dataGene <span class="ot">&lt;-</span> kpGene<span class="sc">$</span>latest.plot<span class="sc">$</span>computed.values</span>
<span id="cb306-2"><a href="#cb306-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb306-3"><a href="#cb306-3" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> density_dataTr<span class="sc">$</span>density<span class="sc">/</span>density_dataGene<span class="sc">$</span>density</span>
<span id="cb306-4"><a href="#cb306-4" aria-hidden="true" tabindex="-1"></a>y[<span class="fu">is.nan</span>(y)] <span class="ot">&lt;-</span> <span class="cn">NA</span>  </span>
<span id="cb306-5"><a href="#cb306-5" aria-hidden="true" tabindex="-1"></a>y[<span class="fu">is.infinite</span>(y)] <span class="ot">&lt;-</span> <span class="cn">NA</span>  </span>
<span id="cb306-6"><a href="#cb306-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb306-7"><a href="#cb306-7" aria-hidden="true" tabindex="-1"></a>ratio_gr <span class="ot">&lt;-</span> density_dataGene<span class="sc">$</span>windows</span>
<span id="cb306-8"><a href="#cb306-8" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(ratio_gr)<span class="sc">$</span>score <span class="ot">&lt;-</span> y</span>
<span id="cb306-9"><a href="#cb306-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb306-10"><a href="#cb306-10" aria-hidden="true" tabindex="-1"></a>KpGeTR <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb306-11"><a href="#cb306-11" aria-hidden="true" tabindex="-1"></a><span class="fu">kpLines</span>(KpGeTR, <span class="at">data=</span>ratio_gr, <span class="at">y=</span>y, <span class="at">col=</span><span class="st">"darkred"</span>, <span class="at">lwd=</span><span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-95-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb307"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb307-1"><a href="#cb307-1" aria-hidden="true" tabindex="-1"></a>TTdegsC1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rownames</span>(TTC1RemovedUpRegSorted.G), <span class="fu">rownames</span>(TTC1RemovedDownRegSorted.G))</span>
<span id="cb307-2"><a href="#cb307-2" aria-hidden="true" tabindex="-1"></a>TT_genesC1 <span class="ot">&lt;-</span> gene_features[gene_features<span class="sc">$</span>gene_id <span class="sc">%in%</span> TTdegsC1]</span>
<span id="cb307-3"><a href="#cb307-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-4"><a href="#cb307-4" aria-hidden="true" tabindex="-1"></a>kpTr <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb307-5"><a href="#cb307-5" aria-hidden="true" tabindex="-1"></a>kpTr <span class="ot">&lt;-</span> <span class="fu">kpPlotDensity</span>(kpTr, <span class="at">window.size =</span> <span class="fl">0.5e6</span>, <span class="at">data=</span> TT_genesC1, <span class="at">data.panel=</span><span class="dv">1</span>, <span class="at">col =</span> <span class="st">"forestgreen"</span>, <span class="at">border=</span><span class="st">"forestgreen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-96-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb308"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb308-1"><a href="#cb308-1" aria-hidden="true" tabindex="-1"></a>density_dataTr <span class="ot">&lt;-</span> kpTr<span class="sc">$</span>latest.plot<span class="sc">$</span>computed.values</span>
<span id="cb308-2"><a href="#cb308-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb308-3"><a href="#cb308-3" aria-hidden="true" tabindex="-1"></a>kpGene <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb308-4"><a href="#cb308-4" aria-hidden="true" tabindex="-1"></a>kpGene <span class="ot">&lt;-</span> <span class="fu">kpPlotDensity</span>(kpGene, <span class="at">window.size =</span> <span class="fl">0.5e6</span>, <span class="at">data=</span> gene_features, <span class="at">data.panel=</span><span class="dv">2</span>, <span class="at">col =</span> <span class="st">"steelblue"</span>, <span class="at">border=</span><span class="st">"steelblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-96-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb309"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb309-1"><a href="#cb309-1" aria-hidden="true" tabindex="-1"></a>density_dataGene <span class="ot">&lt;-</span> kpGene<span class="sc">$</span>latest.plot<span class="sc">$</span>computed.values</span>
<span id="cb309-2"><a href="#cb309-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-3"><a href="#cb309-3" aria-hidden="true" tabindex="-1"></a>yC1 <span class="ot">&lt;-</span> density_dataTr<span class="sc">$</span>density<span class="sc">/</span>density_dataGene<span class="sc">$</span>density</span>
<span id="cb309-4"><a href="#cb309-4" aria-hidden="true" tabindex="-1"></a>yC1[<span class="fu">is.nan</span>(yC1)] <span class="ot">&lt;-</span> <span class="cn">NA</span>  </span>
<span id="cb309-5"><a href="#cb309-5" aria-hidden="true" tabindex="-1"></a>yC1[<span class="fu">is.infinite</span>(yC1)] <span class="ot">&lt;-</span> <span class="cn">NA</span>  </span>
<span id="cb309-6"><a href="#cb309-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-7"><a href="#cb309-7" aria-hidden="true" tabindex="-1"></a>ratio_grC1 <span class="ot">&lt;-</span> density_dataGene<span class="sc">$</span>windows</span>
<span id="cb309-8"><a href="#cb309-8" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(ratio_grC1)<span class="sc">$</span>score <span class="ot">&lt;-</span> yC1</span>
<span id="cb309-9"><a href="#cb309-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-10"><a href="#cb309-10" aria-hidden="true" tabindex="-1"></a>KpGeTRc1 <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb309-11"><a href="#cb309-11" aria-hidden="true" tabindex="-1"></a><span class="fu">kpLines</span>(KpGeTRc1, <span class="at">data=</span>ratio_grC1, <span class="at">y=</span>yC1, <span class="at">col=</span><span class="st">"darkred"</span>, <span class="at">lwd=</span><span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-96-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb310"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb310-1"><a href="#cb310-1" aria-hidden="true" tabindex="-1"></a>peaksC1 <span class="ot">&lt;-</span> <span class="fu">GRanges</span>()</span>
<span id="cb310-2"><a href="#cb310-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (chr <span class="cf">in</span> <span class="fu">runValue</span>(<span class="fu">seqnames</span>(ratio_grC1))){</span>
<span id="cb310-3"><a href="#cb310-3" aria-hidden="true" tabindex="-1"></a>  subset_gtf <span class="ot">&lt;-</span> ratio_grC1[<span class="fu">seqnames</span>(ratio_grC1)<span class="sc">==</span>chr]</span>
<span id="cb310-4"><a href="#cb310-4" aria-hidden="true" tabindex="-1"></a>  score_vec <span class="ot">&lt;-</span> subset_gtf[<span class="sc">!</span><span class="fu">is.na</span>(subset_gtf<span class="sc">$</span>score)]</span>
<span id="cb310-5"><a href="#cb310-5" aria-hidden="true" tabindex="-1"></a>  min_value <span class="ot">&lt;-</span> <span class="fl">0.01</span>  </span>
<span id="cb310-6"><a href="#cb310-6" aria-hidden="true" tabindex="-1"></a>  nonzero_scores <span class="ot">&lt;-</span> score_vec[score_vec<span class="sc">$</span>score <span class="sc">&gt;</span> min_value]</span>
<span id="cb310-7"><a href="#cb310-7" aria-hidden="true" tabindex="-1"></a>  quantile_cutoff <span class="ot">&lt;-</span> <span class="fl">0.90</span></span>
<span id="cb310-8"><a href="#cb310-8" aria-hidden="true" tabindex="-1"></a>  threshold <span class="ot">&lt;-</span> <span class="fu">quantile</span>(nonzero_scores<span class="sc">$</span>score,quantile_cutoff)</span>
<span id="cb310-9"><a href="#cb310-9" aria-hidden="true" tabindex="-1"></a>  peaksC1 <span class="ot">&lt;-</span> <span class="fu">c</span>(peaksC1, nonzero_scores[nonzero_scores<span class="sc">$</span>score <span class="sc">&gt;</span> threshold <span class="sc">&amp;</span> nonzero_scores<span class="sc">$</span>score <span class="sc">&gt;</span> min_value])</span>
<span id="cb310-10"><a href="#cb310-10" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb310-11"><a href="#cb310-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-12"><a href="#cb310-12" aria-hidden="true" tabindex="-1"></a>dummy <span class="ot">&lt;-</span> ratio_grC1</span>
<span id="cb310-13"><a href="#cb310-13" aria-hidden="true" tabindex="-1"></a>hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(dummy, peaksC1, <span class="at">type =</span> <span class="st">"start"</span>)</span>
<span id="cb310-14"><a href="#cb310-14" aria-hidden="true" tabindex="-1"></a>dummy<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb310-15"><a href="#cb310-15" aria-hidden="true" tabindex="-1"></a>dummy<span class="sc">$</span>score[<span class="fu">queryHits</span>(hits)] <span class="ot">&lt;-</span> peaksC1<span class="sc">$</span>score[<span class="fu">subjectHits</span>(hits)]</span>
<span id="cb310-16"><a href="#cb310-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-17"><a href="#cb310-17" aria-hidden="true" tabindex="-1"></a>KpGeTRdummy <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb310-18"><a href="#cb310-18" aria-hidden="true" tabindex="-1"></a><span class="fu">kpLines</span>(KpGeTRdummy, <span class="at">data=</span>dummy, <span class="at">y=</span>dummy<span class="sc">$</span>score, <span class="at">col=</span><span class="st">"darkgreen"</span>, <span class="at">lwd=</span><span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-97-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb311"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb311-1"><a href="#cb311-1" aria-hidden="true" tabindex="-1"></a>peaks_genes1 <span class="ot">&lt;-</span> <span class="fu">mcols</span>(<span class="fu">subsetByOverlaps</span>(TT_genesC1, peaksC1, <span class="at">type=</span><span class="st">"within"</span>))<span class="sc">$</span>gene_name</span>
<span id="cb311-2"><a href="#cb311-2" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(peaks_genes1, <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/peaks_potential_gene.txt"</span>)</span>
<span id="cb311-3"><a href="#cb311-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-4"><a href="#cb311-4" aria-hidden="true" tabindex="-1"></a>peaks <span class="ot">&lt;-</span> <span class="fu">GRanges</span>()</span>
<span id="cb311-5"><a href="#cb311-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (chr <span class="cf">in</span> <span class="fu">runValue</span>(<span class="fu">seqnames</span>(ratio_gr))){</span>
<span id="cb311-6"><a href="#cb311-6" aria-hidden="true" tabindex="-1"></a>  subset_gtf <span class="ot">&lt;-</span> ratio_gr[<span class="fu">seqnames</span>(ratio_gr)<span class="sc">==</span>chr]</span>
<span id="cb311-7"><a href="#cb311-7" aria-hidden="true" tabindex="-1"></a>  score_vec <span class="ot">&lt;-</span> subset_gtf[<span class="sc">!</span><span class="fu">is.na</span>(subset_gtf<span class="sc">$</span>score)]</span>
<span id="cb311-8"><a href="#cb311-8" aria-hidden="true" tabindex="-1"></a>  min_value <span class="ot">&lt;-</span> <span class="fl">0.01</span>  </span>
<span id="cb311-9"><a href="#cb311-9" aria-hidden="true" tabindex="-1"></a>  nonzero_scores <span class="ot">&lt;-</span> score_vec[score_vec<span class="sc">$</span>score <span class="sc">&gt;</span> min_value]</span>
<span id="cb311-10"><a href="#cb311-10" aria-hidden="true" tabindex="-1"></a>  quantile_cutoff <span class="ot">&lt;-</span> <span class="fl">0.90</span></span>
<span id="cb311-11"><a href="#cb311-11" aria-hidden="true" tabindex="-1"></a>  threshold <span class="ot">&lt;-</span> <span class="fu">quantile</span>(nonzero_scores<span class="sc">$</span>score,quantile_cutoff)</span>
<span id="cb311-12"><a href="#cb311-12" aria-hidden="true" tabindex="-1"></a>  peaks <span class="ot">&lt;-</span> <span class="fu">c</span>(peaks, nonzero_scores[nonzero_scores<span class="sc">$</span>score <span class="sc">&gt;</span> threshold <span class="sc">&amp;</span> nonzero_scores<span class="sc">$</span>score <span class="sc">&gt;</span> min_value])</span>
<span id="cb311-13"><a href="#cb311-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb311-14"><a href="#cb311-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-15"><a href="#cb311-15" aria-hidden="true" tabindex="-1"></a>dummy2 <span class="ot">&lt;-</span> ratio_gr</span>
<span id="cb311-16"><a href="#cb311-16" aria-hidden="true" tabindex="-1"></a>hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(dummy2, peaks, <span class="at">type =</span> <span class="st">"start"</span>)</span>
<span id="cb311-17"><a href="#cb311-17" aria-hidden="true" tabindex="-1"></a>dummy2<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb311-18"><a href="#cb311-18" aria-hidden="true" tabindex="-1"></a>dummy2<span class="sc">$</span>score[<span class="fu">queryHits</span>(hits)] <span class="ot">&lt;-</span> peaks<span class="sc">$</span>score[<span class="fu">subjectHits</span>(hits)]</span>
<span id="cb311-19"><a href="#cb311-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-20"><a href="#cb311-20" aria-hidden="true" tabindex="-1"></a>KpGeTRdummy <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb311-21"><a href="#cb311-21" aria-hidden="true" tabindex="-1"></a><span class="fu">kpLines</span>(KpGeTRdummy, <span class="at">data=</span>dummy2, <span class="at">y=</span>dummy2<span class="sc">$</span>score, <span class="at">col=</span><span class="st">"darkred"</span>, <span class="at">lwd=</span><span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-97-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb312"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb312-1"><a href="#cb312-1" aria-hidden="true" tabindex="-1"></a>peaks_genes2 <span class="ot">&lt;-</span> <span class="fu">mcols</span>(<span class="fu">subsetByOverlaps</span>(TT_genesC1, peaks, <span class="at">type=</span><span class="st">"within"</span>))<span class="sc">$</span>gene_name</span>
<span id="cb312-2"><a href="#cb312-2" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(peaks_genes2, <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/peaksC1_potential_gene.txt"</span>)</span>
<span id="cb312-3"><a href="#cb312-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-4"><a href="#cb312-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-5"><a href="#cb312-5" aria-hidden="true" tabindex="-1"></a>hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(peaks, peaksC1, <span class="at">type =</span> <span class="st">"start"</span>)</span>
<span id="cb312-6"><a href="#cb312-6" aria-hidden="true" tabindex="-1"></a>common_peaks <span class="ot">&lt;-</span> peaks[<span class="fu">queryHits</span>(hits)]</span>
<span id="cb312-7"><a href="#cb312-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-8"><a href="#cb312-8" aria-hidden="true" tabindex="-1"></a>dummy3 <span class="ot">&lt;-</span> ratio_gr</span>
<span id="cb312-9"><a href="#cb312-9" aria-hidden="true" tabindex="-1"></a>hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(dummy3, common_peaks, <span class="at">type =</span> <span class="st">"start"</span>)</span>
<span id="cb312-10"><a href="#cb312-10" aria-hidden="true" tabindex="-1"></a>dummy3<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb312-11"><a href="#cb312-11" aria-hidden="true" tabindex="-1"></a>dummy3<span class="sc">$</span>score[<span class="fu">queryHits</span>(hits)] <span class="ot">&lt;-</span> peaks<span class="sc">$</span>score[<span class="fu">subjectHits</span>(hits)]</span>
<span id="cb312-12"><a href="#cb312-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-13"><a href="#cb312-13" aria-hidden="true" tabindex="-1"></a>KpGeTRdummy <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">plot.type=</span><span class="dv">1</span>, <span class="at">genome =</span> custom.genome, <span class="at">cytobands =</span> cyto, <span class="at">chromosomes=</span>mus_chrs, <span class="at">cex=</span><span class="dv">1</span>)</span>
<span id="cb312-14"><a href="#cb312-14" aria-hidden="true" tabindex="-1"></a><span class="fu">kpLines</span>(KpGeTRdummy, <span class="at">data=</span>dummy3, <span class="at">y=</span>dummy3<span class="sc">$</span>score, <span class="at">col=</span><span class="st">"steelblue"</span>, <span class="at">lwd=</span><span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-97-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb313"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb313-1"><a href="#cb313-1" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">mcols</span>(<span class="fu">subsetByOverlaps</span>(TT_genesC1, common_peaks, <span class="at">type=</span><span class="st">"within"</span>))<span class="sc">$</span>gene_name</span>
<span id="cb313-2"><a href="#cb313-2" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(common_genes, <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/common_potential_gene.txt"</span>)</span>
<span id="cb313-3"><a href="#cb313-3" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(<span class="fu">setdiff</span>(peaks_genes2, common_genes), <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/only_peaksC1_potential_gene.txt"</span>)</span>
<span id="cb313-4"><a href="#cb313-4" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(<span class="fu">setdiff</span>(peaks_genes1, common_genes), <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/only_peaks_potential_gene.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb314"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb314-1"><a href="#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(RnaUpRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_STAR_UPREG.txt"</span>)</span>
<span id="cb314-2"><a href="#cb314-2" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(RnaDownRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_STAR_DOWNREG.txt"</span>)</span>
<span id="cb314-3"><a href="#cb314-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-4"><a href="#cb314-4" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TxiUpRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_SALMON_GENE_PREMRNA_UPREG.txt"</span>)</span>
<span id="cb314-5"><a href="#cb314-5" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TxiDownRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_SALMON_GENE_PREMRNA_DOWNREG.txt"</span>)</span>
<span id="cb314-6"><a href="#cb314-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-7"><a href="#cb314-7" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TxiTTUpRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_SALMON_PREMRNA_UPREG.txt"</span>)</span>
<span id="cb314-8"><a href="#cb314-8" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TxiTTDownRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_SALMON_PREMRNA_DOWNREG.txt"</span>)</span>
<span id="cb314-9"><a href="#cb314-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-10"><a href="#cb314-10" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TxiTTUpRegSortedC1), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_SALMON_PREMRNA_C1REMOVED_UPREG.txt"</span>)</span>
<span id="cb314-11"><a href="#cb314-11" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TxiTTDownRegSortedC1), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_SALMON_PREMRNA_C1REMOVED_DOWNREG.txt"</span>)</span>
<span id="cb314-12"><a href="#cb314-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-13"><a href="#cb314-13" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(RnaUpRegSortedC1), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_STAR_C1REMOVED_UPREG.txt"</span>)</span>
<span id="cb314-14"><a href="#cb314-14" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(RnaDownRegSortedC1), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/RNA_STAR_C1REMOVED_DOWNREG.txt"</span>)</span>
<span id="cb314-15"><a href="#cb314-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-16"><a href="#cb314-16" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TTUpRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/TT_STAR_UPREG.txt"</span>)</span>
<span id="cb314-17"><a href="#cb314-17" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TTDownRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/TT_STAR_DOWNREG.txt"</span>)</span>
<span id="cb314-18"><a href="#cb314-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-19"><a href="#cb314-19" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TTBowtieUpRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/TT_BOWTIE2_UPREG.txt"</span>)</span>
<span id="cb314-20"><a href="#cb314-20" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TTBowtieDownRegSorted), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/TT_BOWTIE2_DOWNREG.txt"</span>)</span>
<span id="cb314-21"><a href="#cb314-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-22"><a href="#cb314-22" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TTC1RemovedUpRegSorted.G), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/TT_BOWTIE2_C1REMOVED_GENOTYPE_UPREG.txt"</span>)</span>
<span id="cb314-23"><a href="#cb314-23" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TTC1RemovedDownRegSorted.G), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/TT_BOWTIE2_C1REMOVED_GENOTYPE_DOWNREG.txt"</span>)</span>
<span id="cb314-24"><a href="#cb314-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-25"><a href="#cb314-25" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TTC1RemovedUpRegSorted.GR), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/TT_BOWTIE2_C1REMOVED_GENOTYPE_REPLICATE_UPREG.txt"</span>)</span>
<span id="cb314-26"><a href="#cb314-26" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(MusGeneNames<span class="sc">$</span>gene_name[<span class="fu">match</span>(<span class="fu">rownames</span>(TTC1RemovedDownRegSorted.GR), MusGeneNames<span class="sc">$</span>gene_id)], <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/TT_BOWTIE2_C1REMOVED_GENOTYPE_REPLICATE_DOWNREG.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb315"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb315-1"><a href="#cb315-1" aria-hidden="true" tabindex="-1"></a>cyto12 <span class="ot">&lt;-</span> cyto[<span class="fu">seqnames</span>(cyto)<span class="sc">==</span><span class="st">"chr12"</span>]</span>
<span id="cb315-2"><a href="#cb315-2" aria-hidden="true" tabindex="-1"></a>chr12_gene_name <span class="ot">&lt;-</span> <span class="fu">mcols</span>(<span class="fu">subsetByOverlaps</span>(TT_genes, cyto12[<span class="fu">length</span>(cyto12)<span class="sc">-</span><span class="dv">1</span>], <span class="at">type =</span> <span class="st">"within"</span>))<span class="sc">$</span>gene_name</span>
<span id="cb315-3"><a href="#cb315-3" aria-hidden="true" tabindex="-1"></a><span class="fu">writeLines</span>(chr12_gene_name, <span class="st">"/home/franck/SCRATCH/fpan/Data/Analysis/GENE_LIST/chr12_TT_Bowtie_potential_gene.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="limma" class="level2">
<h2 class="anchored" data-anchor-id="limma">limma</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb316"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb316-1"><a href="#cb316-1" aria-hidden="true" tabindex="-1"></a>all_mus_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(MusGeneNames<span class="sc">$</span>gene_id)</span>
<span id="cb316-2"><a href="#cb316-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-3"><a href="#cb316-3" aria-hidden="true" tabindex="-1"></a>rnaC1_vs_ttC1_up <span class="ot">&lt;-</span> <span class="fu">cbind</span>(</span>
<span id="cb316-4"><a href="#cb316-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">RNA =</span> all_mus_genes <span class="sc">%in%</span> <span class="fu">rownames</span>(RnaUpRegSortedC1),</span>
<span id="cb316-5"><a href="#cb316-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">TT =</span> all_mus_genes <span class="sc">%in%</span> <span class="fu">rownames</span>(TTC1RemovedUpRegSorted.G)</span>
<span id="cb316-6"><a href="#cb316-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb316-7"><a href="#cb316-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(rnaC1_vs_ttC1_up) <span class="ot">&lt;-</span> all_mus_genes</span>
<span id="cb316-8"><a href="#cb316-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-9"><a href="#cb316-9" aria-hidden="true" tabindex="-1"></a>rnaC1_vs_ttC1_down <span class="ot">&lt;-</span> <span class="fu">cbind</span>(</span>
<span id="cb316-10"><a href="#cb316-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">RNA =</span> all_mus_genes <span class="sc">%in%</span> <span class="fu">rownames</span>(RnaDownRegSortedC1),</span>
<span id="cb316-11"><a href="#cb316-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">TT =</span> all_mus_genes <span class="sc">%in%</span> <span class="fu">rownames</span>(TTC1RemovedDownRegSorted.G)</span>
<span id="cb316-12"><a href="#cb316-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb316-13"><a href="#cb316-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(rnaC1_vs_ttC1_down) <span class="ot">&lt;-</span> all_mus_genes</span>
<span id="cb316-14"><a href="#cb316-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-15"><a href="#cb316-15" aria-hidden="true" tabindex="-1"></a>rna_only <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rownames</span>(rnaC1_vs_ttC1_up[rnaC1_vs_ttC1_up[,<span class="dv">1</span>] <span class="sc">&amp;</span> <span class="sc">!</span>rnaC1_vs_ttC1_up[,<span class="dv">2</span>],]),</span>
<span id="cb316-16"><a href="#cb316-16" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rownames</span>(rnaC1_vs_ttC1_down[rnaC1_vs_ttC1_down[,<span class="dv">1</span>] <span class="sc">&amp;</span> <span class="sc">!</span>rnaC1_vs_ttC1_down[,<span class="dv">2</span>],]))</span>
<span id="cb316-17"><a href="#cb316-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-18"><a href="#cb316-18" aria-hidden="true" tabindex="-1"></a>tt_only <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rownames</span>(rnaC1_vs_ttC1_up[<span class="sc">!</span>rnaC1_vs_ttC1_up[,<span class="dv">1</span>] <span class="sc">&amp;</span> rnaC1_vs_ttC1_up[,<span class="dv">2</span>],]),</span>
<span id="cb316-19"><a href="#cb316-19" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rownames</span>(rnaC1_vs_ttC1_down[<span class="sc">!</span>rnaC1_vs_ttC1_down[,<span class="dv">1</span>] <span class="sc">&amp;</span> rnaC1_vs_ttC1_down[,<span class="dv">2</span>],]))</span>
<span id="cb316-20"><a href="#cb316-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-21"><a href="#cb316-21" aria-hidden="true" tabindex="-1"></a>both_present <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rownames</span>(rnaC1_vs_ttC1_up[rnaC1_vs_ttC1_up[,<span class="dv">1</span>] <span class="sc">&amp;</span> rnaC1_vs_ttC1_up[,<span class="dv">2</span>],]),</span>
<span id="cb316-22"><a href="#cb316-22" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rownames</span>(rnaC1_vs_ttC1_down[rnaC1_vs_ttC1_down[,<span class="dv">1</span>] <span class="sc">&amp;</span> rnaC1_vs_ttC1_down[,<span class="dv">2</span>],]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb317"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb317-1"><a href="#cb317-1" aria-hidden="true" tabindex="-1"></a>vc_up <span class="ot">&lt;-</span> <span class="fu">vennCounts</span>(rnaC1_vs_ttC1_up)</span>
<span id="cb317-2"><a href="#cb317-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(vc_up)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  RNA TT Counts
1   0  0  53333
2   0  1   1494
3   1  0    218
4   1  1    442
attr(,"class")
[1] "VennCounts"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb319"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb319-1"><a href="#cb319-1" aria-hidden="true" tabindex="-1"></a>vc_down <span class="ot">&lt;-</span> <span class="fu">vennCounts</span>(rnaC1_vs_ttC1_down)</span>
<span id="cb319-2"><a href="#cb319-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(vc_down)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  RNA TT Counts
1   0  0  55217
2   0  1    260
3   1  0      7
4   1  1      3
attr(,"class")
[1] "VennCounts"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb321"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb321-1"><a href="#cb321-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vennDiagram</span>(vc_up)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-103-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb322"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb322-1"><a href="#cb322-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vennDiagram</span>(vc_down)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-103-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<section id="fold2change-rna-vs-tt-plot" class="level3">
<h3 class="anchored" data-anchor-id="fold2change-rna-vs-tt-plot">Fold2Change RNA vs TT Plot</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb323"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb323-1"><a href="#cb323-1" aria-hidden="true" tabindex="-1"></a>RnaTTmergedF2change <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="fu">data.frame</span>(RnaResNoNa), <span class="fu">data.frame</span>(TTResNoNa), <span class="at">by=</span><span class="st">"index"</span>, <span class="at">suffixes=</span><span class="fu">c</span>(<span class="st">"RNA"</span>, <span class="st">"TT"</span>))</span>
<span id="cb323-2"><a href="#cb323-2" aria-hidden="true" tabindex="-1"></a>RnaTTmergedF2change<span class="sc">$</span>Presence <span class="ot">&lt;-</span> <span class="st">"None"</span></span>
<span id="cb323-3"><a href="#cb323-3" aria-hidden="true" tabindex="-1"></a>RnaTTmergedF2change[RnaTTmergedF2change<span class="sc">$</span>index <span class="sc">%in%</span> rna_only,]<span class="sc">$</span>Presence <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb323-4"><a href="#cb323-4" aria-hidden="true" tabindex="-1"></a>RnaTTmergedF2change[RnaTTmergedF2change<span class="sc">$</span>index <span class="sc">%in%</span> tt_only,]<span class="sc">$</span>Presence <span class="ot">&lt;-</span> <span class="st">"TT"</span></span>
<span id="cb323-5"><a href="#cb323-5" aria-hidden="true" tabindex="-1"></a>RnaTTmergedF2change[RnaTTmergedF2change<span class="sc">$</span>index <span class="sc">%in%</span> both_present,]<span class="sc">$</span>Presence <span class="ot">&lt;-</span> <span class="st">"Both"</span></span>
<span id="cb323-6"><a href="#cb323-6" aria-hidden="true" tabindex="-1"></a>RnaTTmergedF2change<span class="sc">$</span>Presence <span class="ot">&lt;-</span> <span class="fu">factor</span>(RnaTTmergedF2change<span class="sc">$</span>Presence, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"None"</span>, <span class="st">"RNA"</span>, <span class="st">"TT"</span>, <span class="st">"Both"</span>))</span>
<span id="cb323-7"><a href="#cb323-7" aria-hidden="true" tabindex="-1"></a>RnaTTmergedF2change <span class="ot">&lt;-</span> RnaTTmergedF2change[<span class="fu">order</span>(RnaTTmergedF2change<span class="sc">$</span>Presence),]</span>
<span id="cb323-8"><a href="#cb323-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-9"><a href="#cb323-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(RnaTTmergedF2change, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChangeRNA, <span class="at">y =</span> log2FoldChangeTT, <span class="at">col =</span> Presence)) <span class="sc">+</span></span>
<span id="cb323-10"><a href="#cb323-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb323-11"><a href="#cb323-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"darkgrey"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb323-12"><a href="#cb323-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"darkgrey"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb323-13"><a href="#cb323-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"steelblue"</span>, <span class="st">"darkgreen"</span>, <span class="st">"darkred"</span>),</span>
<span id="cb323-14"><a href="#cb323-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"None"</span>, <span class="st">"RNA"</span>, <span class="st">"TT"</span>, <span class="st">"Both"</span>)) <span class="sc">+</span></span>
<span id="cb323-15"><a href="#cb323-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="quarto_files/figure-html/unnamed-chunk-104-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>